From 092086c2a40c18e2a3fe985e7fe9d1778b4d7af2 Mon Sep 17 00:00:00 2001
From: Prashant S Chauhan <psinghchauha@vmware.com>
Date: Tue, 27 Jun 2023 02:59:19 +0000
Subject: [PATCH] Fix compilation error

---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 7d3c39151..4886dd78f 100644
--- a/Makefile
+++ b/Makefile
@@ -116,8 +116,8 @@ build_tools:
 	$(HOSTGO) build -o ./tools/readme_linter/readme_linter$(EXEEXT) ./tools/readme_linter
 
 embed_readme_%:
-	go generate -run="tools/config_includer/generator" ./plugins/$*/...
-	go generate -run="tools/readme_config_includer/generator" ./plugins/$*/...
+	go generate -run="tools/config_includer/generator.go" ./plugins/$*/...
+	go generate -run="tools/readme_config_includer/generator.go" ./plugins/$*/...
 
 .PHONY: config
 config:
-- 
2.39.0


