From f2b53f6766f8133c37b60b2c4bacc5e0dfca0a07 Mon Sep 17 00:00:00 2001
From: Clark Williams <williams@redhat.com>
Date: Fri, 29 Oct 2021 09:49:56 -0500
Subject: [PATCH 340/345] Linux 4.19.214-rt93 REBASE

Signed-off-by: Clark Williams <williams@redhat.com>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index bd5aee366a48..e98a1fe050bd 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt87
+-rt93
-- 
2.34.1

