diff --git a/src/java/org/apache/cassandra/utils/JMXServerUtils.java b/src/java/org/apache/cassandra/utils/JMXServerUtils.java
index e78ed01746..072f237049 100644
--- a/src/java/org/apache/cassandra/utils/JMXServerUtils.java
+++ b/src/java/org/apache/cassandra/utils/JMXServerUtils.java
@@ -46,6 +46,7 @@ import org.slf4j.LoggerFactory;
 import com.sun.jmx.remote.internal.RMIExporter;
 import com.sun.jmx.remote.security.JMXPluggableAuthenticator;
 import org.apache.cassandra.auth.jmx.AuthenticationProxy;
+import sun.misc.ObjectInputFilter;
 import sun.rmi.registry.RegistryImpl;
 import sun.rmi.server.UnicastServerRef2;

@@ -308,10 +309,10 @@ public class JMXServerUtils
         // to our custom Registry too.
         private Remote connectorServer;

-        public Remote exportObject(Remote obj, int port, RMIClientSocketFactory csf, RMIServerSocketFactory ssf)
+        public Remote exportObject(Remote obj, int port, RMIClientSocketFactory csf, RMIServerSocketFactory ssf, ObjectInputFilter filter)
         throws RemoteException
         {
-            Remote remote = new UnicastServerRef2(port, csf, ssf).exportObject(obj, null, true);
+            Remote remote = new UnicastServerRef2(port, csf, ssf, filter).exportObject(obj, null, true);
             // Keep a reference to the first object exported, the JMXConnectorServer
             if (connectorServer == null)
                 connectorServer = remote;
