From 500e8ecd945aaee434d8933858a85a260137ebf3 Mon Sep 17 00:00:00 2001
From: Susant Sahani <susant.sahani@broadcom.com>
Date: Tue, 2 Jan 2024 15:01:58 +0000
Subject: [PATCH 2/3] Fetch dns servers from environment

---
 src/resolve/resolved-conf.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/resolve/resolved-conf.c b/src/resolve/resolved-conf.c
index 2f08ed0..4b1a85b 100644
--- a/src/resolve/resolved-conf.c
+++ b/src/resolve/resolved-conf.c
@@ -580,7 +580,7 @@ int manager_parse_config_file(Manager *m) {
         read_proc_cmdline(m);  /* … but kernel command line overrides local configuration. */
 
         if (m->need_builtin_fallbacks) {
-                r = manager_parse_dns_server_string_and_warn(m, DNS_SERVER_FALLBACK, DNS_SERVERS);
+                r = manager_parse_dns_server_string_and_warn(m, DNS_SERVER_FALLBACK, secure_getenv("DEFAULT_DNS_SERVERS") ? : DNS_SERVERS);
                 if (r < 0)
                         return r;
         }
-- 
2.39.0

