diff -rupN systemd-239/src/basic/parse-util.h systemd-239-safe-atou32-full/src/basic/parse-util.h
--- systemd-239/src/basic/parse-util.h	2018-06-22 11:11:49.000000000 +0000
+++ systemd-239-safe-atou32-full/src/basic/parse-util.h	2020-06-06 03:14:19.879278652 +0000
@@ -50,9 +50,13 @@ static inline int safe_atoux16(const cha
 
 int safe_atoi16(const char *s, int16_t *ret);
 
-static inline int safe_atou32(const char *s, uint32_t *ret_u) {
+static inline int safe_atou32_full(const char *s, unsigned base, uint32_t *ret_u) {
         assert_cc(sizeof(uint32_t) == sizeof(unsigned));
-        return safe_atou(s, (unsigned*) ret_u);
+        return safe_atou_full(s, base, (unsigned*) ret_u);
+}
+
+static inline int safe_atou32(const char *s, uint32_t *ret_u) {
+        return safe_atou32_full(s, 0, (unsigned*) ret_u);
 }
 
 static inline int safe_atoi32(const char *s, int32_t *ret_i) {
