diff --color -rupN systemd-stable-249/src/resolve/resolved-conf.c systemd-stable-249-dns/src/resolve/resolved-conf.c
--- systemd-stable-249/src/resolve/resolved-conf.c	2021-07-07 17:41:29.000000000 +0000
+++ systemd-stable-249-dns/src/resolve/resolved-conf.c	2021-07-12 14:06:49.205284151 +0000
@@ -493,7 +493,7 @@ int manager_parse_config_file(Manager *m
                 return r;
 
         if (m->need_builtin_fallbacks) {
-                r = manager_parse_dns_server_string_and_warn(m, DNS_SERVER_FALLBACK, DNS_SERVERS);
+                r = manager_parse_dns_server_string_and_warn(m, DNS_SERVER_FALLBACK, secure_getenv("DEFAULT_DNS_SERVERS") ?: DNS_SERVERS);
                 if (r < 0)
                         return r;
         }
