diff -uNr systemd-228/src/resolve/resolved-manager.c systemd-228-new/src/resolve/resolved-manager.c
--- systemd-228/src/resolve/resolved-manager.c	2016-06-16 15:16:30.974053938 +0000
+++ systemd-228-new/src/resolve/resolved-manager.c	2016-06-16 19:44:52.102821700 +0000
@@ -463,6 +463,7 @@
 int manager_new(Manager **ret) {
         _cleanup_(manager_freep) Manager *m = NULL;
         int r;
+        char *default_dns_servers;
 
         assert(ret);
 
@@ -477,7 +478,10 @@
         m->llmnr_support = SUPPORT_YES;
         m->read_resolv_conf = true;
 
-        r = manager_parse_dns_server(m, DNS_SERVER_FALLBACK, DNS_SERVERS);
+        default_dns_servers = getenv("DEFAULT_DNS_SERVERS");
+        if (default_dns_servers == NULL)
+                default_dns_servers = DNS_SERVERS;
+        r = manager_parse_dns_server(m, DNS_SERVER_FALLBACK, default_dns_servers);
         if (r < 0)
                 return r;
 
