diff -rupN systemd-239/src/network/networkd-routing-policy-rule.c systemd-239-bz-2471962/src/network/networkd-routing-policy-rule.c
--- systemd-239/src/network/networkd-routing-policy-rule.c	2019-12-05 19:43:15.170930208 +0000
+++ systemd-239-bz-2471962/src/network/networkd-routing-policy-rule.c	2019-12-05 19:44:29.335113044 +0000
@@ -1024,6 +1024,26 @@ int routing_policy_load_rules(const char
         return 0;
 }
 
+static bool manager_links_have_routing_policy_rule(Manager *m, RoutingPolicyRule *rule) {
+        RoutingPolicyRule *link_rule;
+        Iterator i;
+        Link *link;
+
+        assert(m);
+        assert(rule);
+
+        HASHMAP_FOREACH(link, m->links, i) {
+                if (!link->network)
+                        continue;
+
+                LIST_FOREACH(rules, link_rule, link->network->rules)
+                        if (routing_policy_rule_compare_func(link_rule, rule) == 0)
+                                return true;
+        }
+
+        return false;
+}
+
 void routing_policy_rule_purge(Manager *m, Link *link) {
         RoutingPolicyRule *rule, *existing;
         Iterator i;
@@ -1034,15 +1054,24 @@ void routing_policy_rule_purge(Manager *
 
         SET_FOREACH(rule, m->rules_saved, i) {
                 existing = set_get(m->rules_foreign, rule);
-                if (existing) {
+                if (!existing) {
+
+                if (manager_links_have_routing_policy_rule(m, existing))
+                        continue; /* Existing links have the saved rule. */
+
+                /* Existing links do not have the saved rule. Let's drop the rule now, and re-configure it
+                 * later when it is requested. */
+
+                 r = routing_policy_rule_remove(existing, link, NULL);
+                 if (r < 0) {
+                         log_warning_errno(r, "Could not remove routing policy rules: %m");
+                         continue;
+                 }
 
-                        r = routing_policy_rule_remove(rule, link, routing_policy_rule_remove_handler);
-                        if (r < 0) {
-                                log_warning_errno(r, "Could not remove routing policy rules: %m");
-                                continue;
-                        }
+                 link->routing_policy_rule_remove_messages++;
 
-                        link->routing_policy_rule_remove_messages++;
+                 assert_se(set_remove(m->rules_foreign, existing) == existing);
+                 routing_policy_rule_free(existing);
                 }
         }
 }
