diff -rupN systemd-233/src/network/networkd-link.c systemd-233-ipv6ll/src/network/networkd-link.c
--- systemd-233/src/network/networkd-link.c	2020-06-22 05:30:35.436768631 +0000
+++ systemd-233-ipv6ll/src/network/networkd-link.c	2020-06-22 05:39:47.645821515 +0000
@@ -235,6 +235,9 @@ static bool link_ipv6_accept_ra_enabled(
         if (!link->network)
                 return false;
 
+        if (!link_ipv6ll_enabled(link))
+                return false;
+
         /* If unset use system default (enabled if local forwarding is disabled.
          * disabled if local forwarding is enabled).
          * If set, ignore or enforce RA independent of local forwarding state.
