diff -rupN systemd-233/src/network/networkd-link.c systemd-233-bz-2257208/src/network/networkd-link.c
--- systemd-233/src/network/networkd-link.c	2019-10-09 14:40:32.000000000 +0200
+++ systemd-233-bz-2257208/src/network/networkd-link.c	2019-10-09 15:57:49.170658942 +0200
@@ -219,7 +219,7 @@ static bool link_ipv6_disabled(Link *lin
         }
 	if (streq(val, "0"))
                 return false;
-        
+
         return true;
 }
 
@@ -803,6 +803,11 @@ static int route_handler(sd_netlink *rtn
 }
 
 static int link_enter_set_routes(Link *link) {
+        enum {
+                PHASE_NON_GATEWAY, /* First phase: Routes without a gateway */
+                PHASE_GATEWAY,     /* Second phase: Routes with a gateway */
+                _PHASE_MAX
+        } phase;
         Route *rt;
         int r;
 
@@ -812,16 +817,19 @@ static int link_enter_set_routes(Link *l
 
         link_set_state(link, LINK_STATE_SETTING_ROUTES);
 
-        LIST_FOREACH(routes, rt, link->network->static_routes) {
-                r = route_configure(rt, link, route_handler);
-                if (r < 0) {
-                        log_link_warning_errno(link, r, "Could not set routes: %m");
-                        link_enter_failed(link);
-                        return r;
-                }
+        /* First add the routes that enable us to talk to gateways, then add in the others that need a gateway. */
+        for (phase = 0; phase < _PHASE_MAX; phase++)
+                LIST_FOREACH(routes, rt, link->network->static_routes) {
 
-                link->link_messages++;
-        }
+                        if (in_addr_is_null(rt->family, &rt->gw) != (phase == PHASE_NON_GATEWAY))
+                                continue;
+
+                        r = route_configure(rt, link, route_handler);
+                        if (r < 0)
+                                return log_link_warning_errno(link, r, "Could not set routes: %m");
+                        if (r > 0)
+                                link->link_messages++;
+                }
 
         if (link->link_messages == 0) {
                 link->static_configured = true;
