--- a/src/login/systemd-user.m4	2015-11-17 23:59:06.000000000 -0800
+++ b/src/login/systemd-user.m4	2016-05-17 13:42:15.575836209 -0700
@@ -2,10 +2,10 @@
 #
 # Used by systemd --user instances.
 
-account  include system-auth
+account  include system-account
 
 m4_ifdef(`HAVE_SELINUX',
 session  required pam_selinux.so close
 session  required pam_selinux.so nottys open
 )m4_dnl
-session  include system-auth
+session  include system-session
