From 52eb0b1c7202a54d735f1ee79c0477b8b50fb2e2 Mon Sep 17 00:00:00 2001
From: Nitesh <kunitesh@vmware.com>
Date: Wed, 10 Jul 2024 11:50:21 +0530
Subject: [PATCH] Fixed incorrect config source for IPv6 address

---
 src/json/network-json.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/json/network-json.c b/src/json/network-json.c
index 61ee73c..a4c3b26 100644
--- a/src/json/network-json.c
+++ b/src/json/network-json.c
@@ -1157,22 +1157,22 @@ static int json_array_to_ip(const json_object *obj, const int family, const int
                 }
         }
 
+        if (prefix > 0)
+                g_string_append_printf(v, "/%d", prefix);
+
         if (family == AF_INET6) {
                 _auto_cleanup_ IPAddress *addr = NULL;
                 int r;
 
-                r = parse_ipv6(v->str, &addr);
+                r = parse_ip_from_str(v->str, &addr);
                 if (r < 0)
                         return r;
 
-                r = ip_to_str(AF_INET6, addr, &ip);
+                r = ip_to_str_prefix(AF_INET6, addr, &ip);
                 if (r < 0)
                         return r;
         }
 
-        if (prefix > 0)
-                g_string_append_printf(v, "/%d", prefix);
-
         if (family == AF_INET6)
                 *ret = steal_ptr(ip);
         else {
-- 
2.34.1

