From 6e7a45e4d44e96973aec9ad35184de88f5cf7fb4 Mon Sep 17 00:00:00 2001
From: Daniel Wagner <wagi@monom.org>
Date: Sat, 26 Aug 2023 16:58:26 +0200
Subject: [PATCH 353/353] Linux 4.19.292-rt128 REBASE

Signed-off-by: Daniel Wagner <wagi@monom.org>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 1199ebade17b..6d2a676e2033 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt16
+-rt128
-- 
2.41.0

