From 3fd03ac6e87a89bd308d8d871cca66552a6de5c4 Mon Sep 17 00:00:00 2001
From: Brennan Lamoreaux <brennan.lamoreaux@broadcom.com>
Date: Fri, 11 Oct 2024 21:41:07 +0000
Subject: [PATCH] compile jitterentropy rng proxy

---
 crypto/Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/crypto/Makefile b/crypto/Makefile
index 4a9c18509..3112e926d 100644
--- a/crypto/Makefile
+++ b/crypto/Makefile
@@ -6,9 +6,10 @@
 obj-$(CONFIG_CRYPTO) += crypto.o
 crypto-y := api.o cipher.o compress.o

-obj-$(CONFIG_CRYPTO_JITTERENTROPY) += jitterentropy_rng.o jitterentropy-3.4.1/ jitterentropy-3.4.1/jitterentropy_canister_wrapper.o jitterentropy-3.4.1/jitterentropy_canister_wrapper_asm.o
+obj-$(CONFIG_CRYPTO_JITTERENTROPY) += jitterentropy_rng.o jitterentropy_rng_proxy.o jitterentropy-3.4.1/ jitterentropy-3.4.1/jitterentropy_canister_wrapper.o jitterentropy-3.4.1/jitterentropy_canister_wrapper_asm.o

 jitterentropy_rng-y := jitterentropy-kcapi.o
+jitterentropy_rng_proxy-y := jitterentropy_rng_proxy.o

 obj-$(CONFIG_CRYPTO_ENGINE) += crypto_engine.o
 obj-$(CONFIG_CRYPTO_FIPS) += fips.o
--
2.39.4

