--- po/xgettext_sh_org	2020-06-23 21:15:00.850038999 +0800
+++ po/xgettext_sh	2020-06-23 21:15:58.975454504 +0800
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 # xgettext_sh -- extracts translatable strings from a shell script
 # Copyright (C) 1999 Conectiva Consultoria e Desenvolvimento de Sistemas LTDA
@@ -18,10 +18,11 @@
 # Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 
 from sys import path
+
 if not '/usr/lib/conectiva/xgettext_sh' in path:
     path[:0] = ['/usr/lib/conectiva/xgettext_sh']
 
 from xgettext_sh import main
 
 if __name__ == '__main__':
-	main()
+    main()
