From 3ae802466dd95ba28ae2506c4848b7b9045576dd Mon Sep 17 00:00:00 2001
From: Clark Williams <clark.williams@gmail.com>
Date: Mon, 9 Mar 2026 10:09:53 -0500
Subject: [PATCH 69/69] Linux 6.1.166-rt61 REBASE

Signed-off-by: Clark Williams <clark.williams@gmail.com>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 8cdeb45db6e2..9b7de9345ef4 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt59
+-rt61
-- 
2.53.0

