diff -Naur a/crypto/fips_canister_wrapper.c b/crypto/fips_canister_wrapper.c
--- a/crypto/fips_canister_wrapper.c	1969-12-31 16:00:00.000000000 -0800
+++ b/crypto/fips_canister_wrapper.c	2021-01-25 14:40:41.321352042 -0800
@@ -0,0 +1,16 @@
+#include <linux/types.h>
+#include <linux/mm.h>
+#include <linux/slab_def.h>
+#include <linux/kasan.h>
+
+//void *kmem_cache_alloc(struct kmem_cache *, gfp_t flags) __assume_slab_alignment __malloc;
+void *kmem_cache_alloc(struct kmem_cache *, gfp_t flags) __malloc;
+
+void __attribute__((weak)) *kmem_cache_alloc_trace(struct kmem_cache *s,
+		gfp_t flags, size_t size)
+{
+	void *ret = kmem_cache_alloc(s, flags);
+
+	ret = kasan_kmalloc(s, ret, size, flags);
+	return ret;
+}
diff -Naur a/crypto/Makefile b/crypto/Makefile
--- a/crypto/Makefile	2021-01-25 14:40:18.669227758 -0800
+++ b/crypto/Makefile	2021-01-25 14:58:28.807368014 -0800
@@ -167,7 +167,7 @@
 crypto_simd-y := simd.o
 obj-$(CONFIG_CRYPTO_SIMD) += crypto_simd.o
 
-obj-$(CONFIG_CRYPTO_FIPS) += fips_canister.o
+obj-$(CONFIG_CRYPTO_FIPS) += fips_canister.o fips_canister_wrapper.o
 
 ifdef CONFIG_CRYPTO_FIPS
 ifneq ($(CONFIG_CRYPTO_FIPS),y)
