From e18132bec4d96f0d1c3cc9a1bce61d36e8af703e Mon Sep 17 00:00:00 2001
From: Daniel Wagner <wagi@monom.org>
Date: Mon, 12 Feb 2024 08:49:28 +0100
Subject: [PATCH 353/353] Linux 4.19.306-rt132 REBASE

Signed-off-by: Daniel Wagner <wagi@monom.org>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 1199ebade17b..ecff281e807f 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt16
+-rt132
-- 
2.43.0

