--- a/source/Plugins/ScriptInterpreter/Python/PythonDataObjects.cpp_org	2020-06-20 18:48:29.776153278 +0800
+++ b/source/Plugins/ScriptInterpreter/Python/PythonDataObjects.cpp	2020-06-20 18:51:19.496359653 +0800
@@ -404,14 +404,16 @@ llvm::StringRef PythonString::GetString(
     return llvm::StringRef();
 
   Py_ssize_t size;
-  char *c;
+  const char *data;
 
 #if PY_MAJOR_VERSION >= 3
-  c = PyUnicode_AsUTF8AndSize(m_py_obj, &size);
+  data = PyUnicode_AsUTF8AndSize(m_py_obj, &size);
 #else
+  char *c;
   PyString_AsStringAndSize(m_py_obj, &c, &size);
+  data = c;
 #endif
-  return llvm::StringRef(c, size);
+  return llvm::StringRef(data, size);
 }
 
 size_t PythonString::GetSize() const {
