From 13fc1412d3c2166ae0d77b4ad95de4303ec28dd9 Mon Sep 17 00:00:00 2001
From: Clark Williams <williams@redhat.com>
Date: Fri, 23 Jul 2021 12:31:54 -0500
Subject: [PATCH 338/338] Linux 4.19.198-rt85 REBASE

Signed-off-by: Clark Williams <williams@redhat.com>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index c1f2720eaf25..168858f0521b 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt76
+-rt85
-- 
2.31.1

