From 0d76fb0bd999f2a05da5d5ef5af76df8c980b724 Mon Sep 17 00:00:00 2001
From: Thomas Gleixner <tglx@linutronix.de>
Date: Fri, 8 Jul 2011 20:25:16 +0200
Subject: [PATCH 192/196] Add localversion for -RT release

Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
---
 localversion-rt | 1 +
 1 file changed, 1 insertion(+)
 create mode 100644 localversion-rt

diff --git a/localversion-rt b/localversion-rt
new file mode 100644
index 000000000000..483ad771f201
--- /dev/null
+++ b/localversion-rt
@@ -0,0 +1 @@
+-rt19
-- 
2.44.0

