From e6c0e3128465785bb2e9eff28a5dbb7e09b1bca1 Mon Sep 17 00:00:00 2001
From: Clark Williams <clark.williams@gmail.com>
Date: Tue, 27 Feb 2024 17:13:28 -0600
Subject: [PATCH 193/196] Linux 6.6.18-rt23 REBASE

Signed-off-by: Clark Williams <clark.williams@gmail.com>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 483ad771f201..9a218ca23053 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt19
+-rt23
-- 
2.44.0

