From 5b500d5f30cb80da523be2bdfd71f97f61ce88c0 Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <shreenidhi.shedi@broadcom.com>
Date: Thu, 7 Sep 2023 12:53:58 +0530
Subject: [PATCH 1/9] azure ds

Signed-off-by: Shreenidhi Shedi <shreenidhi.shedi@broadcom.com>
Signed-off-by: Shivani Agarwal <shivani.agarwal@broadcom.com>
---
 cloudinit/sources/helpers/azure.py | 30 ++++++++++++++++++++++++++++++
 1 file changed, 30 insertions(+)

diff --git a/cloudinit/sources/helpers/azure.py b/cloudinit/sources/helpers/azure.py
index 3a200ee27..be32e0479 100644
--- a/cloudinit/sources/helpers/azure.py
+++ b/cloudinit/sources/helpers/azure.py
@@ -3,6 +3,7 @@
 # TODO: Importing this file without first importing
 # cloudinit.sources.azure.errors will result in a circular import.
 import base64
+import configobj
 import json
 import logging
 import os
@@ -11,6 +12,7 @@ import textwrap
 import zlib
 from contextlib import contextmanager
 from datetime import datetime, timezone
+from io import StringIO
 from time import sleep, time
 from typing import Callable, List, Optional, TypeVar, Union
 from xml.etree import ElementTree as ET  # nosec B405
@@ -37,6 +39,8 @@ azure_ds_reporter = events.ReportEventStack(
 
 T = TypeVar("T")
 
+NETWORKD_LEASES_DIR = '/run/systemd/netif/leases'
+
 
 def azure_ds_telemetry_reporter(func: Callable[..., T]) -> Callable[..., T]:
     def impl(*args, **kwargs):
@@ -849,6 +853,32 @@ class WALinuxAgentShim:
         report_diagnostic_event(msg, logger_func=LOG.debug)
         return goal_state
 
+    @staticmethod
+    def networkd_parse_lease(content):
+        """Parse a systemd lease file content as in /run/systemd/netif/leases/
+        Parse this (almost) ini style file even though it says:
+          # This is private data. Do not parse.
+        Simply return a dictionary of key/values."""
+
+        return dict(configobj.ConfigObj(StringIO(content), list_values=False))
+
+    @staticmethod
+    def networkd_load_leases(leases_d=None):
+        """Return a dictionary of dictionaries representing each lease
+        found in lease_d.i
+        The top level key will be the filename, which is typically the ifindex."""
+
+        if leases_d is None:
+            leases_d = NETWORKD_LEASES_DIR
+
+        ret = {}
+        if not os.path.isdir(leases_d):
+            return ret
+        for lfile in os.listdir(leases_d):
+            ret[lfile] = WALinuxAgentShim.networkd_parse_lease(
+                util.load_file(os.path.join(leases_d, lfile)))
+        return ret
+
     @azure_ds_telemetry_reporter
     def _get_user_pubkeys(
         self, goal_state: GoalState, pubkey_info: list
-- 
2.48.1

