From 2d261e721ca5a4cca3bd59c2f3ccfc55a351d4c0 Mon Sep 17 00:00:00 2001
From: "Luis Claudio R. Goncalves" <lgoncalv@redhat.com>
Date: Mon, 9 Oct 2023 20:24:07 -0300
Subject: [PATCH 322/322] Linux 5.10.197-rt96 REBASE

Signed-off-by: Luis Claudio R. Goncalves <lgoncalv@redhat.com>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 21988f9ad53f1..45b639347fb1b 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt34
+-rt96
-- 
2.41.0

