From 5bac4b53f8d8e41f53395df21657c3b663c01a70 Mon Sep 17 00:00:00 2001
From: Clark Williams <clrkwllms@kernel.org>
Date: Tue, 28 Jan 2025 13:16:20 -0600
Subject: [PATCH 69/69] Linux 6.1.127-rt48 REBASE

Signed-off-by: Clark Williams <clrkwllms@kernel.org>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 272158183778..24707986c321 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt46
+-rt48
-- 
2.48.1

