From 292c609ac8d05cc993af24022582456312d5c53d Mon Sep 17 00:00:00 2001
From: Thomas Gleixner <tglx@linutronix.de>
Date: Fri, 8 Jul 2011 20:25:16 +0200
Subject: [PATCH 287/323] Add localversion for -RT release

Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
---
 localversion-rt | 1 +
 1 file changed, 1 insertion(+)
 create mode 100644 localversion-rt

diff --git a/localversion-rt b/localversion-rt
new file mode 100644
index 0000000000000..21988f9ad53f1
--- /dev/null
+++ b/localversion-rt
@@ -0,0 +1 @@
+-rt34
-- 
2.44.0

