From 4b17dfeb41ed02a42bafe6ee97ca8258abfcba19 Mon Sep 17 00:00:00 2001
From: Clark Williams <clark.williams@gmail.com>
Date: Mon, 23 Feb 2026 20:56:10 -0600
Subject: [PATCH 69/69] Linux 6.1.164-rt60 REBASE

Signed-off-by: Clark Williams <clark.williams@gmail.com>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 8cdeb45db6e2..66fa05e70f29 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt59
+-rt60
-- 
2.53.0

