diff --git a/setup.py b/setup.py
index babc7f2..c4abbcd 100755
--- a/setup.py
+++ b/setup.py
@@ -109,8 +109,6 @@ for dirprefix in PREFIX_TUPLE:
                        'images/helloworld.jpg',
                        'images/highlighter.png',
                        'images/highlight.jpg',
-                       'images/latexmath_source.png',
-                       'images/logo_asciidoc3.png',
                        'images/redsquare.jpg',
                        'images/smallnew.png',
                        'images/tiger.png', ]))
