From a1a504093803cea0c0238ef482448f738a16acb1 Mon Sep 17 00:00:00 2001
From: Keerthana K <keerthanak@vmware.com>
Date: Fri, 3 Nov 2023 07:51:23 +0000
Subject: [PATCH 1/3] List canister objs in a file

Signed-off-by: Keerthana K <keerthanak@vmware.com>
---
 crypto/Makefile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/crypto/Makefile b/crypto/Makefile
index b19245935..934761af2 100644
--- a/crypto/Makefile
+++ b/crypto/Makefile
@@ -226,14 +226,17 @@ OBJECT_FILES_NON_STANDARD_x86-aesni-intel_avx-x86_64.o := y
 crypto/x86-%.o: arch/x86/crypto/%.c $(recordmcount_source) $(objtool_dep) FORCE
 	$(call cmd,force_checksrc)
 	$(call if_changed_rule,cc_o_c)
+	@$(foreach obj,$(aesni-intel-y),echo x86-$(obj) >> objs;)
 
 crypto/x86-%.o: arch/x86/crypto/%.S $(objtool_dep)
 	$(call if_changed_rule,as_o_S)
+	@$(foreach obj,$(aesni-intel-y),echo x86-$(obj) >> objs;)
 
 lib-crypto-y := aes.o sha256.o sha1.o
 crypto/lib-crypto-%.o: lib/crypto/%.c $(recordmcount_source) $(objtool_dep) FORCE
 	$(call cmd,force_checksrc)
 	$(call if_changed_rule,cc_o_c)
+	@$(foreach obj,$(lib-crypto-y),echo lib-crypto-$(obj) >> objs;)
 
 canister += crypto_self_test.o
 
@@ -242,6 +245,7 @@ canister += fips_integrity.o
 extra-y += $(canister)
 $(obj)/canister.o: $(addprefix crypto/x86-,$(aesni-intel-y)) $(addprefix crypto/lib-crypto-,$(lib-crypto-y)) $(addprefix $(obj)/,$(canister))
 	$(LD) -z noexecstack -T $(obj)/canister_combine.lds -r $^ -o $@
+	@$(foreach obj,$(canister),echo $(obj) >> objs;)
 define UPDATE_CFLAGS =
   CFLAGS_$1 += -DFIPS_CANISTER
   CFLAGS_REMOVE_$1 += -DLATENT_ENTROPY_PLUGIN -fplugin=./scripts/gcc-plugins/latent_entropy_plugin.so -fplugin-arg-rap_plugin-check=call $(RETPOLINE_CFLAGS) $(RETHUNK_CFLAGS)
-- 
2.35.6

