From 94563db5617bfbf256ba8783f1074b3c22a5351d Mon Sep 17 00:00:00 2001
From: Clark Williams <clrkwllms@kernel.org>
Date: Sat, 13 Jun 2026 14:02:35 -0500
Subject: [PATCH 72/72] Linux 6.1.175-rt63 REBASE

Signed-off-by: Clark Williams <clrkwllms@kernel.org>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 40d81d8e61b6..b0e8dd7bd707 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt62
+-rt63
-- 
2.54.0

