From 7a19a0dc0003e2a132320c166c6773816b335192 Mon Sep 17 00:00:00 2001
From: Clark Williams <clark.williams@gmail.com>
Date: Sun, 26 May 2024 21:16:48 -0500
Subject: [PATCH 64/64] Linux 6.1.92-rt32 REBASE

Signed-off-by: Clark Williams <clark.williams@gmail.com>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index c5b71f9a229d..ce6a482618d5 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt25
+-rt32
-- 
2.45.1

