From 43b751d88347f5b601300733b7cf59ac2f009b93 Mon Sep 17 00:00:00 2001
From: Daniel Wagner <wagi@monom.org>
Date: Mon, 4 Mar 2024 08:25:59 +0100
Subject: [PATCH 353/353] Linux 4.19.307-rt133 REBASE

Signed-off-by: Daniel Wagner <wagi@monom.org>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 1199ebade17b..c2c7e0fb6685 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt16
+-rt133
-- 
2.44.0

