From 50f42c030cf236f4b0e32f997152d53f453be82a Mon Sep 17 00:00:00 2001
From: Clark Williams <clrkwllms@kernel.org>
Date: Wed, 2 Apr 2025 15:02:44 -0500
Subject: [PATCH 69/69] Linux 6.1.132-rt50 REBASE

Signed-off-by: Clark Williams <clrkwllms@kernel.org>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 272158183778..42c384668389 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt46
+-rt50
-- 
2.49.0

