From 6fd680c293747ff085b00f971f2fbf037c6c3a1f Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <sshedi@vmware.com>
Date: Wed, 9 Feb 2022 23:35:52 +0530
Subject: [PATCH] Remove libdb support

Signed-off-by: Shreenidhi Shedi <sshedi@vmware.com>
---
 Configure | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/Configure b/Configure
index 684a9c0..9599af2 100755
--- a/Configure
+++ b/Configure
@@ -20147,12 +20147,13 @@ int main(int argc, char *argv[])
 EOCP
 	set try
 	if eval $compile_ok && $run ./try; then
-		echo 'Looks OK.' >&4
-		set `$run ./try 1`
-		db_version_major=$1
-		db_version_minor=$2
-		db_version_patch=$3
-	else
+    if false; then
+      echo 'Looks OK.' >&4
+      set `$run ./try 1`
+      db_version_major=$1
+      db_version_minor=$2
+      db_version_patch=$3
+    fi
 		echo "I can't use Berkeley DB with your <db.h>.  I'll disable Berkeley DB." >&4
 		i_db=$undef
 		case " $libs " in
-- 
2.25.1

