From 27a637049e08bf17aa865e00003a19cff85486c7 Mon Sep 17 00:00:00 2001
From: Clark Williams <clark.williams@gmail.com>
Date: Sun, 28 Jul 2024 16:49:58 -0500
Subject: [PATCH 64/64] Linux 6.1.102-rt37 REBASE

Signed-off-by: Clark Williams <clark.williams@gmail.com>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index c5b71f9a229d..a3b2408c1da6 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt25
+-rt37
-- 
2.45.2

