From cedcd4f4dacd601c993b5358f15d45ab5cae9b10 Mon Sep 17 00:00:00 2001
From: Bo Gan <ganb@vmware.com>
Date: Fri, 23 Dec 2022 07:45:06 +0000
Subject: [PATCH] Enable AUXILIARY_BUS by default

Signed-off-by: Bo Gan <ganb@vmware.com>
---
 drivers/base/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/base/Kconfig b/drivers/base/Kconfig
index 6f04b831a5c0..6568f9b6c457 100644
--- a/drivers/base/Kconfig
+++ b/drivers/base/Kconfig
@@ -2,6 +2,7 @@
 menu "Generic Driver Options"
 
 config AUXILIARY_BUS
+	default y
 	bool
 
 config UEVENT_HELPER
-- 
2.25.1

