From 2704c1902adc1a5d0348f8bcd27eef08ffe622d0 Mon Sep 17 00:00:00 2001
From: Oliver Kurth <okurth@gmail.com>
Date: Sat, 29 Oct 2022 18:17:32 -0700
Subject: [PATCH] bump version to 3.4.2

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e45dd05..7a29cae 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,7 +8,7 @@
 
 CMAKE_MINIMUM_REQUIRED(VERSION 3.0 FATAL_ERROR)
 
-project(tdnf VERSION 3.4.0 LANGUAGES C)
+project(tdnf VERSION 3.4.2 LANGUAGES C)
 set(VERSION ${PROJECT_VERSION})
 set(PROJECT_YEAR 2022)
 
-- 
2.25.1

