From 2d213641523466299c0a3340fea30c14a5298b65 Mon Sep 17 00:00:00 2001
From: Oliver Kurth <okurth@gmail.com>
Date: Mon, 24 Oct 2022 11:52:19 -0700
Subject: [PATCH 1/2] fix skip obsoletes/conflicts options

---
 solv/tdnfpackage.c | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/solv/tdnfpackage.c b/solv/tdnfpackage.c
index 6f80a20..bedb305 100644
--- a/solv/tdnfpackage.c
+++ b/solv/tdnfpackage.c
@@ -1692,6 +1692,11 @@ SolvReportProblems(
         type = solver_ruleinfo(pSolv, dwProblemId,
                                &dwSource, &dwTarget, &dwDep);
 
+       if (SkipBasedOnType(pSolv, type, dwSource, dwSkipProblem))
+       {
+           continue;
+       }
+
         pszProblem = solver_problemruleinfo2str(pSolv, type, dwSource,
                                                 dwTarget, dwDep);
 
@@ -1704,11 +1709,7 @@ SolvReportProblems(
             }
         }
 
-        if (!SkipBasedOnType(pSolv, type, dwSource, dwSkipProblem))
-        {
-            dwError = ERROR_TDNF_SOLV_FAILED;
-        }
-        
+        dwError = ERROR_TDNF_SOLV_FAILED;
         pr_err("%u. %s\n", ++total_prblms, pszProblem);
     }
 
-- 
2.25.1

