From 10b59e347f827968c7a3232179aafcc1f74dfb55 Mon Sep 17 00:00:00 2001
From: Oliver Kurth <oliver.kurth@broadcom.com>
Date: Thu, 18 Dec 2025 21:53:57 +0000
Subject: [PATCH] bug fix: snapshort URL was broken in crossport from dev

---
 client/repolist.c | 13 +------------
 1 file changed, 1 insertion(+), 12 deletions(-)

diff --git a/client/repolist.c b/client/repolist.c
index 1224725..8aa21a0 100644
--- a/client/repolist.c
+++ b/client/repolist.c
@@ -544,18 +544,7 @@ TDNFLoadReposFromFile(
             }
             else if (strcmp(cn->name, TDNF_REPO_KEY_SNAPSHOT_URL) == 0)
             {
-                /* we do not support URLs yet, just filenames */
-                if (cn->value[0] == '/')
-                    pRepo->pszSnapshotUrl = strdup(cn->value);
-                else {
-                    /* path should be relative to repo dir */
-                    dwError = TDNFJoinPath(
-                                  &pRepo->pszSnapshotUrl,
-                                  pTdnf->pConf->pszRepoDir,
-                                  cn->value,
-                                  NULL);
-                    BAIL_ON_TDNF_ERROR(dwError);
-                }
+                pRepo->pszSnapshotUrl = strdup(cn->value);
             }
             else if (strcmp(cn->name, TDNF_REPO_KEY_SKIP) == 0)
             {
-- 
2.43.0

