From 9202811e627672ad212c02b8ff2603faeb438261 Mon Sep 17 00:00:00 2001
From: Oliver Kurth <oliver.kurth@broadcom.com>
Date: Fri, 17 Apr 2026 20:41:59 +0000
Subject: [PATCH] ensure history db directory exists

Signed-off-by: Oliver Kurth <oliver.kurth@broadcom.com>
---
 client/rpmtrans.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/client/rpmtrans.c b/client/rpmtrans.c
index 1220b22..6d9267e 100644
--- a/client/rpmtrans.c
+++ b/client/rpmtrans.c
@@ -138,6 +138,7 @@ TDNFRunTransactionWithHistory(
 {
     uint32_t dwError = 0;
     int rc;
+    char *pszDataDir = NULL;
 
     if(!pTdnf || !pTS || !pHistoryCtx)
     {
@@ -145,6 +146,18 @@ TDNFRunTransactionWithHistory(
         BAIL_ON_TDNF_ERROR(dwError);
     }
 
+    dwError = TDNFJoinPath(&pszDataDir,
+                           pTdnf->pArgs->pszInstallRoot,
+                           pTdnf->pConf->pszPersistDir,
+                           NULL);
+    if (dwError == 0 && pszDataDir)
+    {
+        dwError = TDNFUtilsMakeDirs(pszDataDir);
+        if (dwError == ERROR_TDNF_ALREADY_EXISTS)
+            dwError = 0;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+
     rc = history_sync(pHistoryCtx, pTS->pTS);
     if (rc != 0)
     {
@@ -162,6 +175,7 @@ TDNFRunTransactionWithHistory(
         BAIL_ON_TDNF_ERROR(dwError);
     }
 cleanup:
+    TDNF_SAFE_FREE_MEMORY(pszDataDir);
     return dwError;
 error:
     goto cleanup;
-- 
2.43.0

