From 9f6aee86738f932e81ab1f807b8cdb3f3e0b8128 Mon Sep 17 00:00:00 2001
From: Oliver Kurth <okurth@gmail.com>
Date: Fri, 10 Mar 2023 11:58:44 -0800
Subject: [PATCH 4/4] include <string.h> to fix strdup() warning

---
 history/history.c | 1 +
 history/main.c    | 1 +
 2 files changed, 2 insertions(+)

diff --git a/history/history.c b/history/history.c
index eb70410..276ab3c 100644
--- a/history/history.c
+++ b/history/history.c
@@ -7,6 +7,7 @@
  */
 
 #include <stdio.h>
+#include <string.h>
 #include <time.h>
 /* for O_RDONLY */
 #include <fcntl.h>
diff --git a/history/main.c b/history/main.c
index c4fdb35..e6bb4ae 100644
--- a/history/main.c
+++ b/history/main.c
@@ -7,6 +7,7 @@
  */
 
 #include <stdio.h>
+#include <string.h>
 #include <time.h>
 /* for O_RDONLY */
 #include <fcntl.h>
-- 
2.25.1

