diff --git a/ci/Dockerfile.fedora b/ci/Dockerfile.fedora
index 80ed1e5..8b4a92b 100644
--- a/ci/Dockerfile.fedora
+++ b/ci/Dockerfile.fedora
@@ -1,4 +1,4 @@
-FROM fedora:latest
+FROM fedora:32
 
 MAINTAINER csiddharth@vmware.com
 
diff --git a/client/api.c b/client/api.c
index 96521d4..fc7ec6d 100644
--- a/client/api.c
+++ b/client/api.c
@@ -540,13 +540,14 @@ TDNFInfo(
                   DETAIL_INFO,
                   &pPkgInfo,
                   &dwCount);
-    BAIL_ON_TDNF_ERROR(dwError);
 
-    if(dwError == ERROR_TDNF_NO_MATCH && !*ppszPackageNameSpecs)
+    if (dwError == ERROR_TDNF_NO_MATCH && !*ppszPackageNameSpecs)
     {
         dwError = 0;
     }
 
+    BAIL_ON_TDNF_ERROR(dwError);
+
     *ppPkgInfo = pPkgInfo;
     *pdwCount = dwCount;
 
@@ -620,13 +621,14 @@ TDNFList(
                   DETAIL_LIST,
                   &pPkgInfo,
                   &dwCount);
-    BAIL_ON_TDNF_ERROR(dwError);
 
-    if(dwError == ERROR_TDNF_NO_MATCH && !*ppszPackageNameSpecs)
+    if (dwError == ERROR_TDNF_NO_MATCH && !*ppszPackageNameSpecs)
     {
         dwError = 0;
     }
 
+    BAIL_ON_TDNF_ERROR(dwError);
+
     *ppPkgInfo = pPkgInfo;
     *pdwCount = dwCount;
 
diff --git a/client/remoterepo.c b/client/remoterepo.c
index 36cfe16..2c2b204 100644
--- a/client/remoterepo.c
+++ b/client/remoterepo.c
@@ -546,6 +546,7 @@ TDNFAllocateResourceURL(
     )
 {
     uint32_t dwError = 0;
+    TDNF_METALINK_URLS *new_metalink_url = NULL;
 
     if (IsNullOrEmptyString(url) || !metalink_url)
     {
@@ -553,8 +554,6 @@ TDNFAllocateResourceURL(
         BAIL_ON_TDNF_ERROR(dwError);
     }
 
-    TDNF_METALINK_URLS *new_metalink_url = NULL;
-
     dwError = TDNFAllocateMemory(1, sizeof(TDNF_METALINK_URLS), (void **)&new_metalink_url);
     BAIL_ON_TDNF_ERROR(dwError);
 
diff --git a/client/repo.c b/client/repo.c
index 9bcbfbb..17817e6 100644
--- a/client/repo.c
+++ b/client/repo.c
@@ -529,7 +529,14 @@ TDNFStoreBaseURLFromMetalink(
         BAIL_ON_TDNF_ERROR(dwError);
     }
 
-    for ( ; strcmp(pszRepo, pRepos->pszId) != 0; pRepos = pRepos->pNext);
+    while(pRepos)
+    {
+        if(!strcmp(pszRepo, pRepos->pszId))
+        {
+            break;
+        }
+        pRepos = pRepos->pNext;
+    }
 
     if (!pRepos)
     {
@@ -629,7 +636,7 @@ TDNFFreeMetalinkUrlsList(
     TDNF_METALINK_URLS *metalink_urls_temp = NULL;
     TDNF_METALINK_URLS *metalink_urls_temp_next = NULL;
 
-    if (!metalink_urls)
+    if (metalink_urls)
     {
         metalink_urls_temp = metalink_urls;
         while (metalink_urls_temp)
diff --git a/client/repolist.c b/client/repolist.c
index 1c1ae99..2a07bae 100644
--- a/client/repolist.c
+++ b/client/repolist.c
@@ -131,7 +131,7 @@ TDNFCreateCmdLineRepo(
     )
 {
     uint32_t dwError;
-    PTDNF_REPO_DATA_INTERNAL pRepo;
+    PTDNF_REPO_DATA_INTERNAL pRepo = NULL;
 
     if(!ppRepo)
     {
@@ -563,6 +563,7 @@ TDNFCloneRepo(
         dwError = TDNFAllocateStringArray(
                       pRepoIn->ppszUrlGPGKeys,
                       &pRepo->ppszUrlGPGKeys);
+        BAIL_ON_TDNF_ERROR(dwError);
     }
 
     *ppRepo = pRepo;
diff --git a/client/repoutils.c b/client/repoutils.c
index 6bc213a..08cf0d1 100644
--- a/client/repoutils.c
+++ b/client/repoutils.c
@@ -385,6 +385,9 @@ TDNFRemoveRpmCache(
                 BAIL_ON_TDNF_ERROR(dwError);
             }
         }
+        closedir(pDir);
+        pDir = NULL;
+
         if(rmdir(pszRpmCacheArchDir))
         {
             dwError = errno;
@@ -438,6 +441,9 @@ TDNFRemoveRpmCache(
                 BAIL_ON_TDNF_ERROR(dwError);
             }
         }
+        closedir(pDir);
+        pDir = NULL;
+
         if(rmdir(pszRpmCacheNoarchDir))
         {
             dwError = errno;
@@ -517,6 +523,11 @@ TDNFRemoveTmpRepodata(
         BAIL_ON_TDNF_SYSTEM_ERROR(dwError);
     }
 cleanup:
+    if (pDir)
+    {
+        closedir(pDir);
+    }
+    TDNF_SAFE_FREE_MEMORY(pszFilePath);
     return dwError;
 error:
     goto cleanup;
diff --git a/tools/cli/lib/parseupdateinfo.c b/tools/cli/lib/parseupdateinfo.c
index 847b608..00ec300 100644
--- a/tools/cli/lib/parseupdateinfo.c
+++ b/tools/cli/lib/parseupdateinfo.c
@@ -119,6 +119,7 @@ TDNFCliParseMode(
 {
     uint32_t dwError = 0;
     int nIndex = 0;
+    int nCount = 0;
     TDNF_UPDATEINFO_OUTPUT nMode = OUTPUT_SUMMARY;
     struct stTemp
     {
@@ -138,7 +139,7 @@ TDNFCliParseMode(
         BAIL_ON_CLI_ERROR(dwError);
     }
 
-    int nCount = ARRAY_SIZE(stModes);
+    nCount = ARRAY_SIZE(stModes);
     for(nIndex = 0; nIndex < nCount; ++nIndex)
     {
         if(!strcasecmp(stModes[nIndex].pszModeName, pszMode))
