From b6b7665358cfcd1ef5eb37a739f87dceb9488759 Mon Sep 17 00:00:00 2001
From: Oliver Kurth <okurth@gmail.com>
Date: Tue, 13 Sep 2022 12:18:43 -0700
Subject: [PATCH] set POOL_FLAG_NOINSTALLEDOBSOLETES and
 POOL_FLAG_IMPLICITOBSOLETEUSESPROVIDES

---
 solv/tdnfpool.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/solv/tdnfpool.c b/solv/tdnfpool.c
index 0a4153b..0d952ee 100644
--- a/solv/tdnfpool.c
+++ b/solv/tdnfpool.c
@@ -116,6 +116,12 @@ SolvInitSack(
     pool_setarch(pPool, systemInfo.machine);
     pool_set_flag(pPool, POOL_FLAG_ADDFILEPROVIDESFILTERED, 1);
 
+    /* libsolv 0.6.x allowed this, 0.7.x changed it to 0 -
+       this would disallow installing coreutils since coreutils-selinux
+       obsoletes coreutils */
+    pool_set_flag(pPool, POOL_FLAG_NOINSTALLEDOBSOLETES, 1);
+    pool_set_flag(pPool, POOL_FLAG_IMPLICITOBSOLETEUSESPROVIDES, 1);
+
     if(nInstallAllDeps)
     {
         pRepo = repo_create(pPool, SYSTEM_REPO_NAME);
-- 
2.25.1

