diff --git a/client/init.c b/client/init.c
index b7e88ff..417b995 100644
--- a/client/init.c
+++ b/client/init.c
@@ -284,9 +284,12 @@ TDNFRefreshRepo(
 
     if(nCleanMetadata || nMetadataExpired)
     {
-        fprintf(stdout,
-                "Refreshing metadata for: '%s'\n",
-                pRepo->pszName);
+        if(!pTdnf->pArgs->nQuiet)
+        {
+            fprintf(stdout,
+                    "Refreshing metadata for: '%s'\n",
+                    pRepo->pszName);
+        }
         dwError = TDNFRepoRemoveCache(pTdnf, pRepo->pszId);
         if(dwError == ERROR_TDNF_FILE_NOT_FOUND)
         {
diff --git a/client/remoterepo.c b/client/remoterepo.c
index 12b4d2d..946aafd 100644
--- a/client/remoterepo.c
+++ b/client/remoterepo.c
@@ -137,7 +137,7 @@ TDNFDownloadFile(
     dwError = curl_easy_setopt(pCurl, CURLOPT_FOLLOWLOCATION, 1L);
     BAIL_ON_TDNF_CURL_ERROR(dwError);
 
-    if(pszProgressData)
+    if(!pTdnf->pArgs->nQuiet && pszProgressData)
     {
         dwError = set_progress_cb(pCurl, pszProgressData);
         BAIL_ON_TDNF_ERROR(dwError);
