From 18050e590aeb669141b6fca988466462b4aa482f Mon Sep 17 00:00:00 2001
From: Tapas Kundu <tkundu@vmware.com>
Date: Sat, 30 May 2020 13:11:14 +0800
Subject: [PATCH] cmake: Set release version

Signed-off-by: Tapas Kundu <tkundu@vmware.com>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 25ee54e..eada511 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,7 +9,7 @@
 
 CMAKE_MINIMUM_REQUIRED(VERSION 3.0 FATAL_ERROR)
 
-project(tdnf VERSION 2.1.0 LANGUAGES C)
+project(tdnf VERSION 2.1.1 LANGUAGES C)
 set(PROJECT_YEAR 2019)
 
 # Ensure correct directory paths are used for installation
