From 514de45a6985a780dd0feb804abba4253db24d28 Mon Sep 17 00:00:00 2001
From: Priyesh Padmavilasom <ppadmavilasom@vmware.com>
Date: Tue, 22 Jan 2019 22:05:21 +0000
Subject: [PATCH] fix curl status type. remove unused code

Change-Id: I6c1a3943280c4ee63314d45a8e3fb947f835e975
---

diff --git a/client/remoterepo.c b/client/remoterepo.c
index 51bac28..45e1334 100644
--- a/client/remoterepo.c
+++ b/client/remoterepo.c
@@ -98,7 +98,8 @@
     CURL *pCurl = NULL;
     FILE *fp = NULL;
     char *pszUserPass = NULL;
-    uint32_t nStatus = 0;
+    /* lStatus reads CURLINFO_RESPONSE_CODE. Must be long */
+    long lStatus = 0;
 
     if(!pTdnf ||
        !pTdnf->pArgs ||
@@ -162,14 +163,14 @@
 
     dwError = curl_easy_getinfo(pCurl,
                                 CURLINFO_RESPONSE_CODE,
-                                &nStatus);
+                                &lStatus);
     BAIL_ON_TDNF_CURL_ERROR(dwError);
 
-    if(nStatus >= 400)
+    if(lStatus >= 400)
     {
         fprintf(stderr,
-                "Error: %d when downloading %s\n. Please check repo url.\n",
-                nStatus,
+                "Error: %ld when downloading %s\n. Please check repo url.\n",
+                lStatus,
                 pszFileUrl);
         dwError = ERROR_TDNF_INVALID_PARAMETER;
         BAIL_ON_TDNF_ERROR(dwError);
