From fdbb6efcfa1db80276a15964826dc015071ee9da Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <shreenidhi.shedi@broadcom.com>
Date: Fri, 28 Nov 2025 00:26:04 +0530
Subject: [PATCH] Refresh repodata cache only when triggered by motdgen service

Signed-off-by: Shreenidhi Shedi <shreenidhi.shedi@broadcom.com>
---
 etc/motdgen.d/02-tdnf-updateinfo.sh | 28 ++++++----------------------
 1 file changed, 6 insertions(+), 22 deletions(-)

diff --git a/etc/motdgen.d/02-tdnf-updateinfo.sh b/etc/motdgen.d/02-tdnf-updateinfo.sh
index 4fb44ab4..81d4f0c4 100755
--- a/etc/motdgen.d/02-tdnf-updateinfo.sh
+++ b/etc/motdgen.d/02-tdnf-updateinfo.sh
@@ -10,33 +10,17 @@
 #
 
 path="/var/cache/tdnf/cached-updateinfo.txt"
-ref_file="/tmp/updateinfo_timestamp.$$"
-tdnf_command="tdnf -q --refresh updateinfo"
 
-update_interval="1 hour ago"
-update_timeout="15s"
-
-if [[ "$PAM_TYPE" == "close_session" ]]; then
+if [[ "$PAM_TYPE" = "close_session" ]]; then
     exit 0
 fi
 
-touch -d "$update_interval" "$ref_file"
-if [ ! -s "${path}" ] || [ "$path" -ot "$ref_file" ]; then
-    # /var/cache/tdnf may not exist, but we pipe the output to a file there
-    mkdir -p "$(dirname ${path})"
-    timeout "$update_timeout" $tdnf_command | grep -vE '^Refreshing|^Disabling' > "${path}"
-    tdnf_status=${PIPESTATUS[0]}
-    if [[ ${tdnf_status} == 124 ]]; then
-        echo "tdnf updateinfo timed out"
-        # remove any partial data
-        rm -f "${path}"
-        exit 0
-    fi
-    if [[ ${tdnf_status} != 0 ]]; then
-        echo "tdnf updateinfo failed with ${tdnf_status}"
-    fi
+# This will be set by motdgen.service
+TRIGGERED_BY_MOTD_TIMER="${TRIGGERED_BY_MOTD_TIMER:-0}"
+
+if [ "$TRIGGERED_BY_MOTD_TIMER" = "1" ]; then
+    tdnf -q --refresh updateinfo | grep -vE '^Refreshing|^Disabling' > "${path}"
 fi
-rm -f "$ref_file"
 
 if [ -s "${path}" ]; then
     grep -qE 'Security|Bugfix|Enhancement' "${path}" || exit 0
