From 2c69a36203fde5327c3575d41da4e78d596ef71a Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <sshedi@vmare.com>
Date: Thu, 6 May 2021 17:30:35 +0530
Subject: [PATCH] erlang: fix vernemq build fail

Signed-off-by: Shreenidhi Shedi <sshedi@vmware.com>
---
 lib/crypto/c_src/openssl_config.h | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/lib/crypto/c_src/openssl_config.h b/lib/crypto/c_src/openssl_config.h
index 647ab25..8697744 100644
--- a/lib/crypto/c_src/openssl_config.h
+++ b/lib/crypto/c_src/openssl_config.h
@@ -258,9 +258,6 @@
 # ifdef HAVE_DH
 #   define HAVE_EDDH
 # endif
-# if OPENSSL_VERSION_NUMBER >= (PACKED_OPENSSL_VERSION_PLAIN(1,1,1))
-#   define HAVE_EDDSA
-# endif
 #endif
 
 #if OPENSSL_VERSION_NUMBER >= PACKED_OPENSSL_VERSION_PLAIN(1,0,1)
-- 
2.37.3

