From 467ddac279a00201bb732e7daa2794d61441075d Mon Sep 17 00:00:00 2001
From: Alexey Makhalov <amakhalov@vmware.com>
Date: Tue, 29 Sep 2015 16:01:17 -0700
Subject: [PATCH] reboot.c: Hardcodded piix4 poweroff.

---
 arch/x86/kernel/reboot.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/x86/kernel/reboot.c b/arch/x86/kernel/reboot.c
index 02693dd..e1eba75 100644
--- a/arch/x86/kernel/reboot.c
+++ b/arch/x86/kernel/reboot.c
@@ -653,6 +653,7 @@ static void native_machine_halt(void)
 
 	tboot_shutdown(TB_SHUTDOWN_HALT);
 
+	outw(0x2000, 0x1004);
 	stop_this_cpu(NULL);
 }
 
-- 
1.9.1

