diff -Naur a/net/sunrpc/xprtsock.c b/net/sunrpc/xprtsock.c
--- a/net/sunrpc/xprtsock.c	2016-09-01 15:08:53.327462189 -0700
+++ b/net/sunrpc/xprtsock.c	2016-09-01 15:08:07.579462068 -0700
@@ -1791,9 +1791,11 @@
 static int xs_bind(struct sock_xprt *transport, struct socket *sock)
 {
 	struct sockaddr_storage myaddr;
-	int err, nloop = 0;
+	int err = -EADDRINUSE;
+	int nloop = 0;
 	unsigned short port = xs_get_srcport(transport);
 	unsigned short last;
+	struct net *net = transport->xprt.xprt_net;
 
 	/*
 	 * If we are asking for any ephemeral port (i.e. port == 0 &&
@@ -1815,6 +1817,8 @@
 
 	memcpy(&myaddr, &transport->srcaddr, transport->xprt.addrlen);
 	do {
+		if (inet_is_local_reserved_port(net, port))
+			goto next_port;
 		rpc_set_port((struct sockaddr *)&myaddr, port);
 		err = kernel_bind(sock, (struct sockaddr *)&myaddr,
 				transport->xprt.addrlen);
@@ -1822,6 +1826,7 @@
 			transport->srcport = port;
 			break;
 		}
+next_port:
 		last = port;
 		port = xs_next_srcport(transport, port);
 		if (port > last)
