Author: Alexey Makhalov <amakhalov@vmware.com>
Lower log level for "Assuming drive cache..." message.

--- linux-4.2_/drivers/scsi/sd.c	2015-08-30 18:34:09.000000000 +0000
+++ linux-4.2/drivers/scsi/sd.c	2015-08-31 21:14:59.861926417 +0000
@@ -2483,7 +2483,7 @@
 				"Assuming drive cache: write back\n");
 		sdkp->WCE = 1;
 	} else {
-		sd_first_printk(KERN_ERR, sdkp,
+		sd_first_printk(KERN_NOTICE, sdkp,
 				"Assuming drive cache: write through\n");
 		sdkp->WCE = 0;
 	}
