From 0fccbc69d6db0b2630651127e37519d529396787 Mon Sep 17 00:00:00 2001
From: Keerthana K <keerthanak@vmware.com>
Date: Wed, 18 Aug 2021 07:05:41 +0000
Subject: [PATCH] ice: Use PTP_SYS_OFFSET_EXTENDED_IOCTL support

Signed-off-by: Keerthana K <keerthanak@vmware.com>
Signed-off-by: Srivatsa S. Bhat (VMware) <srivatsa@csail.mit.edu>
---
 src/kcompat.h      | 22 ----------------------
 src/kcompat_impl.h |  8 +++-----
 2 files changed, 3 insertions(+), 27 deletions(-)

diff --git a/src/kcompat.h b/src/kcompat.h
index 6549382..fba7100 100644
--- a/src/kcompat.h
+++ b/src/kcompat.h
@@ -3080,30 +3080,8 @@ _kc_dev_change_flags(struct net_device *netdev, unsigned int flags,
 
 #define dev_change_flags _kc_dev_change_flags
 #endif /* !(RHEL_RELEASE_CODE && RHEL > RHEL(8,0)) */
-#if (RHEL_RELEASE_CODE && \
-     (RHEL_RELEASE_CODE >= RHEL_RELEASE_VERSION(7,7) && \
-      RHEL_RELEASE_CODE < RHEL_RELEASE_VERSION(8,0)) || \
-     (RHEL_RELEASE_CODE >= RHEL_RELEASE_VERSION(8,1)))
 #define HAVE_PTP_SYS_OFFSET_EXTENDED_IOCTL
 #define HAVE_PTP_CLOCK_INFO_GETTIMEX64
-#else /* RHEL >= 7.7 && RHEL < 8.0 || RHEL >= 8.1 */
-struct ptp_system_timestamp {
-	struct timespec64 pre_ts;
-	struct timespec64 post_ts;
-};
-
-static inline void
-ptp_read_system_prets(struct ptp_system_timestamp __always_unused *sts)
-{
-	;
-}
-
-static inline void
-ptp_read_system_postts(struct ptp_system_timestamp __always_unused *sts)
-{
-	;
-}
-#endif /* !(RHEL >= 7.7 && RHEL != 8.0) */
 #if (RHEL_RELEASE_CODE && (RHEL_RELEASE_CODE >= RHEL_RELEASE_VERSION(8,1)))
 #define HAVE_NDO_BRIDGE_SETLINK_EXTACK
 #endif /* RHEL 8.1 */
diff --git a/src/kcompat_impl.h b/src/kcompat_impl.h
index 10d268c..973c3a2 100644
--- a/src/kcompat_impl.h
+++ b/src/kcompat_impl.h
@@ -41,13 +41,11 @@ static inline void net_prefetch(void *p)
  *
  * Implementing the wrappers directly for older kernels which still have the
  * old implementation of skb_frag_t is trivial.
+ *
+ * skb_frag_off was added to linux-stable 4.19.y in 4.19.201, so
+ * remove that from the compat implementation.
  */
 #ifdef NEED_SKB_FRAG_OFF_ACCESSORS
-static inline unsigned int skb_frag_off(const skb_frag_t *frag)
-{
-	return frag->page_offset;
-}
-
 static inline void skb_frag_off_add(skb_frag_t *frag, int delta)
 {
 	frag->page_offset += delta;
