--- a/SPIRV-LLVM-Translator-22.1.0/CMakeLists.txt
+++ b/SPIRV-LLVM-Translator-22.1.0/CMakeLists.txt
@@ -1,3 +1,4 @@
+set(LLVM_EXTERNAL_SPIRV_HEADERS_SOURCE_DIR "/usr/include/spirv/unified1")
 cmake_minimum_required(VERSION 3.13.4)
 
 if(NOT DEFINED BASE_LLVM_VERSION)
--- a/SPIRV-LLVM-Translator-22.1.0/lib/SPIRV/CMakeLists.txt
+++ b/SPIRV-LLVM-Translator-22.1.0/lib/SPIRV/CMakeLists.txt
@@ -60,7 +60,7 @@ if(SPIRV_BACKEND_FOUND)
   list(APPEND SPIRVLIB_LINK_COMPONENTS "SPIRVCodeGen")
 endif()
 
-add_llvm_library(LLVMSPIRVLib
+add_llvm_library(LLVMSPIRVLib SHARED
   ${SRC_LIST}
   LINK_COMPONENTS
     ${SPIRVLIB_LINK_COMPONENTS}
