--- src/third_party/swiftshader/BUILD.gn	2025-09-17 22:09:27.000000000 +0530
+++ src/third_party/swiftshader/BUILD.gn	2025-09-17 23:15:29.897025486 +0530
@@ -74,21 +74,3 @@
     "src/Vulkan:swiftshader_libvulkan",
   ]
 }
-
-if (build_with_chromium) {
-  group("swiftshader_tests") {
-    testonly = true
-
-    data_deps = [ "tests/SystemUnitTests:swiftshader_system_unittests" ]
-
-    if (supports_llvm) {
-      data_deps +=
-          [ "tests/ReactorUnitTests:swiftshader_reactor_llvm_unittests" ]
-    }
-
-    if (supports_subzero) {
-      data_deps +=
-          [ "tests/ReactorUnitTests:swiftshader_reactor_subzero_unittests" ]
-    }
-  }
-}
