From cb274c28c8e72fbf1e4c6cd41833442fd3f1a3db Mon Sep 17 00:00:00 2001
From: Guillaume GARDET <guillaume.gardet@free.fr>
Date: Thu, 16 Nov 2017 13:20:12 +0100
Subject: [PATCH] Fix MMC1 (external SD slot) on Samsung Snow (Chromebook ARM)

---
 drivers/mmc/dw_mmc.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/mmc/dw_mmc.c b/drivers/mmc/dw_mmc.c
index 23f642980b..a6a0934eef 100644
--- a/drivers/mmc/dw_mmc.c
+++ b/drivers/mmc/dw_mmc.c
@@ -508,7 +508,8 @@ void dwmci_setup_cfg(struct mmc_config *cfg, struct dwmci_host *host,
 		cfg->host_caps |= MMC_MODE_4BIT;
 		cfg->host_caps &= ~MMC_MODE_8BIT;
 	}
-	cfg->host_caps |= MMC_MODE_HS | MMC_MODE_HS_52MHz;
+	/* MMC_MODE_HS_52MHz is broken (at least) on Samsung Snow, so disbale it for now */
+	cfg->host_caps |= MMC_MODE_HS;
 
 	cfg->b_max = CONFIG_SYS_MMC_MAX_BLK_COUNT;
 }
