--- likewise-open-6.2.11.4/makekit/mk/module/platform.sh.orig	2017-10-02 20:15:04.188000000 -0400
+++ likewise-open-6.2.11.4/makekit/mk/module/platform.sh	2017-10-02 20:15:22.800000000 -0400
@@ -285,6 +285,9 @@
                 x86_64|amd64)
                     _default_MK_BUILD_ARCH="x86_64"
                     ;;
+                aarch64|arm64)
+                    _default_MK_BUILD_ARCH="aarch64"
+                    ;;
                 *)
                     mk_fail "unknown architecture: `uname -m`"
                     ;;
@@ -690,4 +693,4 @@
 {
     _MK_MULTIARCH_SYS="${_MK_MULTIARCH_SYS_MAKE}"
     _MK_COMPAT_SYS="${_MK_MULTIARCH_COMPAT_MAKE}"
-}
\ No newline at end of file
+}
