--- a/include/configs/rpi.h	2019-02-22 11:39:43.610750574 +0000
+++ b/include/configs/rpi.h	2019-02-22 11:41:22.498671719 +0000
@@ -146,12 +146,20 @@
 	"fdt_addr_r=0x02600000\0" \
 	"ramdisk_addr_r=0x02700000\0"
 
+#ifdef CONFIG_USB
 #define BOOT_TARGET_DEVICES(func) \
 	func(MMC, mmc, 0) \
 	func(MMC, mmc, 1) \
 	func(USB, usb, 0) \
 	func(PXE, pxe, na) \
 	func(DHCP, dhcp, na)
+#else
+#define BOOT_TARGET_DEVICES(func) \
+        func(MMC, mmc, 0) \
+        func(MMC, mmc, 1) \
+        func(PXE, pxe, na) \
+        func(DHCP, dhcp, na)
+#endif
 #include <config_distro_bootcmd.h>
 
 #define CONFIG_EXTRA_ENV_SETTINGS \
