From feb077627a83bd24f6bcf502a7a74f2317498946 Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <sshedi@vmware.com>
Date: Wed, 1 Mar 2023 01:29:02 +0530
Subject: [PATCH 6/7] dracut.sh: validate instmods calls

Signed-off-by: Shreenidhi Shedi <sshedi@vmware.com>
---
 dracut.sh | 15 ++++++++++++---
 1 file changed, 12 insertions(+), 3 deletions(-)

diff --git a/dracut.sh b/dracut.sh
index 7107ce6..e917fd1 100755
--- a/dracut.sh
+++ b/dracut.sh
@@ -2202,11 +2202,17 @@ if [[ $no_kernel != yes ]]; then
 
     if [[ -n ${add_drivers// /} ]]; then
         # shellcheck disable=SC2086
-        hostonly='' instmods -c $add_drivers
+        if ! hostonly='' instmods -c $add_drivers; then
+            dfatal "instmods failed for add_drivers: $add_drivers"
+            exit 1
+        fi
     fi
     if [[ $force_drivers ]]; then
         # shellcheck disable=SC2086
-        hostonly='' instmods -c $force_drivers
+        if ! hostonly='' instmods -c $force_drivers; then
+            dfatal "instmods failed for force_drivers: $force_drivers"
+            exit 1
+        fi
         if [[ $kernel_only != yes ]]; then
             rm -f "$initdir"/etc/cmdline.d/20-force_driver.conf
             for mod in $force_drivers; do
@@ -2216,7 +2222,10 @@ if [[ $no_kernel != yes ]]; then
     fi
     if [[ $filesystems ]]; then
         # shellcheck disable=SC2086
-        hostonly='' instmods -c $filesystems
+        if ! hostonly='' instmods -c $filesystems; then
+            dfatal "instmods failed for filesystems: $filesystems"
+            exit 1
+        fi
     fi
 
     dinfo "*** Installing kernel module dependencies ***"
-- 
2.43.7

