diff -rupN dracut-050/modules.d/01fips/fips.sh dracut-050-fips/modules.d/01fips/fips.sh
--- dracut-050/modules.d/01fips/fips.sh	2020-03-04 11:43:22.000000000 +0100
+++ dracut-050-fips/modules.d/01fips/fips.sh	2020-04-24 07:45:51.808192241 +0200
@@ -75,10 +75,38 @@ fips_load_crypto()
 
     info "Loading and integrity checking all crypto modules"
     mv /etc/modprobe.d/fips.conf /etc/modprobe.d/fips.conf.bak
+
+    if ! insmod "/lib/modules/${KERNEL}/extra/hmac_generator.ko.xz"; then
+       warn "Not able to load hmac_generator"
+       return 1
+    fi
+
     for _module in $FIPSMODULES; do
-        if [ "$_module" != "tcrypt" ]; then
+
+        x=$(find /lib/modules/${KERNEL}/ -name ${_module}.ko.xz)
+        if [ -z $x ]; then
+          warn "Module $_module not found"
+            return 1
+        fi
+        DIRECTORY=$(dirname $x)
+        FILENAME=$(basename $x)
+        MOD_HMAC=$DIRECTORY/.$FILENAME.hmac
+        if [ ! -f $MOD_HMAC ]; then
+           warn ".hmac file for Module $_module not found"
+            return 1
+        fi
+        HMAC_SUM_ORIG=$(cat $MOD_HMAC | while read a b || [ -n "$a" ]; do printf "%s\n" $b; done)
+        HMAC_SUM_CALC=$(hmacgen HMAC-SHA256 "FIPS-PH3-VMW2020" $x)
+        if [ -z "$HMAC_SUM_ORIG" ] || [ -z "$HMAC_SUM_CALC" ] || [ "${HMAC_SUM_ORIG}" != "${HMAC_SUM_CALC}" ]; then
+           warn "$_module HMAC sum mismatch"
+           return 1
+        fi
+     done
+     for _module in $FIPSMODULES; do
+        if [ "$_module" != "tcrypt" ] && [ "$_module" != "hmac_generator" ]; then
+
             if ! modprobe "${_module}" 2>/tmp/fips.modprobe_err; then
-                # check if kernel provides generic algo
+    warn "modprobe for $_module failed"
                 _found=0
                 while read _k _s _v || [ -n "$_k" ]; do
                     [ "$_k" != "name" -a "$_k" != "driver" ] && continue
@@ -91,10 +119,6 @@ fips_load_crypto()
         fi
     done
     mv /etc/modprobe.d/fips.conf.bak /etc/modprobe.d/fips.conf
-
-    info "Self testing crypto algorithms"
-    modprobe tcrypt || return 1
-    rmmod tcrypt
 }
 
 do_fips()
@@ -137,14 +161,40 @@ do_fips()
         fi
 
         BOOT_IMAGE_HMAC="/boot/${BOOT_IMAGE_PATH}/.${BOOT_IMAGE_NAME}.hmac"
+  BOOT_IMAGE_KERN="/boot/${BOOT_IMAGE_PATH}${BOOT_IMAGE_NAME}"
         if ! [ -e "${BOOT_IMAGE_HMAC}" ]; then
-            warn "${BOOT_IMAGE_HMAC} does not exist"
-            return 1
+            found=0
+            umount /boot/
+            PARTS=$(find /dev/ -name "sda*")
+            for part in $PARTS; do
+                mount -o ro $part /boot
+                if [ $? -eq 0 ]; then
+                   if [ -e "/boot/${BOOT_IMAGE}" ]; then
+                       found=1
+                       break
+                   else
+                       umount /boot/
+                   fi
+                fi
+            done
+            if [ "$found" = 0 ]; then
+                warn "${BOOT_IMAGE_HMAC} does not exist"
+                return 1
+            fi
+        fi
+       HMAC_SUM_ORIG=$(cat $BOOT_IMAGE_HMAC | while read a b || [ -n "$a" ]; do printf "%s\n" $b; done)
+       HMAC_SUM_CALC=$(hmacgen HMAC-SHA256 "FIPS-PH3-VMW2020" $BOOT_IMAGE_KERN )
+        if [ -z "$HMAC_SUM_ORIG" ] || [ -z "$HMAC_SUM_CALC" ] || [ "${HMAC_SUM_ORIG}" != "${HMAC_SUM_CALC}" ]; then
+           warn "HMAC sum mismatch"
+           return 1
         fi
-
-        (cd "${BOOT_IMAGE_HMAC%/*}" && sha512hmac -c "${BOOT_IMAGE_HMAC}") || return 1
     fi
 
+    #integrity check
+    warn "Self testing crypto algorithms"
+    modprobe tcrypt || return 1
+    rmmod tcrypt
+
     info "All initrd crypto checks done"
 
     > /tmp/fipsdone
