diff -rupN dracut-048/modules.d/00systemd/module-setup.sh dracut-048-systemd-tty-ask-password/modules.d/00systemd/module-setup.sh
--- dracut-048/modules.d/00systemd/module-setup.sh	2020-04-01 12:10:42.257733876 +0530
+++ dracut-048-systemd-tty-ask-password/modules.d/00systemd/module-setup.sh	2020-04-01 12:11:59.616096656 +0530
@@ -135,6 +135,7 @@ install() {
         mount umount reboot poweroff \
         systemd-run systemd-escape \
         systemd-cgls systemd-tmpfiles \
+        systemd-ask-password systemd-tty-ask-password-agent \
         /etc/udev/udev.hwdb \
         ${NULL}
 
