From dbf29962bd50e334a0389c5c464671f4b0faf44f Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <yesshedi@gmail.com>
Date: Sat, 9 May 2026 17:35:56 +0530
Subject: [PATCH 17/18] Fix -Wdiscarded-qualifiers error with glibc-2.43

Signed-off-by: Shreenidhi Shedi <yesshedi@gmail.com>
---
 tools/perf/util/parse-events.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/perf/util/parse-events.c b/tools/perf/util/parse-events.c
index 5973f46c2375..e3b7950331bc 100644
--- a/tools/perf/util/parse-events.c
+++ b/tools/perf/util/parse-events.c
@@ -1647,7 +1647,7 @@ int parse_events__modifier_group(struct list_head *list,
  */
 static bool is_same_uncore_block(const char *pmu_name_a, const char *pmu_name_b)
 {
-	char *end_a, *end_b;
+	const char *end_a, *end_b;
 
 	end_a = strrchr(pmu_name_a, '_');
 	end_b = strrchr(pmu_name_b, '_');
-- 
2.54.0

