From 56c9e3ba52d1f30bad6ff3ba2042ef9fe30525e8 Mon Sep 17 00:00:00 2001
From: Daniel Wagner <wagi@monom.org>
Date: Wed, 20 Dec 2023 13:45:00 +0100
Subject: [PATCH 354/354] Linux 4.19.302-rt131 REBASE

Signed-off-by: Daniel Wagner <wagi@monom.org>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 1199ebade17b..a328b97369c2 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt16
+-rt131
-- 
2.43.0

