From 246058930e6410d50b2f4d1fb7813f432e4714b7 Mon Sep 17 00:00:00 2001
From: Clark Williams <clrkwllms@kernel.org>
Date: Tue, 31 Dec 2024 11:14:40 -0600
Subject: [PATCH 69/69] Linux 6.1.120-rt47 REBASE

Signed-off-by: Clark Williams <clrkwllms@kernel.org>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 272158183778..8a777ac42aab 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt46
+-rt47
-- 
2.47.1

