diff --git a/Makefile b/Makefile
index 49bfd68..891e045 100644
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
-CFLAGS=-D_GNU_SOURCE -Wall -O2
+CFLAGS=-D_GNU_SOURCE -Wall -O2 -g
 PRG=proctools
 DIR=${DESTDIR}/sbin
 
@@ -12,7 +12,7 @@ clean:
 
 install:
 	install -dm 0755 $(DIR)
-	install -sm 0755  $(PRG) $(DIR)
+	install -m 0755  $(PRG) $(DIR)
 	ln -sf $(PRG) ${DIR}/checkproc
 	ln -sf $(PRG) ${DIR}/killproc
 	ln -sf $(PRG) ${DIR}/startproc
