From 5b2e78ad57c792ba0570bf0feff7352c69598c27 Mon Sep 17 00:00:00 2001
From: Clark Williams <clark.williams@gmail.com>
Date: Sat, 7 Sep 2024 14:39:10 -0500
Subject: [PATCH 63/63] Linux 6.1.108-rt40 REBASE

Signed-off-by: Clark Williams <clark.williams@gmail.com>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index c5b71f9a229d..2af6c89aee6d 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt25
+-rt40
-- 
2.46.0

