From 1b1ba8beab04ad392b03eeee84e91fa4169658d8 Mon Sep 17 00:00:00 2001
From: Clark Williams <clark.williams@gmail.com>
Date: Mon, 28 Oct 2024 13:01:42 -0500
Subject: [PATCH 63/69] Linux 6.1.114-rt44 REBASE

Signed-off-by: Clark Williams <clark.williams@gmail.com>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index c5b71f9a229d..ac4d836a809d 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt25
+-rt44
-- 
2.48.1

