From 4d63444f84936ffcd84db31c6951bfad10c2de8c Mon Sep 17 00:00:00 2001
From: Clark Williams <clrkwllms@kernel.org>
Date: Tue, 17 Dec 2024 13:34:26 -0600
Subject: [PATCH 67/69] Linux 6.1.120-rt46 REBASE

Signed-off-by: Clark Williams <clrkwllms@kernel.org>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index ac4d836a809d..272158183778 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt44
+-rt46
-- 
2.48.1

