From dd227ccc7c80851d2458fdc5cc49920fe31b2a08 Mon Sep 17 00:00:00 2001
From: Clark Williams <clrkwllms@kernel.org>
Date: Tue, 11 Feb 2025 15:00:34 -0600
Subject: [PATCH 69/69] Linux 6.1.128-rt49 REBASE

Signed-off-by: Clark Williams <clrkwllms@kernel.org>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 272158183778..4b7dca68a5b4 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt46
+-rt49
-- 
2.48.1

