diff --git a/crypto/engine/eng_all.c b/crypto/engine/eng_all.c
index 48ad0d2..482481d 100644
--- a/crypto/engine/eng_all.c
+++ b/crypto/engine/eng_all.c
@@ -64,6 +64,7 @@ void ENGINE_load_builtin_engines(void)
 {
     /* Some ENGINEs need this */
     OPENSSL_cpuid_setup();
+ OPENSSL_init(); 
 #if 0
     /*
      * There's no longer any need for an "openssl" ENGINE unless, one day, it
diff --git a/crypto/evp/c_allc.c b/crypto/evp/c_allc.c
index 280e584..f177b65 100644
--- a/crypto/evp/c_allc.c
+++ b/crypto/evp/c_allc.c
@@ -64,6 +64,7 @@
 
 void OpenSSL_add_all_ciphers(void)
 {
+ OPENSSL_init(); 
 
 #ifndef OPENSSL_NO_DES
     EVP_add_cipher(EVP_des_cfb());
diff --git a/crypto/evp/c_alld.c b/crypto/evp/c_alld.c
index fdbe3ee..d7f9357 100644
--- a/crypto/evp/c_alld.c
+++ b/crypto/evp/c_alld.c
@@ -64,6 +64,7 @@
 
 void OpenSSL_add_all_digests(void)
 {
+ OPENSSL_init(); 
 #ifndef OPENSSL_NO_MD4
     EVP_add_digest(EVP_md4());
 #endif
diff --git a/ssl/ssl_algs.c b/ssl/ssl_algs.c
index e6f515f..64914ad 100644
--- a/ssl/ssl_algs.c
+++ b/ssl/ssl_algs.c
@@ -63,6 +63,7 @@
 
 int SSL_library_init(void)
 {
+ OPENSSL_init(); 
 
 #ifndef OPENSSL_NO_DES
     EVP_add_cipher(EVP_des_cbc());
