diff --git a/crypto/engine/eng_all.c b/crypto/engine/eng_all.c
index ebe0277..47ff9d1 100644
--- a/crypto/engine/eng_all.c
+++ b/crypto/engine/eng_all.c
@@ -14,7 +14,7 @@ void ENGINE_load_builtin_engines(void)
 {
     /* Some ENGINEs need this */
     OPENSSL_cpuid_setup();
-
+    OPENSSL_init();
     OPENSSL_init_crypto(OPENSSL_INIT_ENGINE_ALL_BUILTIN, NULL);
 }
 
diff --git a/crypto/evp/c_allc.c b/crypto/evp/c_allc.c
index 6ed31ed..94ddd7d 100644
--- a/crypto/evp/c_allc.c
+++ b/crypto/evp/c_allc.c
@@ -16,6 +16,7 @@
 
 void openssl_add_all_ciphers_int(void)
 {
+    OPENSSL_init();
 
 #ifndef OPENSSL_NO_DES
     EVP_add_cipher(EVP_des_cfb());
diff --git a/crypto/evp/c_alld.c b/crypto/evp/c_alld.c
index ec79734..817f475 100644
--- a/crypto/evp/c_alld.c
+++ b/crypto/evp/c_alld.c
@@ -16,6 +16,7 @@
 
 void openssl_add_all_digests_int(void)
 {
+    OPENSSL_init();
 #ifndef OPENSSL_NO_MD4
     EVP_add_digest(EVP_md4());
 #endif
diff --git a/ssl/ssl_init.c b/ssl/ssl_init.c
index 3e62d48..1ab705c 100644
--- a/ssl/ssl_init.c
+++ b/ssl/ssl_init.c
@@ -191,6 +191,7 @@ int OPENSSL_init_ssl(uint64_t opts, const OPENSSL_INIT_SETTINGS * settings)
         return 0;
     }
 
+    OPENSSL_init();
     if (!RUN_ONCE(&ssl_base, ossl_init_ssl_base))
         return 0;
 
