From 98aa9845f2a8356ad7df51880d9431823f3abffe Mon Sep 17 00:00:00 2001
From: Clark Williams <clark.williams@gmail.com>
Date: Thu, 28 Mar 2024 16:06:31 -0500
Subject: [PATCH 64/64] Linux 6.1.83-rt28 REBASE

Signed-off-by: Clark Williams <clark.williams@gmail.com>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index c5b71f9a229d..5a64c9c2ab61 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt25
+-rt28
-- 
2.44.0

