From 952eb41a53c74f3c6701b84a85cdb715fadd171e Mon Sep 17 00:00:00 2001
From: Thomas Gleixner <tglx@linutronix.de>
Date: Fri, 8 Jul 2011 20:25:16 +0200
Subject: [PATCH 260/347] Add localversion for -RT release

Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
---
 localversion-rt | 1 +
 1 file changed, 1 insertion(+)
 create mode 100644 localversion-rt

diff --git a/localversion-rt b/localversion-rt
new file mode 100644
index 000000000000..1199ebade17b
--- /dev/null
+++ b/localversion-rt
@@ -0,0 +1 @@
+-rt16
-- 
2.35.1

