From 0e168adcbb50d9082991263abb3bce3b6708ff46 Mon Sep 17 00:00:00 2001
From: Daniel Wagner <wagi@monom.org>
Date: Thu, 31 Mar 2022 20:37:13 +0200
Subject: [PATCH 345/347] Linux 4.19.237-rt107 REBASE

Signed-off-by: Daniel Wagner <wagi@monom.org>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 9ea5981fa956..2e0a3c960575 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt101
+-rt107
-- 
2.35.1

