From 6a068c4c66411b0584c44fe4696dab2d2c036984 Mon Sep 17 00:00:00 2001
From: Brennan Lamoreaux <blamoreaux@vmware.com>
Date: Thu, 13 Apr 2023 22:38:21 +0000
Subject: [PATCH] directly link libabsl_synchronization for grpc test

---
 tests/lib/subdir.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/lib/subdir.am b/tests/lib/subdir.am
index 62b1944..8f63a77 100644
--- a/tests/lib/subdir.am
+++ b/tests/lib/subdir.am
@@ -19,7 +19,7 @@ EXTRA_DIST += tests/lib/test_frrscript.py
 
 
 ##############################################################################
-GRPC_TESTS_LDADD = staticd/libstatic.a grpc/libfrrgrpc_pb.la -lgrpc++ -lprotobuf $(ALL_TESTS_LDADD) $(LIBYANG_LIBS) -lm
+GRPC_TESTS_LDADD = staticd/libstatic.a grpc/libfrrgrpc_pb.la -lgrpc++ -lprotobuf -labsl_synchronization $(ALL_TESTS_LDADD) $(LIBYANG_LIBS) -lm
 
 if GRPC
 check_PROGRAMS += tests/lib/test_grpc
-- 
2.39.0

