diff --git a/isisd/isis_circuit.c b/isisd/isis_circuit.c
index ffa6ad3..8c28b17 100644
--- a/isisd/isis_circuit.c
+++ b/isisd/isis_circuit.c
@@ -1543,6 +1543,10 @@ ferr_r isis_circuit_passwd_set(struct isis_circuit *circuit,
 		return ferr_code_bug(
 			"circuit password too long (max 254 chars)");
 
+	//When in FIPS mode, the password never gets set in MD5
+	if((passwd_type == ISIS_PASSWD_TYPE_HMAC_MD5) && FIPS_mode())
+		return ferr_cfg_invalid("FIPS mode is enabled, md5 authentication is disabled");
+
 	circuit->passwd.len = len;
 	strlcpy((char *)circuit->passwd.passwd, passwd,
 		sizeof(circuit->passwd.passwd));
diff --git a/isisd/isisd.c b/isisd/isisd.c
index b1064d8..fbcd097 100644
--- a/isisd/isisd.c
+++ b/isisd/isisd.c
@@ -3040,6 +3040,10 @@ static int isis_area_passwd_set(struct isis_area *area, int level,
 		if (len > 254)
 			return -1;
 
+		//When in FIPS mode, the password never get set in MD5
+		if ((passwd_type == ISIS_PASSWD_TYPE_HMAC_MD5) && (FIPS_mode()))
+			return ferr_cfg_invalid("FIPS mode is enabled, md5 authentication is disabled");
+
 		modified.len = len;
 		strlcpy((char *)modified.passwd, passwd,
 			sizeof(modified.passwd));
diff --git a/isisd/isisd.h b/isisd/isisd.h
index f5042e4..2c05ca5 100644
--- a/isisd/isisd.h
+++ b/isisd/isisd.h
@@ -29,6 +29,8 @@
 
 DECLARE_MGROUP(ISISD);
 
+# define FIPS_mode() EVP_default_properties_is_fips_enabled(NULL)
+
 #ifdef FABRICD
 static const bool fabricd = true;
 #define PROTO_TYPE ZEBRA_ROUTE_OPENFABRIC
diff --git a/ospfd/ospf_vty.c b/ospfd/ospf_vty.c
index 355742a..0ede0da 100644
--- a/ospfd/ospf_vty.c
+++ b/ospfd/ospf_vty.c
@@ -1085,6 +1085,11 @@ DEFUN (ospf_area_vlink,
 		vl_config.keychain = argv[idx+1]->arg;
 	} else if (argv_find(argv, argc, "message-digest", &idx)) {
 		/* authentication message-digest */
+        if(FIPS_mode())
+        {
+            vty_out(vty, "FIPS mode is enabled, md5 authentication is disabled\n");
+            return CMD_WARNING_CONFIG_FAILED;
+        }
 		vl_config.auth_type = OSPF_AUTH_CRYPTOGRAPHIC;
 	} else if (argv_find(argv, argc, "null", &idx)) {
 		/* "authentication null" */
@@ -1997,6 +2002,15 @@ DEFUN (ospf_area_authentication_message_digest,
 				  ? OSPF_AUTH_NULL
 				  : OSPF_AUTH_CRYPTOGRAPHIC;
 
+	if(area->auth_type == OSPF_AUTH_CRYPTOGRAPHIC)
+	{
+		if(FIPS_mode())
+		{
+			vty_out(vty, "FIPS mode is enabled, md5 authentication is disabled\n");
+			return CMD_WARNING_CONFIG_FAILED;
+		}
+	}
+
 	return CMD_SUCCESS;
 }
 
@@ -7621,6 +7635,11 @@ DEFUN (ip_ospf_authentication_args,
 
 	/* Handle message-digest authentication */
 	if (argv[idx_encryption]->arg[0] == 'm') {
+        if(FIPS_mode())
+        {
+            vty_out(vty, "FIPS mode is enabled, md5 authentication is disabled\n");
+            return CMD_WARNING_CONFIG_FAILED;
+        }
 		SET_IF_PARAM(params, auth_type);
 		params->auth_type = OSPF_AUTH_CRYPTOGRAPHIC;
 		UNSET_IF_PARAM(params, keychain_name);
@@ -7949,6 +7968,11 @@ DEFUN (ip_ospf_message_digest_key,
        "The OSPF password (key)\n"
        "Address of interface\n")
 {
+	if(FIPS_mode())
+	{
+		vty_out(vty, "FIPS mode is enabled, md5 authentication is disabled\n");
+		return CMD_WARNING_CONFIG_FAILED;
+	}
 	VTY_DECLVAR_CONTEXT(interface, ifp);
 	struct crypt_key *ck;
 	uint8_t key_id;
diff --git a/ospfd/ospfd.h b/ospfd/ospfd.h
index 2ab7db1..04d1ff4 100644
--- a/ospfd/ospfd.h
+++ b/ospfd/ospfd.h
@@ -20,6 +20,8 @@
 #include "ospf_memory.h"
 #include "ospf_dump_api.h"
 
+# define FIPS_mode() EVP_default_properties_is_fips_enabled(NULL)
+
 #define OSPF_VERSION            2
 
 /* VTY port number. */
diff --git a/ripd/rip_cli.c b/ripd/rip_cli.c
index 097c708..854a16e 100644
--- a/ripd/rip_cli.c
+++ b/ripd/rip_cli.c
@@ -876,6 +876,12 @@ DEFPY_YANG (ip_rip_authentication_mode,
 			value = "20";
 	}
 
+	if(strmatch(mode, "md5") && FIPS_mode())
+	{
+		vty_out(vty, "FIPS mode is enabled, md5 authentication id disabled\n");
+		return CMD_WARNING_CONFIG_FAILED;
+	}
+
 	nb_cli_enqueue_change(vty, "./authentication-scheme/mode", NB_OP_MODIFY,
 			      strmatch(mode, "md5") ? "md5" : "plain-text");
 	if (strmatch(mode, "md5"))
diff --git a/ripd/ripd.h b/ripd/ripd.h
index ac4a51f..4a68993 100644
--- a/ripd/ripd.h
+++ b/ripd/ripd.h
@@ -6,6 +6,8 @@
 #ifndef _ZEBRA_RIP_H
 #define _ZEBRA_RIP_H
 
+# define FIPS_mode() EVP_default_properties_is_fips_enabled(NULL)
+
 #include "hook.h"
 #include "nexthop.h"
 #include "distribute.h"
