From fd689eadb419d7fde009a6d3e4fe47136878d90d Mon Sep 17 00:00:00 2001
From: Ashwin Dayanand Kamat <kashwindayan@vmware.com>
Date: Fri, 26 Nov 2021 23:25:57 +0000
Subject: [PATCH] disable md5 algorithm for sctp if fips is enabled

Signed-off-by: Keerthana K <keerthanak@vmware.com>
---
 net/sctp/protocol.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/net/sctp/protocol.c b/net/sctp/protocol.c
index bcd3384..eb27841 100644
--- a/net/sctp/protocol.c
+++ b/net/sctp/protocol.c
@@ -34,6 +34,7 @@
 #include <linux/memblock.h>
 #include <linux/highmem.h>
 #include <linux/slab.h>
+#include <linux/fips.h>
 #include <net/net_namespace.h>
 #include <net/protocol.h>
 #include <net/ip.h>
@@ -1312,6 +1313,7 @@ static int __net_init sctp_defaults_init(struct net *net)
 	/* Whether Cookie Preservative is enabled(1) or not(0) */
 	net->sctp.cookie_preserve_enable 	= 1;
 
+	if (!fips_enabled) {
 	/* Default sctp sockets to use md5 as their hmac alg */
 #if defined (CONFIG_SCTP_DEFAULT_COOKIE_HMAC_MD5)
 	net->sctp.sctp_hmac_alg			= "md5";
@@ -1320,6 +1322,15 @@ static int __net_init sctp_defaults_init(struct net *net)
 #else
 	net->sctp.sctp_hmac_alg			= NULL;
 #endif
+	}
+
+	else {
+#if defined (CONFIG_SCTP_DEFAULT_COOKIE_HMAC_SHA1)
+	net->sctp.sctp_hmac_alg			= "sha1";
+#else
+	net->sctp.sctp_hmac_alg			= NULL;
+#endif
+	}
 
 	/* Max.Burst		    - 4 */
 	net->sctp.max_burst			= SCTP_DEFAULT_MAX_BURST;
-- 
2.23.3

