--- blfs-systemd-units-20140907/Makefile	2014-08-24 12:26:39.000000000 -0700
+++ blfs-systemd-units-20140907/Makefile1	2014-10-28 15:54:23.184338761 -0700
@@ -183,7 +183,7 @@
 	install -m ${CONFMODE} blfs/units/sshd.service ${UNITSDIR}/
 	install -m ${CONFMODE} blfs/units/sshdat.service ${UNITSDIR}/sshd@.service
 	install -m ${CONFMODE} blfs/units/sshd.socket ${UNITSDIR}/
-	systemctl enable sshd.service
+	#systemctl enable sshd.service
 
 install-svnserve: create-dirs
 	install -m ${CONFMODE} blfs/default/svnserve ${DEFAULTSDIR}/

--- blfs-systemd-units-20140907/Makefile	2014-08-24 12:26:39.000000000 -0700
+++ blfs-systemd-units-20140907/Makefile1	2015-01-07 10:08:49.515838075 -0800
@@ -75,7 +75,7 @@
 
 install-iptables: create-dirs
 	install -m ${CONFMODE} blfs/units/iptables.service ${UNITSDIR}/
-	systemctl enable iptables.service
+	#systemctl enable iptables.service
 
 install-kdm: create-dirs
 	install -m ${CONFMODE} blfs/units/kdm.service ${UNITSDIR}/

