diff -rup openssh-7.5p1-fips/readconf.c openssh-7.5p1-fips-enable-1/readconf.c
--- openssh-7.5p1-fips/readconf.c	2017-02-02 16:19:05.624875801 -0800
+++ openssh-7.5p1-fips-enable-1/readconf.c	2017-02-15 17:08:04.946982460 -0800
@@ -171,7 +171,8 @@ typedef enum {
 	oStreamLocalBindMask, oStreamLocalBindUnlink, oRevokedHostKeys,
 	oFingerprintHash, oUpdateHostkeys, oHostbasedKeyTypes,
 	oPubkeyAcceptedKeyTypes, oProxyJump,
-	oIgnoredUnknownOption, oDeprecated, oUnsupported
+	oIgnoredUnknownOption, oDeprecated, oUnsupported,
+	oFipsMode
 } OpCodes;
 
 /* Textual representations of the tokens. */
@@ -291,6 +292,7 @@ static struct {
 	{ "streamlocalbindunlink", oStreamLocalBindUnlink },
 	{ "revokedhostkeys", oRevokedHostKeys },
 	{ "fingerprinthash", oFingerprintHash },
+	{ "fipsmode", oFipsMode },
 	{ "updatehostkeys", oUpdateHostkeys },
 	{ "hostbasedkeytypes", oHostbasedKeyTypes },
 	{ "pubkeyacceptedkeytypes", oPubkeyAcceptedKeyTypes },
@@ -965,6 +967,10 @@ parse_time:
 		intptr = &options->gss_deleg_creds;
 		goto parse_flag;
 
+        case oFipsMode:
+                intptr = &options->fips_mode;
+                goto parse_flag;
+
 	case oBatchMode:
 		intptr = &options->batch_mode;
 		goto parse_flag;
@@ -1857,6 +1863,7 @@ initialize_options(Options * options)
 	options->update_hostkeys = -1;
 	options->hostbased_key_types = NULL;
 	options->pubkey_key_types = NULL;
+	options->fips_mode = -1;
 }
 
 /*
@@ -2044,6 +2051,13 @@ fill_default_options(Options * options)
 		options->fingerprint_hash = SSH_FP_HASH_DEFAULT;
 	if (options->update_hostkeys == -1)
 		options->update_hostkeys = 0;
+        if (options->fips_mode == -1)
+                options->fips_mode = 0;
+        /* Call FIPS_mode_set depending on whether fips mode is set or not */
+        if (options->fips_mode)
+                if (!FIPS_mode_set(1))
+                        fatal("FIPS mode could not be set");
+
 	if (kex_assemble_names((FIPS_mode() ? KEX_FIPS_ENCRYPT
 	        : KEX_CLIENT_ENCRYPT), &options->ciphers) != 0 ||
 	    kex_assemble_names((FIPS_mode() ? KEX_FIPS_MAC
@@ -2535,6 +2549,7 @@ dump_client_config(Options *o, const cha
 	dump_cfg_fmtint(oVerifyHostKeyDNS, o->verify_host_key_dns);
 	dump_cfg_fmtint(oVisualHostKey, o->visual_host_key);
 	dump_cfg_fmtint(oUpdateHostkeys, o->update_hostkeys);
+	dump_cfg_fmtint(oFipsMode, o->fips_mode);
 
 	/* Integer options */
 	dump_cfg_int(oCanonicalizeMaxDots, o->canonicalize_max_dots);
Only in openssh-7.5p1-fips-enable-1: readconf.c.orig
Only in openssh-7.5p1-fips-enable-1: readconf.c.rej
diff -rup openssh-7.5p1-fips/readconf.h openssh-7.5p1-fips-enable-1/readconf.h
--- openssh-7.5p1-fips/readconf.h	2017-02-02 15:26:47.532946693 -0800
+++ openssh-7.5p1-fips-enable-1/readconf.h	2017-02-15 17:03:52.938988153 -0800
@@ -157,6 +157,7 @@ typedef struct {
 	char	*revoked_host_keys;
 
 	int	 fingerprint_hash;
+	int	 fips_mode;
 
 	int	 update_hostkeys; /* one of SSH_UPDATE_HOSTKEYS_* */
 
diff -rup openssh-7.5p1-fips/servconf.c openssh-7.5p1-fips-enable-1/servconf.c
--- openssh-7.5p1-fips/servconf.c	2017-02-02 16:23:48.132869419 -0800
+++ openssh-7.5p1-fips-enable-1/servconf.c	2017-02-15 17:03:52.938988153 -0800
@@ -164,6 +164,7 @@ initialize_server_options(ServerOptions
 	options->version_addendum = NULL;
 	options->fingerprint_hash = -1;
 	options->disable_forwarding = -1;
+	options->fips_mode = -1;
 }
 
 /* Returns 1 if a string option is unset or set to "none" or 0 otherwise. */
@@ -336,6 +337,13 @@ fill_default_server_options(ServerOption
 		options->fingerprint_hash = SSH_FP_HASH_DEFAULT;
 	if (options->disable_forwarding == -1)
 		options->disable_forwarding = 0;
+	if (options->fips_mode == -1)
+		options->fips_mode = 0;
+
+        /* Call FIPS_mode_set depending on whether fips mode is set or not */
+        if (options->fips_mode)
+                if (!FIPS_mode_set(1))
+                        fatal("FIPS mode could not be set");
 
 	assemble_algorithms(options);
 
@@ -421,7 +429,8 @@ typedef enum {
 	sAuthenticationMethods, sHostKeyAgent, sPermitUserRC,
 	sStreamLocalBindMask, sStreamLocalBindUnlink,
 	sAllowStreamLocalForwarding, sFingerprintHash, sDisableForwarding,
-	sDeprecated, sIgnore, sUnsupported
+	sDeprecated, sIgnore, sUnsupported,
+	sFipsMode
 } ServerOpCodes;
 
 #define SSHCFG_GLOBAL	0x01	/* allowed in main section of sshd_config */
@@ -564,6 +573,7 @@ static struct {
 	{ "allowstreamlocalforwarding", sAllowStreamLocalForwarding, SSHCFG_ALL },
 	{ "fingerprinthash", sFingerprintHash, SSHCFG_GLOBAL },
 	{ "disableforwarding", sDisableForwarding, SSHCFG_ALL },
+	{ "fipsmode", sFipsMode, SSHCFG_GLOBAL },
 	{ NULL, sBadOption, 0 }
 };
 
@@ -1839,6 +1849,10 @@ process_server_config_line(ServerOptions
 			options->fingerprint_hash = value;
 		break;
 
+	case sFipsMode:
+		intptr = &options->fips_mode;
+		goto parse_flag;
+
 	case sDeprecated:
 	case sIgnore:
 	case sUnsupported:
@@ -2278,6 +2285,7 @@ dump_config(ServerOptions *o)
 	dump_cfg_fmtint(sAllowStreamLocalForwarding, o->allow_streamlocal_forwarding);
 	dump_cfg_fmtint(sStreamLocalBindUnlink, o->fwd_opts.streamlocal_bind_unlink);
 	dump_cfg_fmtint(sFingerprintHash, o->fingerprint_hash);
+	dump_cfg_fmtint(sFipsMode, o->fips_mode);
 
 	/* string arguments */
 	dump_cfg_string(sPidFile, o->pid_file);
Only in openssh-7.5p1-fips-enable-1: servconf.c.orig
diff -rup openssh-7.5p1-fips/servconf.h openssh-7.5p1-fips-enable-1/servconf.h
--- openssh-7.5p1-fips/servconf.h	2017-02-08 15:46:03.518738277 -0800
+++ openssh-7.5p1-fips-enable-1/servconf.h	2017-02-15 17:03:52.938988153 -0800
@@ -189,6 +189,7 @@ typedef struct {
 	char   *auth_methods[MAX_AUTH_METHODS];
 
 	int	fingerprint_hash;
+	int fips_mode;
 }       ServerOptions;
 
 /* Information about the incoming connection as used by Match */
diff -rup openssh-7.5p1-fips/ssh_config openssh-7.5p1-fips-enable-1/ssh_config
--- openssh-7.5p1-fips/ssh_config	2017-02-02 15:26:47.536946693 -0800
+++ openssh-7.5p1-fips-enable-1/ssh_config	2017-02-15 17:03:52.938988153 -0800
@@ -37,6 +37,7 @@
 #   IdentityFile ~/.ssh/id_ecdsa
 #   IdentityFile ~/.ssh/id_ed25519
 #   Port 22
+#   FipsMode no
 #   Protocol 2
 #   Cipher 3des
 #   Ciphers aes128-ctr,aes192-ctr,aes256-ctr,arcfour256,arcfour128,aes128-cbc,3des-cbc
diff -rup openssh-7.5p1-fips/ssh_config.0 openssh-7.5p1-fips-enable-1/ssh_config.0
--- openssh-7.5p1-fips/ssh_config.0	2017-02-02 15:26:47.536946693 -0800
+++ openssh-7.5p1-fips-enable-1/ssh_config.0	2017-02-15 17:03:52.938988153 -0800
@@ -362,6 +362,10 @@ DESCRIPTION
              Specifies the hash algorithm used when displaying key
              fingerprints.  Valid options are: md5 and sha256 (the default).
 
+     FipsMode
+             Enables or disables FIPS mode. Requires FIPS capable ssl modules.
+             The default is no.
+
      ForwardAgent
              Specifies whether the connection to the authentication agent (if
              any) will be forwarded to the remote machine.  The argument must
diff -rup openssh-7.5p1-fips/ssh_config.5 openssh-7.5p1-fips-enable-1/ssh_config.5
--- openssh-7.5p1-fips/ssh_config.5	2017-02-02 15:26:47.536946693 -0800
+++ openssh-7.5p1-fips-enable-1/ssh_config.5	2017-02-15 17:03:52.938988153 -0800
@@ -658,6 +658,10 @@ Valid options are:
 and
 .Cm sha256
 (the default).
+.It Cm FipsMode
+Enables or disables FIPS mode. Requires FIPS capable ssl modules.
+The default is
+.Cm no .
 .It Cm ForwardAgent
 Specifies whether the connection to the authentication agent (if any)
 will be forwarded to the remote machine.
Only in openssh-7.5p1-fips-enable-1: ssh.c.orig
diff -rup openssh-7.5p1-fips/sshd_config openssh-7.5p1-fips-enable-1/sshd_config
--- openssh-7.5p1-fips/sshd_config	2017-02-02 15:26:47.536946693 -0800
+++ openssh-7.5p1-fips-enable-1/sshd_config	2017-02-15 17:03:52.938988153 -0800
@@ -105,6 +105,8 @@ AuthorizedKeysFile	.ssh/authorized_keys
 #ChrootDirectory none
 #VersionAddendum none
 
+#FipsMode no
+
 # no default banner path
 #Banner none
 
diff -rup openssh-7.5p1-fips/sshd_config.0 openssh-7.5p1-fips-enable-1/sshd_config.0
--- openssh-7.5p1-fips/sshd_config.0	2017-02-02 15:26:47.536946693 -0800
+++ openssh-7.5p1-fips-enable-1/sshd_config.0	2017-02-15 17:03:52.938988153 -0800
@@ -331,6 +331,10 @@ DESCRIPTION
              Specifies the hash algorithm used when logging key fingerprints.
              Valid options are: md5 and sha256.  The default is sha256.
 
+     FipsMode
+             Enables or disables FIPS mode. Requires FIPS capable ssl modules.
+             The default is no.
+
      ForceCommand
              Forces the execution of the command specified by ForceCommand,
              ignoring any command supplied by the client and ~/.ssh/rc if
diff -rup openssh-7.5p1-fips/sshd_config.5 openssh-7.5p1-fips-enable-1/sshd_config.5
--- openssh-7.5p1-fips/sshd_config.5	2017-02-02 15:26:47.536946693 -0800
+++ openssh-7.5p1-fips-enable-1/sshd_config.5	2017-02-15 17:03:52.942988153 -0800
@@ -578,6 +578,10 @@ and
 .Cm sha256 .
 The default is
 .Cm sha256 .
+.It Cm FipsMode
+Enables or disables FIPS mode. Requires FIPS capable ssl modules.
+The default is 
+.Cm no .
 .It Cm ForceCommand
 Forces the execution of the command specified by
 .Cm ForceCommand ,
