From a778da75578051d31a4badcf781f6d3af2b6b65b Mon Sep 17 00:00:00 2001
From: Bo Gan <ganb@vmware.com>
Date: Mon, 28 Oct 2019 17:14:22 -0700
Subject: [PATCH] disable git invocation

---
 CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 490bec8..0c897a7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,6 +13,8 @@ if(MINIMAL)
 	add_definitions(-DTINI_MINIMAL=1)
 endif()
 
+if(FALSE)
+
 # Extract git version and dirty-ness
 execute_process (
   COMMAND git --git-dir "${PROJECT_SOURCE_DIR}/.git" --work-tree "${PROJECT_SOURCE_DIR}" log -n 1 --date=local --pretty=format:%h
@@ -27,6 +29,8 @@ execute_process(
   OUTPUT_VARIABLE git_dirty_check_out
 )
 
+endif()
+
 if("${git_version_check_ret}" EQUAL 0)
   set(tini_VERSION_GIT " - git.${tini_VERSION_GIT}")
   if(NOT "${git_dirty_check_out}" STREQUAL "")
-- 
2.7.4

