From 180ae0478053739a0a401dc81e79fb7127fcf8cf Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <sshedi@vmware.com>
Date: Thu, 8 Sep 2022 13:31:03 +0530
Subject: [PATCH] Don't check for libssp, this brings gcc while installing
 ntpsec

Signed-off-by: Shreenidhi Shedi <sshedi@vmware.com>
---
 wscript | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/wscript b/wscript
index c7c5468..049ef0c 100644
--- a/wscript
+++ b/wscript
@@ -312,6 +312,7 @@ def configure(ctx):
     if ret:
         ctx.env.LDFLAGS += ["-lbsd"]
 
+    """
     # -lssp and -lssp_nonshared may be needed by older gcc to
     # support "-fstack-protector-all"
     ret = ctx.check_cc(lib="ssp", mandatory=False,
@@ -323,6 +324,7 @@ def configure(ctx):
                        comment="libssp_nonshared")
     if ret:
         ctx.env.LDFLAGS += ["-lssp_nonshared"]
+    """
 
     cc_test_flags = [
         ('f_stack_protector_all', '-fstack-protector-all'),
-- 
2.25.1

