From 146bd711714a9129f9237e665c0d98ce4ab75d27 Mon Sep 17 00:00:00 2001
From: Bo Gan <ganb@vmware.com>
Date: Mon, 14 Sep 2020 16:22:53 -0700
Subject: [PATCH] i40e: Fix minor compilation error

Signed-off-by: Bo Gan <ganb@vmware.com>
---
 src/i40e_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/i40e_main.c b/src/i40e_main.c
index 33f7fc3..eda18c0 100644
--- a/src/i40e_main.c
+++ b/src/i40e_main.c
@@ -16382,7 +16382,7 @@ debug_mode_clear:
  * remediation.
  **/
 static pci_ers_result_t i40e_pci_error_detected(struct pci_dev *pdev,
-						enum pci_channel_state error)
+						pci_channel_state_t error)
 {
 	struct i40e_pf *pf = pci_get_drvdata(pdev);
 
-- 
2.7.4

