diff --git a/photon_installer/installer.py b/photon_installer/installer.py
index 7c50b29..7b9ab39 100644
--- a/photon_installer/installer.py
+++ b/photon_installer/installer.py
@@ -2343,6 +2343,10 @@ password_pbkdf2 {grub_user} {grub_password_hash}
             # add lvm2 package to install list
             self._add_packages_to_install('lvm2')
 
+        # add btrfs-progs if any partition uses btrfs
+        if any(p.get('filesystem') == 'btrfs' for p in partitions):
+            self._add_packages_to_install('btrfs-progs')
+
         # Create partitions_data (needed for mk-setup-grub.sh)
         for partition in partitions:
             if 'mountpoint' in partition and not partition.get('shadow', False):
