diff --git a/photon_installer/ostree-release-repo.conf b/photon_installer/ostree-release-repo.conf
index e9af1518..07fc1544 100644
--- a/photon_installer/ostree-release-repo.conf
+++ b/photon_installer/ostree-release-repo.conf
@@ -1,2 +1,2 @@
-OSTREEREPOURL=https://packages.vmware.com/photon/rpm-ostree/base/dev/x86_64/repo
-OSTREEREFS=photon/4.0/x86_64/minimal
+OSTREEREPOURL=https://packages.vmware.com/photon/rpm-ostree/base/5.0/{}/repo
+OSTREEREFS=photon/5.0/{}/minimal
diff --git a/photon_installer/ostreeinstaller.py b/photon_installer/ostreeinstaller.py
index 49da9385..680819c0 100644
--- a/photon_installer/ostreeinstaller.py
+++ b/photon_installer/ostreeinstaller.py
@@ -20,10 +20,9 @@ class OstreeInstaller(object):
     def __init__(self, installer):
         self.repo_config = {}
         self.installer_path = installer.installer_path
-        self.repo_read_conf()
-
         # simulate inheritance
         self.install_config = installer.install_config
+        self.repo_read_conf()
         self.logger = installer.logger
         if self.install_config['ui']:
             self.progress_bar = installer.progress_bar
@@ -45,7 +44,7 @@ class OstreeInstaller(object):
         with open(conf_path) as repo_conf:
             for line in repo_conf:
                 name, value = line.partition("=")[::2]
-                self.repo_config[name] = value.strip(' \n\t\r')
+                self.repo_config[name] = str(value.strip(' \n\t\r')).format(self.install_config['arch'])
 
     def pull_repo(self, repo_url, repo_ref):
         self.run([['ostree', 'remote', 'add', '--repo={}/ostree/repo'.format(self.photon_root), '--set=gpg-verify=false', 'photon', '{}'
