diff --git a/leveldb.c b/leveldb.c
index 5a4d9de..b42a57e 100644
--- a/leveldb.c
+++ b/leveldb.c
@@ -743,24 +743,8 @@ int parseServiceInfo(int fd, char *name, struct service *service, int honorHide,
 
 /* returns -1 on error */
 int currentRunlevel(void) {
-    FILE *p;
-    char response[50];
-
-    p = popen("/sbin/runlevel", "r");
-    if (!p)
-        return -1;
-
-    if (!fgets(response, sizeof(response), p)) {
-        pclose(p);
-        return -1;
-    }
-
-    pclose(p);
-
-    if (response[1] != ' ' || !isdigit(response[2]) || response[3] != '\n')
-        return -1;
-
-    return response[2] - '0';
+    // Photon OS only supports running chkconfig from runlevel 3
+    return 3;
 }
 
 int findServiceEntries(char *name, int level, glob_t *globresptr) {
