From 2c927d164b5836284a88caae498eb2b525c70042 Mon Sep 17 00:00:00 2001
From: Keerthana K <keerthanak@vmware.com>
Date: Mon, 31 Oct 2022 08:14:51 +0000
Subject: [PATCH] stalld: Add debug print for starving tasks

Signed-off-by: Keerthana K <keerthanak@vmware.com>
---
 src/stalld.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/stalld.c b/src/stalld.c
index 1eb0a19..dccf0a8 100644
--- a/src/stalld.c
+++ b/src/stalld.c
@@ -1703,8 +1703,8 @@ int boost_cpu_starving_vector(struct cpu_starving_task_info *vector, int nr_cpus
 
 		cpu = &cpu_starving_vector[i];
 
-		if (config_verbose && cpu->pid)
-			log_msg("\t cpu %d: pid: %d starving for %llu\n", i, cpu->pid, (now - cpu->since));
+		if (cpu->pid)
+			log_msg("\t cpu %d: task: %s-%d starving for %llu seconds\n", i, cpu->task.comm, cpu->pid, (now - cpu->since));
 
 		if (config_log_only)
 			continue;
-- 
2.28.0

