From 10f3f5ff169af1c7b26cdc0494e3519e0fe253e2 Mon Sep 17 00:00:00 2001
From: "Luis Claudio R. Goncalves" <lgoncalv@redhat.com>
Date: Thu, 12 May 2022 11:34:59 -0300
Subject: [PATCH 317/319] Linux 5.10.111-rt66 REBASE

Signed-off-by: Luis Claudio R. Goncalves <lgoncalv@redhat.com>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 21988f9ad53f1..d42c0971b041d 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt34
+-rt66
-- 
2.35.3

