diff -dupr a/Makefile-tests.am b/Makefile-tests.am
--- a/Makefile-tests.am	2017-04-13 15:42:06.000000000 +0000
+++ b/Makefile-tests.am	2017-09-16 01:16:03.069641948 +0000
@@ -55,7 +55,7 @@ $(foreach pkg,$(testpackages),$(eval $(c
 
 tests/common/compose/yum/repo/repodata/repomd.xml: $(testpackages)
 	(cd tests/common/compose/yum/repo && \
-	 createrepo_c --no-database $$PWD)
+	 createrepo --no-database $$PWD)
 
 tests/common/compose/test-repo.repo: tests/common/compose/test-repo.repo.in \
                                      tests/common/compose/yum/repo/repodata/repomd.xml
@@ -79,7 +79,6 @@ tests_check_postprocess_LDADD = $(PKGDEP
 
 tests/check/test-compose.sh: tests/common/compose/test-repo.repo
 
-tests/check/test-ucontainer.sh: tests/common/compose/test-repo.repo
 
 uninstalled_test_programs = \
 	tests/check/jsonutil			\
@@ -90,7 +89,6 @@ uninstalled_test_programs = \
 uninstalled_test_scripts = \
 	tests/check/test-basic.sh \
 	tests/check/test-compose.sh \
-	tests/check/test-ucontainer.sh \
 	$(NULL)
 
 uninstalled_test_extra_programs = dbus-run-session
