diff -dupr a/tests/check/test-compose.sh b/tests/check/test-compose.sh
--- a/tests/check/test-compose.sh	2017-04-13 15:42:06.000000000 +0000
+++ b/tests/check/test-compose.sh	2017-09-15 23:38:06.424734259 +0000
@@ -31,7 +31,7 @@ export RPMOSTREE_UNINSTALLED_PKGLIBDIR=$
 
 composedir=${commondir}/compose
 
-arch=$(arch)
+arch=x86_64
 if ! test "${arch}" = x86_64; then
     echo 1>&2 "$0 can be run only on x86_64"; echo "1..0" ; exit 77
 fi
