diff -Naur a/jdk/make/src/classes/build/tools/generatecurrencydata/GenerateCurrencyData.java b/jdk/make/src/classes/build/tools/generatecurrencydata/GenerateCurrencyData.java
--- a/jdk/make/src/classes/build/tools/generatecurrencydata/GenerateCurrencyData.java	2018-04-20 19:48:45.000000000 +0000
+++ b/jdk/make/src/classes/build/tools/generatecurrencydata/GenerateCurrencyData.java	2024-01-10 09:22:08.998451848 +0000
@@ -284,8 +284,8 @@
             checkCurrencyCode(newCurrency);
             String timeString = currencyInfo.substring(4, length - 4);
             long time = format.parse(timeString).getTime();
-            if (Math.abs(time - System.currentTimeMillis()) > ((long) 10) * 365 * 24 * 60 * 60 * 1000) {
-                throw new RuntimeException("time is more than 10 years from present: " + time);
+            if (Math.abs(time - System.currentTimeMillis()) > ((long) 10) * 365 * 24 * 60 * 60 * 1000 * 5) {
+                throw new RuntimeException("time is more than 50 years from present: " + time);
             }
             specialCaseCutOverTimes[specialCaseCount] = time;
             specialCaseOldCurrencies[specialCaseCount] = oldCurrency;
