From 74581ec4f4ebd7abf68c5ab042164dc0b77133ba Mon Sep 17 00:00:00 2001
From: Nikola Pajkovsky <npajkovs@redhat.com>
Date: Wed, 17 Mar 2010 17:24:48 +0100
Subject: [PATCH] fix strcpy overlap memory

Signed-off-by: Nikola Pajkovsky <npajkovs@redhat.com>
---
 src/ifaces.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/ifaces.c b/src/ifaces.c
index 8447f51..ecb8a2d 100644
--- a/src/ifaces.c
+++ b/src/ifaces.c
@@ -51,7 +51,7 @@ char *ltrim(char *buf)
     while ((*tmp == ' ') || (*tmp == '\t'))
         tmp++;
 
-    strcpy(buf, tmp);
+    memmove(buf, tmp, strlen(buf));
     return buf;
 }
 
-- 
1.6.6.1

