--- iptraf-3.0.0/src/iptraf.c.orig	2005-09-13 08:42:54.000000000 +0200
+++ iptraf-3.0.0/src/iptraf.c	2006-01-30 08:38:49.000000000 +0100
@@ -39,6 +39,7 @@
 #include <menurt.h>
 #include <winops.h>
 #include <msgboxes.h>
+#include <locale.h>
 #include "dirs.h"
 #include "deskman.h"
 #include "fltdefs.h"
@@ -499,6 +500,8 @@
     }
 #endif
 
+    setlocale(LC_ALL, "");
+
     strcpy(current_logfile, "");
     strcpy(graphing_logfile, "");
     strcpy(graphing_filter, "");
