diff -up iptraf-3.0.1/src/ifstats.c.old iptraf-3.0.1/src/ifstats.c
--- iptraf-3.0.1/src/ifstats.c.old	2007-11-02 12:49:25.000000000 +0100
+++ iptraf-3.0.1/src/ifstats.c	2007-11-02 12:49:38.000000000 +0100
@@ -26,6 +26,7 @@ details.
 #include <time.h>
 #include <sys/time.h>
 #include <sys/types.h>
+#include <linux/types.h>
 #include <sys/socket.h>
 #include <sys/ioctl.h>
 #include <netinet/in.h>
diff -up iptraf-3.0.1/src/serv.c.old iptraf-3.0.1/src/serv.c
--- iptraf-3.0.1/src/serv.c.old	2007-11-02 12:50:04.000000000 +0100
+++ iptraf-3.0.1/src/serv.c	2007-11-02 12:50:25.000000000 +0100
@@ -32,6 +32,7 @@ details.
 #include <netinet/ip.h>
 #include <netinet/ip6.h>
 #include <netinet/udp.h>
+#include <linux/types.h>
 #include <linux/if_packet.h>
 #include <linux/if_ether.h>
 #include <listbox.h>
diff -up iptraf-3.0.1/src/ifaces.c.old iptraf-3.0.1/src/ifaces.c
--- iptraf-3.0.1/src/ifaces.c.old	2007-11-02 12:50:37.000000000 +0100
+++ iptraf-3.0.1/src/ifaces.c	2007-11-02 12:50:54.000000000 +0100
@@ -27,6 +27,7 @@ details.
 #include <net/if.h>
 #include <fcntl.h>
 #include <string.h>
+#include <linux/types.h>
 #include <linux/if_packet.h>
 #include "links.h"
 #include "error.h"
