diff -up iptraf-3.0.1/src/serv.c.servmon-fix iptraf-3.0.1/src/serv.c
--- iptraf-3.0.1/src/serv.c.servmon-fix	2009-12-28 14:37:25.000000000 +0100
+++ iptraf-3.0.1/src/serv.c	2009-12-28 14:40:41.580853867 +0100
@@ -1052,7 +1052,6 @@ void servmon(char *ifname, struct portta
                     list.baridx = 1;
                 }
             }
-        }
         } else {
             if ((((struct ip6_hdr *) ipacket)->ip6_nxt == IPPROTO_TCP)
                 || (((struct ip6_hdr *) ipacket)->ip6_nxt ==
@@ -1073,6 +1072,7 @@ void servmon(char *ifname, struct portta
                 }
             }
         }
+        }
     }
 
     if (logging) {
