diff -up iptraf-3.0.1/src/packet.c.old iptraf-3.0.1/src/packet.c
--- iptraf-3.0.1/src/packet.c.old	2015-11-30 12:20:11.403710380 -0800
+++ iptraf-3.0.1/src/packet.c	2015-11-30 12:21:01.793669693 -0800
@@ -36,7 +36,7 @@ details.
 #include <linux/if_ether.h>
 #include <linux/netdevice.h>
 #include <linux/if_fddi.h>
-#include <linux/if_tr.h>
+#include <netinet/if_tr.h>
 #include <linux/isdn.h>
 #include <linux/sockios.h>
 #include <msgboxes.h>

diff -up iptraf-3.0.1/src/hostmon.c.old iptraf-3.0.1/src/hostmon.c
--- iptraf-3.0.1/src/hostmon.c.old	2015-11-30 12:42:19.445405357 -0800
+++ iptraf-3.0.1/src/hostmon.c	2015-11-30 12:43:01.623684187 -0800
@@ -32,7 +32,7 @@ details.
 #include <linux/if_ether.h>
 #include <linux/netdevice.h>
 #include <linux/if_fddi.h>
-#include <linux/if_tr.h>
+#include <netinet/if_tr.h>
 #include <net/if_arp.h>
 #include <stdlib.h>
 #include <time.h>
