diff -up iptraf-3.0.1/src/tr.c.old iptraf-3.0.1/src/tr.c
--- iptraf-3.0.1/src/tr.c.old	2005-09-13 08:42:54.000000000 +0200
+++ iptraf-3.0.1/src/tr.c	2007-11-02 12:30:26.000000000 +0100
@@ -7,7 +7,7 @@
  */
 
 #include <asm/types.h>
-#include <linux/if_tr.h>
+#include <netinet/if_tr.h>
 #include <netinet/in.h>
 
 unsigned int get_tr_ip_offset(unsigned char *pkt)
diff -up iptraf-3.0.1/src/othptab.c.old iptraf-3.0.1/src/othptab.c
--- iptraf-3.0.1/src/othptab.c.old	2007-09-25 11:47:35.000000000 +0200
+++ iptraf-3.0.1/src/othptab.c	2007-11-02 12:28:24.000000000 +0100
@@ -19,10 +19,10 @@ details.
 #include <stdlib.h>
 #include <asm/types.h>
 #include <sys/socket.h>
-#include <linux/if_ether.h>
-#include <linux/if_tr.h>
-#include <linux/netdevice.h>
-#include <linux/if_fddi.h>
+#include <netinet/if_ether.h>
+#include <netinet/if_tr.h>
+#include <linux/netdevice.h>
+#include <netinet/if_fddi.h>
 #include <winops.h>
 #include "arphdr.h"
 #include "options.h"
diff -up iptraf-3.0.1/src/Makefile.old iptraf-3.0.1/src/Makefile
--- iptraf-3.0.1/src/Makefile.old	2007-11-02 12:27:07.000000000 +0100
+++ iptraf-3.0.1/src/Makefile	2007-11-02 12:29:50.000000000 +0100
@@ -18,7 +18,7 @@ VERSION 	= -DVERSION=\"$(VERNUMBER)\"
 BINDIR		= ../../iptraf-$(VERNUMBER).bin.$(ARCH)
 
 CC		= gcc
-LIBS		= -L../support -ltextbox -lpanel -lncurses   # in this order!
+LIBS		= -L../support -ltextbox -lpanelw -lncursesw   # in this order!
 
 # comment this one out to omit debug code when done.
 
@@ -39,7 +39,7 @@ LDOPTS		= #-static
 # you may want to change this to point to your ncurses include directory
 # if the ncurses include files are not in the default location.
 
-INCLUDEDIR	= -I/usr/include/ncurses -I../support
+INCLUDEDIR	= -I/usr/include/ncursesw -I../support
 
 # You can uncomment this one to disable the backspace key in input fields.
 # This means you must use the Del key or Ctrl+H combination to erase the
@@ -162,7 +162,7 @@ install:
 # and the compiled programs for release as a ready-to-run distribution.
 
 dist-bin: all
-	$(CC) $(LDOPTS) $(PROF) -o iptraf $(OBJS) -L../support -ltextbox /usr/lib/libpanel.a /usr/lib/libncurses.a
+	$(CC) $(LDOPTS) $(PROF) -o iptraf $(OBJS) -L../support -ltextbox /usr/lib/libpanelw.a /usr/lib/libncursesw.a
 	rm -rf $(BINDIR)
 	mkdir $(BINDIR)
 	/bin/cp -p ../CHANGES ../LICENSE ../FAQ ../INSTALL ../README* \
@@ -178,7 +178,7 @@ dist-bin: all
 # Just in case anyone needs to link a static binary
 #
 static: $(OBJS)
-	$(CC) -static $(PROC) -o iptraf $(OBJS) ../support/libtextbox.a -lpanel -lncurses
+	$(CC) -static $(PROC) -o iptraf $(OBJS) ../support/libtextbox.a -lpanelw -lncursesw
 	$(CC) -static -o rvnamed rvnamed.o getpath.o
 	$(CC) -static -o rawtime rawtime.c
 
diff -up iptraf-3.0.1/src/tcptable.h.old iptraf-3.0.1/src/tcptable.h
--- iptraf-3.0.1/src/tcptable.h.old	2007-09-25 11:46:03.000000000 +0200
+++ iptraf-3.0.1/src/tcptable.h	2007-11-02 12:35:56.000000000 +0100
@@ -20,11 +20,11 @@
 #include <arpa/inet.h>
 #include <net/if_arp.h>
 #include <asm/types.h>
-#include <linux/if_packet.h>
-#include <linux/if_ether.h>
-#include <linux/netdevice.h>
-#include <linux/if_fddi.h>
-#include <linux/if_tr.h>
+#include <linux/if_packet.h>
+#include <netinet/if_ether.h>
+#include <linux/netdevice.h>
+#include <netinet/if_fddi.h>
+#include <netinet/if_tr.h>
 // #include <net/if.h>
 #include <netinet/ip.h>
 #include <netinet/udp.h>
