diff -u iptraf-3.0.0/src/install.sh iptraf-3.0.0/src/install.sh
--- iptraf-3.0.0/src/install.sh	2006-01-30 15:37:34.000000000 +0100
+++ iptraf-3.0.0/src/install.sh	2006-01-31 15:10:09.000000000 +0100
@@ -23,9 +23,9 @@
 echo "*** Installing executable programs and preparing work directories"
 echo
 echo ">>> Installing iptraf in $TARGET"
-$INSTALL -m 0700 -o root -g root -s iptraf $TARGET
+$INSTALL -m 0700 -s iptraf $TARGET
 echo ">>> Installing rvnamed in $TARGET"
-$INSTALL -m 0700 -o root -g root -s rvnamed $TARGET
+$INSTALL -m 0700 -s rvnamed $TARGET
 
 if [ ! -d $WORKDIR ]; then
     echo ">>> Creating IPTraf work directory $WORKDIR"
@@ -34,21 +34,21 @@
     rm -f $WORKDIR/othfilter.dat
 fi
 
-$INSTALL -m 0700 -o root -g root -d $WORKDIR
+$INSTALL -m 0700 -d $WORKDIR
 
 if [ ! -d $LOGDIR ]; then
     echo ">>> Creating IPTraf log directory $LOGDIR"
 else
     echo ">>> IPTraf log directory $LOGDIR already exists"
 fi
-$INSTALL -m 0700 -o root -g root -d $LOGDIR
+$INSTALL -m 0700 -d $LOGDIR
 
 if [ ! -d $LOCKDIR ]; then
     echo ">>> Creating IPTraf lockfile directory $LOCKDIR"
 else
     echo ">>> IPTraf lockfile directory $LOCKDIR already exists"
 fi
-$INSTALL -m 0700 -o root -g root -d $LOCKDIR
+$INSTALL -m 0700 -d $LOCKDIR
 echo
 echo
 echo "*** iptraf, and rvnamed executables are in $TARGET"
@@ -74,7 +74,6 @@
 fi
 ####################################################################
 
-cat version > $WORKDIR/version
 
 echo 
 echo
@@ -85,13 +84,13 @@
 echo "this version. You can view this file now (will require the 'less'"
 echo "program in /usr/bin.  Press Q to quit when done)."
 echo
-echo -n "Would you like to view the RELEASE-NOTES file now (Y/N)? "; read YESNO
-
-if [ "$YESNO" = "y" -o "$YESNO" = "Y" ]; then
-    less ../RELEASE-NOTES
-fi
-
-clear
+#echo -n "Would you like to view the RELEASE-NOTES file now (Y/N)? "; read YESNO
+#
+#if [ "$YESNO" = "y" -o "$YESNO" = "Y" ]; then
+#    less ../RELEASE-NOTES
+#fi
+#
+#clear
 echo
 echo "====================================================================="
 echo
