From 99b578501643377e0b1994b2a068b790d189d5ad Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Wed, 13 Jun 2018 09:41:01 -0400
Subject: [PATCH] remove extra decl

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 src/efibootmgr.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/efibootmgr.c b/src/efibootmgr.c
index de38f01..4e1a680 100644
--- a/src/efibootmgr.c
+++ b/src/efibootmgr.c
@@ -1536,9 +1536,6 @@ parse_opts(int argc, char **argv)
 					       "invalid numeric value %s\n",
 					       optarg);
 			}
-                        /* XXX efivar-36 accidentally doesn't have a public
-                         * header for this */
-			extern int efi_set_verbose(int verbosity, FILE *errlog);
 			efi_set_verbose(opts.verbose - 2, stderr);
 			break;
 		case 'V':
