From fd704d3341738045e1b2695db4779014c40df25f Mon Sep 17 00:00:00 2001
From: Kazuyoshi Kato <katokazu@amazon.com>
Date: Mon, 18 Apr 2022 21:02:06 +0000
Subject: [PATCH] Build bin/gen-manpages instead of using "go run"

Signed-off-by: Kazuyoshi Kato <katokazu@amazon.com>
Signed-off-by: Srivatsa S. Bhat (VMware) <srivatsa@csail.mit.edu>
---
 Makefile | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index f3ae858b043..a6bad72efec 100644
--- a/Makefile
+++ b/Makefile
@@ -253,13 +253,13 @@ mandir:
 # Kept for backwards compatibility
 genman: man/containerd.8 man/ctr.8
 
-man/containerd.8: FORCE
+man/containerd.8: bin/gen-manpages FORCE
 	@echo "$(WHALE) $@"
-	$(GO) run -mod=readonly ${GO_TAGS} cmd/gen-manpages/main.go $(@F) $(@D)
+	$< $(@F) $(@D)
 
-man/ctr.8: FORCE
+man/ctr.8: bin/gen-manpages FORCE
 	@echo "$(WHALE) $@"
-	$(GO) run -mod=readonly ${GO_TAGS} cmd/gen-manpages/main.go $(@F) $(@D)
+	$< $(@F) $(@D)
 
 man/%: docs/man/%.md FORCE
 	@echo "$(WHALE) $@"
