From f9ea3c26bda85a265b2c7e211bac4f91ba760661 Mon Sep 17 00:00:00 2001
From: Daniel Wagner <wagi@monom.org>
Date: Thu, 16 Mar 2023 08:06:42 +0100
Subject: [PATCH 353/353] Linux 4.19.277-rt122 REBASE

Signed-off-by: Daniel Wagner <wagi@monom.org>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 1199ebade17b..4de2714cd138 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt16
+-rt122
-- 
2.39.2

