From a7c6caecd9c6fe1286cba5b1a43de479b003adf1 Mon Sep 17 00:00:00 2001
From: Thomas Gleixner <tglx@linutronix.de>
Date: Fri, 8 Jul 2011 20:25:16 +0200
Subject: [PATCH 288/320] Add localversion for -RT release

Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
---
 localversion-rt | 1 +
 1 file changed, 1 insertion(+)
 create mode 100644 localversion-rt

diff --git a/localversion-rt b/localversion-rt
new file mode 100644
index 000000000000..21988f9ad53f
--- /dev/null
+++ b/localversion-rt
@@ -0,0 +1 @@
+-rt34
-- 
2.39.2

