From 9d0ab6ecadf34055e670f887606724517d68dbae Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <shreenidhi.shedi@broadcom.com>
Date: Mon, 10 Aug 2026 20:29:08 +0530
Subject: [PATCH] Disable tegra in aarch64

Signed-off-by: Shreenidhi Shedi <shreenidhi.shedi@broadcom.com>
---
 meson.build | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/meson.build b/meson.build
index 58be803..3ea2bb8 100644
--- a/meson.build
+++ b/meson.build
@@ -174,13 +174,13 @@ if gallium_drivers.contains('auto')
     elif ['aarch64'].contains(host_machine.cpu_family())
       gallium_drivers = [
         'v3d', 'vc4', 'freedreno', 'etnaviv', 'svga',
-        'tegra', 'virgl', 'lima', 'panfrost', 'llvmpipe', 'softpipe', 'iris',
+        'virgl', 'lima', 'panfrost', 'llvmpipe', 'softpipe', 'iris',
         'zink', 'asahi'
       ]
     elif ['arm'].contains(host_machine.cpu_family())
       gallium_drivers = [
         'v3d', 'vc4', 'freedreno', 'etnaviv', 'svga',
-        'tegra', 'virgl', 'lima', 'panfrost', 'llvmpipe', 'softpipe', 'iris',
+        'virgl', 'lima', 'panfrost', 'llvmpipe', 'softpipe', 'iris',
         'zink'
       ]
     elif ['mips', 'mips64', 'ppc', 'ppc64', 'riscv32', 'riscv64'].contains(host_machine.cpu_family())
@@ -206,7 +206,7 @@ if gallium_drivers.contains('auto')
 elif gallium_drivers.contains('all')
    gallium_drivers = [
      'r300', 'r600', 'radeonsi', 'crocus', 'v3d', 'vc4', 'freedreno', 'etnaviv', 'i915',
-     'nouveau', 'svga', 'tegra', 'virgl', 'lima', 'panfrost', 'llvmpipe', 'softpipe', 'iris',
+     'nouveau', 'svga', 'virgl', 'lima', 'panfrost', 'llvmpipe', 'softpipe', 'iris',
      'zink', 'd3d12', 'asahi', 'rocket', 'ethosu'
    ]
 endif
-- 
2.55.0

