--- a/pam/crond	2010-08-13 05:31:07.000000000 -0700
+++ b/pam/crond	2016-08-29 11:39:40.536626806 -0700
@@ -4,7 +4,7 @@
 #
 # No PAM authentication called, auth modules not needed
 account    required   pam_access.so
-account    include    password-auth
+account    include    system-account
 session    required   pam_loginuid.so
-session    include    password-auth
-auth       include    password-auth
+session    include    system-session
+auth       include    system-auth
