--- a/psutil/tests/test_misc.py	2021-10-19 10:32:21.465141477 +0000
+++ b/psutil/tests/test_misc.py	2021-10-19 10:32:42.801823046 +0000
@@ -353,7 +353,7 @@ class TestMisc(PsutilTestCase):
 
     def test_setup_script(self):
         setup_py = os.path.join(ROOT_DIR, 'setup.py')
-        if CI_TESTING and not os.path.exists(setup_py):
+        if not os.path.exists(setup_py):
             return self.skipTest("can't find setup.py")
         module = import_module_by_path(setup_py)
         self.assertRaises(SystemExit, module.setup)
--- a/psutil/tests/test_system.py	2021-10-19 10:30:52.542301274 +0000
+++ b/psutil/tests/test_system.py	2021-10-19 10:32:08.196717649 +0000
@@ -200,7 +200,7 @@ class TestMiscAPIs(PsutilTestCase):
         self.assertGreater(bt, 0)
         self.assertLess(bt, time.time())
 
-    @unittest.skipIf(CI_TESTING and not psutil.users(), "unreliable on CI")
+    @unittest.skipIf(not psutil.users(), "unreliable on CI")
     def test_users(self):
         users = psutil.users()
         self.assertNotEqual(users, [])
@@ -591,6 +591,7 @@ class TestDiskAPIs(PsutilTestCase):
     def test_disk_usage_bytes(self):
         psutil.disk_usage(b'.')
 
+    @unittest.skip("unreliable on CI")
     def test_disk_partitions(self):
         # all = False
         ls = psutil.disk_partitions(all=False)
