From 25368e9fd6fbd6a6c05b90bc6c1872f48f53d920 Mon Sep 17 00:00:00 2001
From: Satya Naga Rajesh <svasamsetty@vmware.com>
Date: Tue, 16 Nov 2021 17:00:33 +0530
Subject: [PATCH] sysdig: Fix build

---
 CMakeListsGtestInclude.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeListsGtestInclude.cmake b/CMakeListsGtestInclude.cmake
index 45ff47b..76c0f1b 100644
--- a/CMakeListsGtestInclude.cmake
+++ b/CMakeListsGtestInclude.cmake
@@ -23,7 +23,7 @@ project(googletest-download NONE)
 include(ExternalProject)
 ExternalProject_Add(googletest
   GIT_REPOSITORY    https://github.com/google/googletest.git
-  GIT_TAG           master
+  GIT_TAG           main
   SOURCE_DIR        "${CMAKE_CURRENT_BINARY_DIR}/googletest-src"
   BINARY_DIR        "${CMAKE_CURRENT_BINARY_DIR}/googletest-build"
   CONFIGURE_COMMAND ""
-- 
2.17.1

