From 5f9afd3679c232d0260f1482d997393532706bee Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <shreenidhi.shedi@broadcom.com>
Date: Sun, 10 May 2026 22:04:16 +0530
Subject: [PATCH 1/8] CMakeListsGtestInclude.cmake: use user provided source
 archive

Signed-off-by: Shreenidhi Shedi <shreenidhi.shedi@broadcom.com>
---
 CMakeListsGtestInclude.cmake | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/CMakeListsGtestInclude.cmake b/CMakeListsGtestInclude.cmake
index 76c0f1b1..6b241fa8 100644
--- a/CMakeListsGtestInclude.cmake
+++ b/CMakeListsGtestInclude.cmake
@@ -20,10 +20,13 @@ cmake_minimum_required(VERSION 2.8.2)
 
 project(googletest-download NONE)
 
+set(GTEST_URL "" CACHE STRING "URL for googletest source tarball")
+set(GTEST_URL_MD5 "" CACHE STRING "MD5 checksum for googletest source tarball")
+
 include(ExternalProject)
 ExternalProject_Add(googletest
-  GIT_REPOSITORY    https://github.com/google/googletest.git
-  GIT_TAG           main
+  URL               "${GTEST_URL}"
+  URL_MD5           "${GTEST_URL_MD5}"
   SOURCE_DIR        "${CMAKE_CURRENT_BINARY_DIR}/googletest-src"
   BINARY_DIR        "${CMAKE_CURRENT_BINARY_DIR}/googletest-build"
   CONFIGURE_COMMAND ""
-- 
2.54.0

