diff --git a/CMakeListsGtestInclude.cmake b/CMakeListsGtestInclude.cmake
index 76c0f1b..801a155 100644
--- a/CMakeListsGtestInclude.cmake
+++ b/CMakeListsGtestInclude.cmake
@@ -22,8 +22,8 @@ project(googletest-download NONE)
 
 include(ExternalProject)
 ExternalProject_Add(googletest
-  GIT_REPOSITORY    https://github.com/google/googletest.git
-  GIT_TAG           main
+  URL               https://packages.vmware.com/photon/photon_sources/1.0/googletest-release-1.12.1.tar.gz
+  URL_MD5           e82199374acdfda3f425331028eb4e2a
   SOURCE_DIR        "${CMAKE_CURRENT_BINARY_DIR}/googletest-src"
   BINARY_DIR        "${CMAKE_CURRENT_BINARY_DIR}/googletest-build"
   CONFIGURE_COMMAND ""
