From 9d1abdb1dcef8747786388c9dbfcc4cbb3d3e899 Mon Sep 17 00:00:00 2001
From: Clark Williams <clrkwllms@kernel.org>
Date: Mon, 15 May 2023 10:23:52 -0500
Subject: [PATCH 52/52] 'Linux 6.1.28-rt10 REBASE'

Signed-off-by: Clark Williams <clrkwllms@kernel.org>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 045478966e9f..d79dde624aaa 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt7
+-rt10
-- 
2.40.1

