From 2bf2ed23897a29e11c2f84f0d81e1d1e70656993 Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <shreenidhi.shedi@broadcom.com>
Date: Wed, 15 May 2024 01:22:31 +0530
Subject: [PATCH] Change order of args to cloud-init in Ph4 and above

Signed-off-by: Shreenidhi Shedi <shreenidhi.shedi@broadcom.com>
---
 gosc-scripts/PhotonCustomization.sh | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/gosc-scripts/PhotonCustomization.sh b/gosc-scripts/PhotonCustomization.sh
index aeca8cf..0446aa3 100644
--- a/gosc-scripts/PhotonCustomization.sh
+++ b/gosc-scripts/PhotonCustomization.sh
@@ -82,9 +82,9 @@ RunCloudConfig()

   ${SLEEP} 2

-  local photon_ver="$(${GREP} -w "VERSION=" /etc/os-release | cut -d= -f2)"
+  local photon_ver="$(${GREP} -w "VERSION" /etc/os-release | cut -d'"' -f2)"
   # cloud-init on Photon 1.0 and 2.0 doesn't have clean option support
-  if [ "${photon_ver}" == "\"1.0\"" -o  "${photon_ver}" == "\"2.0\"" ]; then
+  if [ "${photon_ver}" \< "3.0" ]; then
     ${RMD} /var/lib/cloud/instance* /var/lib/cloud/seed
   else
     ${CLOUD_INIT} clean -s
@@ -97,7 +97,11 @@ RunCloudConfig()
   # If you don't quote the variable when you echo it, newlines are lost.
   ${ECHO} "${content}" > $ciConfigPath

-  cloud-init --file $ciConfigPath init 2>&1
+  if [ "${photon_ver}" \< "4.0" ]; then
+    ${CLOUD_INIT} --file ${ciConfigPath} init 2>&1
+  else
+    ${CLOUD_INIT} init --file ${ciConfigPath} 2>&1
+  fi

   Exec "${SYSTEMCTL} restart systemd-networkd" '' ''

--
2.39.3 (Apple Git-146)
