--- a/java/org/apache/tomcat/util/net/jsse/JSSEUtil.java
+++ b/java/org/apache/tomcat/util/net/jsse/JSSEUtil.java
@@ -266,13 +266,7 @@ public class JSSEUtil extends SSLUtilBas
             if (k != null && !"DKS".equalsIgnoreCase(certificate.getCertificateKeystoreType()) &&
                     "PKCS#8".equalsIgnoreCase(k.getFormat())) {
                 // Switch to in-memory key store
-                String provider = certificate.getCertificateKeystoreProvider();
-                if (provider == null) {
-                    ksUsed = KeyStore.getInstance(certificate.getCertificateKeystoreType());
-                } else {
-                    ksUsed = KeyStore.getInstance(certificate.getCertificateKeystoreType(),
-                            provider);
-                }
+                ksUsed = KeyStore.getInstance("JKS");
                 ksUsed.load(null,  null);
                 ksUsed.setKeyEntry(keyAlias, k, keyPassArray, ks.getCertificateChain(keyAlias));
             }
