--- a/java/org/apache/tomcat/util/net/jsse/JSSEUtil.java	2018-09-04 06:40:32.721928699 +0530
+++ b/java/org/apache/tomcat/util/net/jsse/JSSEUtil.java	2018-09-04 06:41:46.281931886 +0530
@@ -247,14 +247,8 @@ public class JSSEUtil extends SSLUtilBas
             if (k != null && !"DKS".equalsIgnoreCase(certificate.getCertificateKeystoreType()) &&
                     "PKCS#8".equalsIgnoreCase(k.getFormat())) {
                 // Switch to in-memory key store
-                String provider = certificate.getCertificateKeystoreProvider();
-                if (provider == null) {
-                    ksUsed = KeyStore.getInstance(certificate.getCertificateKeystoreType());
-                } else {
-                    ksUsed = KeyStore.getInstance(certificate.getCertificateKeystoreType(),
-                            provider);
-                }
-                ksUsed.load(null,  null);
+                ksUsed = KeyStore.getInstance("JKS");
+		ksUsed.load(null,  null);
                 ksUsed.setKeyEntry(keyAlias, k, keyPassArray, ks.getCertificateChain(keyAlias));
             }
             // Non-PKCS#8 key stores will use the original key store
