From 949ac0bc71b820c7abd320002fe1027387c9aab2 Mon Sep 17 00:00:00 2001
From: Brennan Lamoreaux <blamoreaux@vmware.com>
Date: Tue, 19 Jul 2022 22:06:03 +0000
Subject: [PATCH] adding option to set description field of module

---
 kpatch-build/kpatch-build | 15 +++++++++++++--
 1 file changed, 13 insertions(+), 2 deletions(-)

diff --git a/kpatch-build/kpatch-build b/kpatch-build/kpatch-build
index 1c5d7f7..b94354a 100755
--- a/kpatch-build/kpatch-build
+++ b/kpatch-build/kpatch-build
@@ -523,6 +523,7 @@ usage() {
 	echo "		-r, --sourcerpm         Specify kernel source RPM" >&2
 	echo "		-s, --sourcedir         Specify kernel source directory" >&2
 	echo "		-c, --config            Specify kernel config file" >&2
+	echo "		-D, --description       Specify module description file" >&2
 	echo "		-v, --vmlinux           Specify original vmlinux" >&2
 	echo "		-j, --jobs              Specify the number of make jobs" >&2
 	echo "		-t, --target            Specify custom kernel build targets" >&2
@@ -540,7 +541,7 @@ usage() {
 	echo "		                        (not recommended)" >&2
 }

-options="$(getopt -o ha:r:s:c:v:j:t:n:o:dR -l "help,archversion:,sourcerpm:,sourcedir:,config:,vmlinux:,jobs:,target:,name:,output:,oot-module:,oot-module-src:,debug,skip-gcc-check,skip-compiler-check,skip-cleanup,non-replace" -- "$@")" || die "getopt failed"
+options="$(getopt -o ha:r:s:c:D:v:j:t:n:o:dR -l "help,archversion:,sourcerpm:,sourcedir:,config:,description:,vmlinux:,jobs:,target:,name:,output:,oot-module:,oot-module-src:,debug,skip-gcc-check,skip-compiler-check,skip-cleanup,non-replace" -- "$@")" || die "getopt failed"

 eval set -- "$options"

@@ -569,6 +570,11 @@ while [[ $# -gt 0 ]]; do
 		CONFIGFILE="$(readlink -f "$2")"
 		shift
 		;;
+    -D|--description)
+		[[ ! -f "$2" ]] && die "description file '$2' not found"
+		DESC_FILE="$(readlink -f "$2")"
+		shift
+		;;
 	-v|--vmlinux)
 		[[ ! -f "$2" ]] && die "vmlinux file '$2' not found"
 		VMLINUX="$(readlink -f "$2")"
@@ -891,7 +897,7 @@ else
 		echo "Getting source rpm"
 		SRC_RPM_URL="https://packages.vmware.com/photon/$PHOTON_VERSION/photon_srpms_${PHOTON_VERSION}_x86_64/$SRC_RPM_NAME"

-		# allow downloading/copying of source rpm from either local or custom urls.
+		# allow downloading/copying of source rpm from either local or custom urls.
 		# Just need these variables to be exported before running to enable these options.
 		if [ -n "${SRC_RPM_LOCAL_PATH}" ]; then
 			cp $SRC_RPM_LOCAL_PATH $SRC_RPM_NAME || die "Couldn't find local src rpm"
@@ -1286,6 +1292,11 @@ KPATCH_LDFLAGS="$KPATCH_LDFLAGS" \
 CROSS_COMPILE="$CROSS_COMPILE"
 save_env

+if [[ $DESC_FILE ]]; then
+	echo "MODULE_DESCRIPTION(\"$(cat $DESC_FILE)\");" >> $TEMPDIR/patch/livepatch-patch-hook.c || die
+fi
+
+
 make "${MAKEVARS[@]}" 2>&1 | logger || die

 if [[ "$USE_KLP" -eq 1 ]]; then
--
2.30.3

