From dc96787571709030eca534382fc8a5addd1f052a Mon Sep 17 00:00:00 2001
From: Brennan Lamoreaux <blamoreaux@vmware.com>
Date: Tue, 19 Jul 2022 22:06:03 +0000
Subject: [PATCH 2/5] adding option to set description field of module

---
 kpatch-build/kpatch-build | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/kpatch-build/kpatch-build b/kpatch-build/kpatch-build
index c58abfa..91b9ac4 100755
--- a/kpatch-build/kpatch-build
+++ b/kpatch-build/kpatch-build
@@ -662,6 +662,7 @@ usage() {
 	echo "		-r, --sourcerpm         Specify kernel source RPM" >&2
 	echo "		-s, --sourcedir         Specify kernel source directory" >&2
 	echo "		-c, --config            Specify kernel config file" >&2
+	echo "		-D, --description       Specify module description file" >&2
 	echo "		-v, --vmlinux           Specify original vmlinux" >&2
 	echo "		-j, --jobs              Specify the number of make jobs" >&2
 	echo "		-t, --target            Specify custom kernel build targets" >&2
@@ -679,7 +680,7 @@ usage() {
 	echo "		                        (not recommended)" >&2
 }
 
-options="$(getopt -o ha:r:s:c:v:j:t:n:o:dR -l "help,archversion:,sourcerpm:,sourcedir:,config:,vmlinux:,jobs:,target:,name:,output:,oot-module:,oot-module-src:,debug,skip-gcc-check,skip-compiler-check,skip-cleanup,non-replace" -- "$@")" || die "getopt failed"
+options="$(getopt -o ha:r:s:c:D:v:j:t:n:o:dR -l "help,archversion:,sourcerpm:,sourcedir:,config:,description:,vmlinux:,jobs:,target:,name:,output:,oot-module:,oot-module-src:,debug,skip-gcc-check,skip-compiler-check,skip-cleanup,non-replace" -- "$@")" || die "getopt failed"
 
 eval set -- "$options"
 
@@ -708,6 +709,11 @@ while [[ $# -gt 0 ]]; do
 		CONFIGFILE="$(readlink -f "$2")"
 		shift
 		;;
+    -D|--description)
+		[[ ! -f "$2" ]] && die "description file '$2' not found"
+		DESC_FILE="$(readlink -f "$2")"
+		shift
+		;;
 	-v|--vmlinux)
 		[[ ! -f "$2" ]] && die "vmlinux file '$2' not found"
 		VMLINUX="$(readlink -f "$2")"
@@ -1429,6 +1435,11 @@ KPATCH_LDFLAGS="$KPATCH_LDFLAGS" \
 CROSS_COMPILE="$CROSS_COMPILE"
 save_env
 
+if [[ $DESC_FILE ]]; then
+	echo "MODULE_DESCRIPTION(\"$(cat $DESC_FILE)\");" >> $TEMPDIR/patch/livepatch-patch-hook.c || die
+fi
+
+
 make "${MAKEVARS[@]}" 2>&1 | logger || die
 
 if [[ "$USE_KLP" -eq 1 ]]; then
-- 
2.53.0

