From 3312975c08ac6f72c6fe74ed04faa8766c8de18f Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <shreenidhi.shedi@broadcom.com>
Date: Sat, 25 May 2024 09:21:42 +0530
Subject: [PATCH 1/2] Makefile: remove static ldflag

Signed-off-by: Shreenidhi Shedi <shreenidhi.shedi@broadcom.com>
---
 Makefile | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Makefile b/Makefile
index 5df1eb2..83649e9 100644
--- a/Makefile
+++ b/Makefile
@@ -117,7 +117,6 @@ clean:
 ver_test: ver
 	./ver
 
-${Bin}: override LDFLAGS += -static -s
 ${Bin}: LDLIBS = -L. -lautil
 ${BinObj}: %.o: %.c ${LibUtilHdr} ${LibUtil}
 
-- 
2.45.1

