diff -Naur cloud-init-18.4/cloudinit/config/cc_kubeadm.py cloud-init-18.4-new/cloudinit/config/cc_kubeadm.py
--- cloud-init-18.4/cloudinit/config/cc_kubeadm.py	1970-01-01 00:00:00.000000000 +0000
+++ cloud-init-18.4-new/cloudinit/config/cc_kubeadm.py	2019-09-19 06:15:31.696000000 +0000
@@ -0,0 +1,59 @@
+# Copyright (C) 2018 Heptio
+#
+# Author: Craig Tracey <craigtracey@gmail.com>
+# Author: Jason DeTiberus <detiber@gmail.com>
+#
+# This file is part of cloud-init. See LICENSE file for license information.
+
+"""
+Kubeadm
+-------
+**Summary:** bootstrap a Kubernetes node using kubeadm
+
+**Internal name:** ``cc_kubeadm``
+
+**Module frequency:** per instance
+
+**Supported distros:** all
+
+**Config keys**::
+
+    kubeadm:
+      operation: <init/join>
+      config: '/home/user/kubeadm.config'
+"""
+
+from cloudinit import util
+
+
+def _run_kubeadm(config, operation):
+    cmd = ["/usr/bin/kubeadm", operation, "--config", config]
+    util.subp(cmd, capture=False)
+
+
+def handle(name, cfg, cloud, log, _args):
+    """Handler method activated by cloud-init."""
+
+    if 'kubeadm' not in cfg:
+        log.debug("Skipping 'kubeadm' module, no config")
+        return
+
+    if 'operation' not in cfg['kubeadm']:
+        log.debug("Skipping 'kubeadm' module, no operation specified")
+        return
+
+    if cfg['kubeadm']['operation'] not in ('init', 'join'):
+        log.debug("Skipping 'kubeadm' module, unknown operation: %s",
+                  cfg['kubeadm']['operation'])
+        return
+
+    if 'config' not in cfg['kubeadm']:
+        log.debug("Skipping 'kubeadm' module, no config specified")
+        return
+
+    try:
+        _run_kubeadm(cfg['kubeadm']['config'], cfg['kubeadm']['operation'])
+    except Exception as e:
+        log.error("Failed to run kubeadm: %s" % e)
+
+# vi: ts=4 expandtab
