diff -uNr cloud-init-0.7.6-orig/config/cloud.cfg.d/05_logging.cfg cloud-init-0.7.6/config/cloud.cfg.d/05_logging.cfg
--- cloud-init-0.7.6-orig/config/cloud.cfg.d/05_logging.cfg	2015-06-17 21:27:10.378051186 -0700
+++ cloud-init-0.7.6/config/cloud.cfg.d/05_logging.cfg	2015-06-17 21:28:29.676352327 -0700
@@ -54,7 +54,8 @@
 
 log_cfgs:
 # These will be joined into a string that defines the configuration
- - [ *log_base, *log_syslog ]
+# Suppress log_syslog to allow logging to /var/log/cloud-init.log
+# - [ *log_base, *log_syslog ]
 # These will be joined into a string that defines the configuration
  - [ *log_base, *log_file ]
 # A file path can also be used
