diff --git a/cloudinit/config/tests/test_resolv_conf.py b/cloudinit/config/tests/test_resolv_conf.py
index 6546a0b5..5a8b877b 100644
--- a/cloudinit/config/tests/test_resolv_conf.py
+++ b/cloudinit/config/tests/test_resolv_conf.py
@@ -15,11 +15,12 @@ EXPECTED_HEADER = """\
 
 class TestGenerateResolvConf:
     @mock.patch("cloudinit.config.cc_resolv_conf.templater.render_to_file")
+    # Needed because of systemd-resolved-config.patch
     def test_default_target_fname_is_etc_resolvconf(self, m_render_to_file):
-        generate_resolv_conf("templates/resolv.conf.tmpl", mock.MagicMock())
+        generate_resolv_conf("templates/systemd.resolved.conf.tmpl", mock.MagicMock())
 
         assert [
-            mock.call(mock.ANY, "/etc/resolv.conf", mock.ANY)
+            mock.call(mock.ANY, "/etc/systemd/resolved.conf", mock.ANY)
         ] == m_render_to_file.call_args_list
 
     @mock.patch("cloudinit.config.cc_resolv_conf.templater.render_to_file")
