diff --git a/cloudinit/distros/__init__.py b/cloudinit/distros/__init__.py
index 2537608f..7fe04d05 100755
--- a/cloudinit/distros/__init__.py
+++ b/cloudinit/distros/__init__.py
@@ -546,6 +546,22 @@ class Distro(metaclass=abc.ABCMeta):
         if 'snapuser' in kwargs:
             return self.add_snap_user(name, **kwargs)
 
+        """
+        Photon provides a way to set password for exisiting user using
+        'passwd' config in userdata.
+        cloud-init supports 'passwd' option only for non-existing user
+        To maintain backward compatibility, this piece of code is added
+
+        Check 'GenerateCloudInitConfig' function in 'CustomizationUtils.sh'
+        in gosc-scripts for the use case
+        """
+        user_exists = True if util.is_user(name) else False
+        # Set hashed_passwd to passwd if hashed_passwd is not present
+        if (user_exists and 'passwd' in kwargs and kwargs['passwd'] and
+                'hashed_passwd' not in kwargs):
+            kwargs['hashed_passwd'] = kwargs['passwd']
+            kwargs.pop('passwd')
+
         # Add the user
         self.add_user(name, **kwargs)
 
