diff -rup cloud-init-0.7.9/cloudinit/config/cc_resizefs.py cloud-init-0.7.9-new/cloudinit/config/cc_resizefs.py
--- cloud-init-0.7.9/cloudinit/config/cc_resizefs.py	2016-12-23 08:37:45.000000000 -0800
+++ cloud-init-0.7.9-new/cloudinit/config/cc_resizefs.py	2017-04-28 15:39:54.499540213 -0700
@@ -85,7 +85,9 @@ def rootdev_from_cmdline(cmdline):
     if found.startswith("LABEL="):
         return "/dev/disk/by-label/" + found[len("LABEL="):]
     if found.startswith("UUID="):
-        return "/dev/disk/by-uuid/" + found[len("UUID="):]
+        return "/dev/disk/by-uuid/" + found[len("UUID="):].lower()
+    if found.startswith("PARTUUID="):
+        return "/dev/disk/by-partuuid/" + found[len("PARTUUID="):].lower()
 
     return "/dev/" + found
 
