diff -rup cloud-init-0.7.9/cloudinit/config/cc_resizefs.py cloud-init-0.7.9-new/cloudinit/config/cc_resizefs.py
--- cloud-init-0.7.9/cloudinit/config/cc_resizefs.py	2016-12-23 08:37:45.000000000 -0800
+++ cloud-init-0.7.9-new/cloudinit/config/cc_resizefs.py	2017-04-13 17:33:14.508595806 -0700
@@ -86,6 +86,8 @@ def rootdev_from_cmdline(cmdline):
         return "/dev/disk/by-label/" + found[len("LABEL="):]
     if found.startswith("UUID="):
         return "/dev/disk/by-uuid/" + found[len("UUID="):]
+    if found.startswith("PARTUUID="):
+        return "/dev/disk/by-partuuid/" + found[len("PARTUUID="):]
 
     return "/dev/" + found
 
