=== modified file './systemd/cloud-init.service'
--- ./systemd/cloud-init.service	2015-01-16 16:36:30 +0000
+++ ./systemd/cloud-init.service	2015-04-01 18:19:12 +0000
@@ -1,9 +1,8 @@
 [Unit]
 Description=Initial cloud-init job (metadata service crawler)
-After=local-fs.target network.target cloud-init-local.service
+After=local-fs.target network-online.target cloud-init-local.service
 Before=sshd.service sshd-keygen.service
-Requires=network.target
-Wants=local-fs.target cloud-init-local.service sshd.service sshd-keygen.service
+Wants=local-fs.target network-online.target cloud-init-local.service sshd.service sshd-keygen.service
 
 [Service]
 Type=oneshot

