diff -ru cloud-init-0.7.6/cloudinit/distros/__init__.py cloud-init-0.7-modified/cloudinit/distros/__init__.py
--- cloud-init-0.7.6/cloudinit/distros/__init__.py	2014-10-10 15:26:25.000000000 +0000
+++ cloud-init-0.7-modified/cloudinit/distros/__init__.py	2015-08-11 14:22:09.814582603 +0000
@@ -376,6 +376,9 @@
         if 'plain_text_passwd' in kwargs and kwargs['plain_text_passwd']:
             self.set_passwd(name, kwargs['plain_text_passwd'])
 
+        if 'passwd' in kwargs and kwargs['passwd']:
+            self.set_passwd(name,kwargs['passwd'],True)
+
         # Default locking down the account.  'lock_passwd' defaults to True.
         # lock account unless lock_password is False.
         if kwargs.get('lock_passwd', True):
