#Forcing ncurses module to compile with stack-protector flag
--- a/setup.py	2016-06-20 18:19:56.332815511 -0700
+++ b/setup.py	2016-06-20 18:20:23.721857009 -0700
@@ -1354,7 +1354,8 @@
                                     [os.path.join(d, 'ncursesw') for d in inc_dirs])
             exts.append( Extension('_curses', ['_cursesmodule.c'],
                                    include_dirs = curses_incs,
-                                   libraries = curses_libs) )
+                                   libraries = curses_libs,
+                                   extra_compile_args=['-fstack-protector']) )
         elif curses_library == 'curses' and host_platform != 'darwin':
                 # OSX has an old Berkeley curses, not good enough for
                 # the _curses module.
