diff --git a/python/mozbuild/mozbuild/configure/__init__.py b/python/mozbuild/mozbuild/configure/__init__.py
index f3167f6c71..21ad8567f9 100644
--- a/python/mozbuild/mozbuild/configure/__init__.py
+++ b/python/mozbuild/mozbuild/configure/__init__.py
@@ -555,8 +555,6 @@ class ConfigureSandbox(dict):
             msg = "Unknown option: %s" % without_value
             if self._help:
                 self._logger.warning(msg)
-            else:
-                raise InvalidOptionError(msg)
 
         # Run the execution queue
         for func, args in self._execution_queue:
