From 1f933cc0c42f9483e68c6c1a63b9dbf722e34fa6 Mon Sep 17 00:00:00 2001
From: Clark Williams <clrkwllms@kernel.org>
Date: Fri, 18 Aug 2023 10:45:35 -0500
Subject: [PATCH 52/69] 'Linux 6.1.46-rt13 REBASE'

Signed-off-by: Clark Williams <clrkwllms@kernel.org>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 045478966e9f..9f7d0bdbffb1 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt7
+-rt13
-- 
2.49.0

