diff --git a/cmake/modules/cpp-httplib.cmake b/cmake/modules/cpp-httplib.cmake
index 28c9de8..48700f2 100644
--- a/cmake/modules/cpp-httplib.cmake
+++ b/cmake/modules/cpp-httplib.cmake
@@ -19,7 +19,7 @@ if(USE_BUNDLED_CPPHTTPLIB)
 	include(FetchContent)
 	FetchContent_Declare(
 		cpp-httplib
-		URL https://github.com/yhirose/cpp-httplib/archive/refs/tags/v0.15.3.tar.gz
+		URL ${CMAKE_CURRENT_SOURCE_DIR}/../../SOURCES/cpp-httplib-0.15.3.tar.gz
 		URL_HASH SHA256=2121bbf38871bb2aafb5f7f2b9b94705366170909f434428352187cb0216124e
 	)
 	FetchContent_MakeAvailable(cpp-httplib)
