diff -Naur a/sysdig-0.10.1/driver/main.c b/sysdig-0.12.1/driver/main.c
--- a/sysdig-0.10.1/driver/main.c	2016-11-10 14:37:56.000000000 -0800
+++ b/sysdig-0.10.1/driver/main.c	2017-01-11 02:12:39.990492919 -0800
@@ -1730,7 +1730,7 @@
 }
 
 int __access_remote_vm(struct task_struct *t, struct mm_struct *mm, unsigned long addr,
-		       void *buf, int len, int write);
+		       void *buf, int len, unsigned int write);
 
 TRACEPOINT_PROBE(syscall_procexit_probe, struct task_struct *p)
 {
