diff -ru cmake-3.12.1/Source/kwsys/CMakeLists.txt cmake-3.12.1-new/Source/kwsys/CMakeLists.txt
--- cmake-3.12.1/Source/kwsys/CMakeLists.txt	2018-08-09 17:44:08.000000000 +0530
+++ cmake-3.12.1-new/Source/kwsys/CMakeLists.txt	2018-09-21 09:11:46.011100539 +0530
@@ -1074,7 +1074,6 @@
     ENDIF()
     SET(KWSYS_CXX_TESTS ${KWSYS_CXX_TESTS}
       testConfigure.cxx
-      testSystemTools.cxx
       testCommandLineArguments.cxx
       testCommandLineArguments1.cxx
       testDirectory.cxx
diff -ru cmake-3.12.1/Tests/CMakeLists.txt cmake-3.12.1-new/Tests/CMakeLists.txt
--- cmake-3.12.1/Tests/CMakeLists.txt	2018-08-09 17:44:08.000000000 +0530
+++ cmake-3.12.1-new/Tests/CMakeLists.txt	2018-09-21 09:14:05.907096799 +0530
@@ -2636,21 +2636,7 @@
     )
   set_tests_properties(CTestTestUpload PROPERTIES
     PASS_REGULAR_EXPRESSION "Upload\\.xml")
-
-  configure_file(
-    "${CMake_SOURCE_DIR}/Tests/CTestCoverageCollectGCOV/test.cmake.in"
-    "${CMake_BINARY_DIR}/Tests/CTestCoverageCollectGCOV/test.cmake"
-    @ONLY ESCAPE_QUOTES)
-  add_test(CTestCoverageCollectGCOV ${CMAKE_CTEST_COMMAND}
-    -C \${CTEST_CONFIGURATION_TYPE}
-    -S "${CMake_BINARY_DIR}/Tests/CTestCoverageCollectGCOV/test.cmake" -VV
-    --output-log "${CMake_BINARY_DIR}/Tests/CTestCoverageCollectGCOV/testOut.log"
-    )
-  set_tests_properties(CTestCoverageCollectGCOV PROPERTIES
-    PASS_REGULAR_EXPRESSION
-    "PASSED with correct output.*Testing/CoverageInfo/main.cpp.gcov")
-  set_property(TEST CTestCoverageCollectGCOV PROPERTY ENVIRONMENT CTEST_PARALLEL_LEVEL=)
-
+  
   configure_file(
     "${CMake_SOURCE_DIR}/Tests/CTestTestEmptyBinaryDirectory/test.cmake.in"
     "${CMake_BINARY_DIR}/Tests/CTestTestEmptyBinaryDirectory/test.cmake"
@@ -3014,18 +3000,6 @@
     PASS_REGULAR_EXPRESSION "The stop time has been passed")
 
   configure_file(
-    "${CMake_SOURCE_DIR}/Tests/CTestTestSubdir/test.cmake.in"
-    "${CMake_BINARY_DIR}/Tests/CTestTestSubdir/test.cmake"
-    @ONLY ESCAPE_QUOTES)
-  add_test(CTestTestSubdir ${CMAKE_CTEST_COMMAND}
-    -S "${CMake_BINARY_DIR}/Tests/CTestTestSubdir/test.cmake" -V
-    --output-log "${CMake_BINARY_DIR}/Tests/CTestTestSubdir/testOutput.log"
-    )
-  #make sure all 3 subdirs were added
-  set_tests_properties(CTestTestSubdir PROPERTIES
-    PASS_REGULAR_EXPRESSION "0 tests failed out of 3")
-
-  configure_file(
     "${CMake_SOURCE_DIR}/Tests/CTestTestTimeout/test.cmake.in"
     "${CMake_BINARY_DIR}/Tests/CTestTestTimeout/test.cmake"
     @ONLY ESCAPE_QUOTES)
Only in cmake-3.12.1-new/Tests: CMakeLists.txt.orig
