diff -ru cmake-3.8.0/Source/kwsys/CMakeLists.txt cmake-3.8.0-modified/Source/kwsys/CMakeLists.txt
--- cmake-3.8.0/Source/kwsys/CMakeLists.txt	2017-04-10 08:23:07.000000000 -0700
+++ cmake-3.8.0-modified/Source/kwsys/CMakeLists.txt	2017-08-09 15:48:48.735319709 -0700
@@ -924,7 +924,6 @@
     ENDIF()
     SET(KWSYS_CXX_TESTS ${KWSYS_CXX_TESTS}
       testIOS
-      testSystemTools
       testCommandLineArguments
       testCommandLineArguments1
       )
diff -ru cmake-3.8.0/Tests/CMakeLists.txt cmake-3.8.0-modified/Tests/CMakeLists.txt
--- cmake-3.8.0/Tests/CMakeLists.txt	2017-04-10 08:23:07.000000000 -0700
+++ cmake-3.8.0-modified/Tests/CMakeLists.txt	2017-08-14 14:00:30.832201561 -0700
@@ -2537,21 +2537,7 @@
     )
   set_tests_properties(CTestTestUpload PROPERTIES
     PASS_REGULAR_EXPRESSION "Upload\\.xml")
-
-  configure_file(
-    "${CMake_SOURCE_DIR}/Tests/CTestCoverageCollectGCOV/test.cmake.in"
-    "${CMake_BINARY_DIR}/Tests/CTestCoverageCollectGCOV/test.cmake"
-    @ONLY ESCAPE_QUOTES)
-  add_test(CTestCoverageCollectGCOV ${CMAKE_CTEST_COMMAND}
-    -C \${CTEST_CONFIGURATION_TYPE}
-    -S "${CMake_BINARY_DIR}/Tests/CTestCoverageCollectGCOV/test.cmake" -VV
-    --output-log "${CMake_BINARY_DIR}/Tests/CTestCoverageCollectGCOV/testOut.log"
-    )
-  set_tests_properties(CTestCoverageCollectGCOV PROPERTIES
-    PASS_REGULAR_EXPRESSION
-    "PASSED with correct output.*Testing/CoverageInfo/main.cpp.gcov")
-  set_property(TEST CTestCoverageCollectGCOV PROPERTY ENVIRONMENT CTEST_PARALLEL_LEVEL=)
-
+  
   configure_file(
     "${CMake_SOURCE_DIR}/Tests/CTestTestEmptyBinaryDirectory/test.cmake.in"
     "${CMake_BINARY_DIR}/Tests/CTestTestEmptyBinaryDirectory/test.cmake"
@@ -2916,18 +2902,6 @@
     PASS_REGULAR_EXPRESSION "The stop time has been passed")
 
   configure_file(
-    "${CMake_SOURCE_DIR}/Tests/CTestTestSubdir/test.cmake.in"
-    "${CMake_BINARY_DIR}/Tests/CTestTestSubdir/test.cmake"
-    @ONLY ESCAPE_QUOTES)
-  add_test(CTestTestSubdir ${CMAKE_CTEST_COMMAND}
-    -S "${CMake_BINARY_DIR}/Tests/CTestTestSubdir/test.cmake" -V
-    --output-log "${CMake_BINARY_DIR}/Tests/CTestTestSubdir/testOutput.log"
-    )
-  #make sure all 3 subdirs were added
-  set_tests_properties(CTestTestSubdir PROPERTIES
-    PASS_REGULAR_EXPRESSION "0 tests failed out of 3")
-
-  configure_file(
     "${CMake_SOURCE_DIR}/Tests/CTestTestTimeout/test.cmake.in"
     "${CMake_BINARY_DIR}/Tests/CTestTestTimeout/test.cmake"
     @ONLY ESCAPE_QUOTES)
