From 88b34bcff95d2dd3c71fa9c5004fc72166a40062 Mon Sep 17 00:00:00 2001
From: Clark Williams <clrkwllms@kernel.org>
Date: Mon, 12 Jun 2023 10:41:56 -0500
Subject: [PATCH 52/52] 'Linux 6.1.33-rt11 REBASE'

Signed-off-by: Clark Williams <clrkwllms@kernel.org>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 045478966e9f..05c35cb58077 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt7
+-rt11
-- 
2.40.1

