From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Leo Sandoval <lsandova@redhat.com>
Date: Mon, 23 Feb 2026 17:28:10 -0600
Subject: [PATCH] commands/efi/tpm.c: check if PCR is enable before TPM measure

Signed-off-by: Leo Sandoval <lsandova@redhat.com>
---
 grub-core/commands/efi/tpm.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/grub-core/commands/efi/tpm.c b/grub-core/commands/efi/tpm.c
index f250c30dbc..cfddeb8719 100644
--- a/grub-core/commands/efi/tpm.c
+++ b/grub-core/commands/efi/tpm.c
@@ -273,6 +273,11 @@ grub_tpm_measure (unsigned char *buf, grub_size_t size, grub_uint8_t pcr,
   grub_efi_handle_t tpm_handle;
   grub_efi_uint8_t protocol_version;
 
+  grub_dprintf ("tpm", "PCR %d %s\n", pcr, grub_tpm_pcr_is_enabled (pcr) ? "enabled" : "disabled");
+
+  if (!grub_tpm_pcr_is_enabled (pcr))
+    return GRUB_ERR_NONE;
+
   grub_cc_log_event(buf, size, pcr, description);
 
   if (!grub_tpm_handle_find (&tpm_handle, &protocol_version))
