--- qemu-7.2.0.orig/meson.build	2022-12-14 08:28:45.000000000 -0800
+++ qemu-7.2.0.mod/meson.build	2025-08-12 03:57:37.000000000 -0700
@@ -3022,7 +3022,6 @@
 subdir('authz')
 subdir('crypto')
 subdir('ui')
-subdir('hw')
 subdir('gdbstub')
 
 
@@ -3622,18 +3621,11 @@
                               libcap_ng, mpathpersist],
                install: true)
   endif
-
-  if have_ivshmem
-    subdir('contrib/ivshmem-client')
-    subdir('contrib/ivshmem-server')
-  endif
 endif
 
 subdir('scripts')
 subdir('tools')
-subdir('pc-bios')
 subdir('docs')
-subdir('tests')
 if gtk.found()
   subdir('po')
 endif
