From 04a979b02d3d868b547a4f2ff316006db7c58c09 Mon Sep 17 00:00:00 2001
From: Steve Grubb <sgrubb@redhat.com>
Date: Feb 24 2022 13:24:30 +0000
Subject: Undo fix to libaudit.h before installing

diff -urp usr.orig/include/libaudit.h.orig usr/include/libaudit.h
--- usr.orig/include/libaudit.h.orig	2022-02-14 14:14:08.000000000 -0500
+++ usr/include/libaudit.h	2022-02-23 13:35:52.638340789 -0500
@@ -39,7 +39,7 @@ extern "C" {
 #include <stdint.h>
 #include <sys/socket.h>
 #include <linux/netlink.h>
-#include "audit.h"
+#include <linux/audit.h>
 #include <stdarg.h>
 #include <syslog.h>
