diff -Naur mesos-1.1.0.orig/src/linux/fs.cpp mesos-1.1.0/src/linux/fs.cpp
--- mesos-1.1.0.orig/src/linux/fs.cpp	2016-11-04 05:46:21.000000000 -0700
+++ mesos-1.1.0/src/linux/fs.cpp	2017-03-24 23:17:15.332465310 -0700
@@ -17,6 +17,7 @@
 #include <errno.h>
 #include <stdio.h>
 #include <string.h>
+#include <sys/sysmacros.h>
 
 #include <linux/limits.h>
 #include <linux/unistd.h>
diff -Naur mesos-1.1.0.orig/src/slave/containerizer/mesos/isolators/gpu/isolator.cpp mesos-1.1.0/src/slave/containerizer/mesos/isolators/gpu/isolator.cpp
--- mesos-1.1.0.orig/src/slave/containerizer/mesos/isolators/gpu/isolator.cpp	2016-11-04 05:46:21.000000000 -0700
+++ mesos-1.1.0/src/slave/containerizer/mesos/isolators/gpu/isolator.cpp	2017-03-25 00:12:23.657869169 -0700
@@ -15,6 +15,7 @@
 // limitations under the License.
 
 #include <stdint.h>
+#include <sys/sysmacros.h>
 
 #include <algorithm>
 #include <list>
