From 2746ad5df8c4925896ce56065afc7cfb79e87911 Mon Sep 17 00:00:00 2001
From: Lukas Nykryn <lnykryn@redhat.com>
Date: Tue, 22 Sep 2015 15:17:05 +0200
Subject: systemd-sysv-install: don't play ping-pong with systemctl

---
 chkconfig.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/chkconfig.c b/chkconfig.c
index 6399ef1..6b54c98 100644
--- a/chkconfig.c
+++ b/chkconfig.c
@@ -789,6 +789,8 @@ int main(int argc, const char ** argv) {
                 if (!name || !state)
                         usage(progname);
 
+                noRedirectItem = 1;
+
                 /* systemd-sysv-install has target and verb in reverse order */
                 c = name;
                 name = state;
