--- a/chkconfig.c	2017-04-07 10:52:18.641601093 -0700
+++ b/chkconfig.c	2017-04-07 10:58:18.657338958 -0700
@@ -839,17 +839,29 @@
 		}
 	    }
 	    rc = readServiceInfo(name, type, &s, 0);
-	    if (rc)
-	       return 1;
+	    if (rc) {
+               fprintf(stderr, "%s: unknown service\n", name);
+ 	       return 1;
+            }
 	    if (s.type == TYPE_XINETD) {
 	       if (isXinetdEnabled())
 		       return !s.levels;
 	       else
 		       return 1;
 	    } else {
-               if (level == -1)
+               if (level == -1) {
                    level = currentRunlevel();
-	       return s.currentLevels & (1 << level) ? 0 : 1;
+                   if (level == -1) {
+                       fprintf(stderr, "can not detect the current runlevel\n");
+                       return 1;
+                   }
+               }
+	       if (s.currentLevels & (1 << level)) {
+                   printf("%s on\n", name);
+               } else {
+                   printf("%s off\n", name);
+               }
+               return 0;
 	    }
 	} else if (!strcmp(state, "on")) {
 	    if (!noRedirectItem) {
