From d0a9ef8fee9ce0f44383d8315c9619dc5a9edfdc Mon Sep 17 00:00:00 2001
From: Clark Williams <clrkwllms@kernel.org>
Date: Tue, 22 Jul 2025 13:12:04 -0500
Subject: [PATCH 67/67] Linux 6.1.146-rt53 REBASE

Signed-off-by: Clark Williams <clrkwllms@kernel.org>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 75493460c41f..e6c0dc6a54cd 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt51
+-rt53
-- 
2.50.1

