diff -rup compute-image-packages-20170426/google_compute_engine/instance_setup/instance_config.py compute-image-packages-20170426-new/google_compute_engine/instance_setup/instance_config.py
--- compute-image-packages-20170426/google_compute_engine/instance_setup/instance_config.py	2017-04-26 11:17:07.000000000 -0700
+++ compute-image-packages-20170426-new/google_compute_engine/instance_setup/instance_config.py	2017-08-23 15:13:20.399352478 -0700
@@ -55,7 +55,7 @@ class InstanceConfig(config_manager.Conf
       'InstanceSetup': {
           'optimize_local_ssd': 'true',
           'network_enabled': 'true',
-          'set_boto_config': 'true',
+          'set_boto_config': 'false',
           'set_host_keys': 'true',
           'set_multiqueue': 'true',
       },
diff -rup compute-image-packages-20170426/setup.py compute-image-packages-20170426-new/setup.py
--- compute-image-packages-20170426/setup.py	2017-04-26 11:17:07.000000000 -0700
+++ compute-image-packages-20170426-new/setup.py	2017-08-23 15:14:25.030957426 -0700
@@ -25,7 +25,7 @@ setuptools.setup(
     author_email='gc-team@google.com',
     description='Google Compute Engine',
     include_package_data=True,
-    install_requires=['boto', 'setuptools'],
+    install_requires=['setuptools'],
     license='Apache Software License',
     long_description='Google Compute Engine guest environment.',
     name='google-compute-engine',
