--- a/js2py/internals/speed.py_org	2019-11-06 04:16:49.172807117 +0000
+++ b/js2py/internals/speed.py	2019-11-06 04:17:34.914220803 +0000
@@ -47,7 +47,7 @@ t = []
 
 Type = None
 try:
-    print timeit(
+    print(timeit(
         """
 
 t.append(4)
@@ -57,6 +57,6 @@ t.pop()
 
 """,
         "from __main__ import X,Y,namedtuple,array,t,add,Type, izip",
-        number=1000000)
+        number=1000000))
 except:
     raise
--- a/js2py/test_internals.py_org	2019-11-06 04:15:41.570716185 +0000
+++ b/js2py/test_internals.py	2019-11-06 04:16:00.615305437 +0000
@@ -6,4 +6,4 @@ x = r'''
 function g() {var h123 = 11; return [function g1() {return h123}, new Function('return h123')]}
 g()[1]()
 '''
-print seval.eval_js_vm(x)
+print(seval.eval_js_vm(x))
