From 6e1bae0cbcbd6aac49a81f882f3b5e9b08e6b9fa Mon Sep 17 00:00:00 2001
From: "Luis Claudio R. Goncalves" <lgoncalv@redhat.com>
Date: Fri, 11 Feb 2022 12:14:30 -0300
Subject: [PATCH 318/318] Linux 5.10.100-rt62 REBASE

Signed-off-by: Luis Claudio R. Goncalves <lgoncalv@redhat.com>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 21988f9ad53f1..40d81d8e61b65 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt34
+-rt62
-- 
2.34.1

