From ef68bf10fd3ab306ce6e9c6fff44e860f4d4c867 Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <sshedi@vmware.com>
Date: Mon, 29 May 2023 11:33:44 +0530
Subject: [PATCH] sslverify.py: use fips compatible sha512 instead of md5

Signed-off-by: Shreenidhi Shedi <sshedi@vmware.com>
---
 src/twisted/internet/_sslverify.py | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/twisted/internet/_sslverify.py b/src/twisted/internet/_sslverify.py
index 6824482..48bb569 100644
--- a/src/twisted/internet/_sslverify.py
+++ b/src/twisted/internet/_sslverify.py
@@ -7,7 +7,7 @@
 import warnings
 from binascii import hexlify
 from functools import lru_cache
-from hashlib import md5
+from hashlib import sha512
 
 from zope.interface import Interface, implementer
 
@@ -519,12 +519,12 @@ class Certificate(CertBase):
         """
         return self.original.get_serial_number()
 
-    def digest(self, method="md5"):
+    def digest(self, method="sha512"):
         """
         Return a digest hash of this certificate using the specified hash
         algorithm.
 
-        @param method: One of C{'md5'} or C{'sha'}.
+        @param method: C{'sha'}.
 
         @return: The digest of the object, formatted as b":"-delimited hex
             pairs
@@ -750,7 +750,7 @@ class PublicKey:
         @rtype: native L{str}
         """
         raw = crypto.dump_publickey(crypto.FILETYPE_ASN1, self.original)
-        h = md5()
+        h = sha512()
         h.update(raw)
         return h.hexdigest()
 
-- 
2.40.1
