From 002166a4207c1c35d97fdbd43cb84105b14a82f4 Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <sshedi@vmware.com>
Date: Wed, 24 May 2023 11:58:47 +0530
Subject: [PATCH] sslverify.py: use fips compatible sha512 instead of md5

Signed-off-by: Shreenidhi Shedi <sshedi@vmware.com>
---
 src/twisted/internet/_sslverify.py | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/twisted/internet/_sslverify.py b/src/twisted/internet/_sslverify.py
index c44ec14..4148c9a 100644
--- a/src/twisted/internet/_sslverify.py
+++ b/src/twisted/internet/_sslverify.py
@@ -8,7 +8,7 @@ from __future__ import division, absolute_import
 import warnings
 
 from constantly import Names, NamedConstant
-from hashlib import md5
+from hashlib import sha512
 
 from OpenSSL import SSL, crypto
 from OpenSSL._util import lib as pyOpenSSLlib
@@ -552,12 +552,12 @@ class Certificate(CertBase):
         return self.original.get_serial_number()
 
 
-    def digest(self, method='md5'):
+    def digest(self, method='sha512'):
         """
         Return a digest hash of this certificate using the specified hash
         algorithm.
 
-        @param method: One of C{'md5'} or C{'sha'}.
+        @param method: C{'sha'}.
 
         @return: The digest of the object, formatted as b":"-delimited hex
             pairs
@@ -789,7 +789,7 @@ class PublicKey:
         @rtype: native L{str}
         """
         raw = crypto.dump_publickey(crypto.FILETYPE_ASN1, self.original)
-        h = md5()
+        h = sha512()
         h.update(raw)
         return h.hexdigest()
 
-- 
2.40.1

