diff -dupr a/src/twisted/python/test/test_setup.py b/src/twisted/python/test/test_setup.py
--- a/src/twisted/python/test/test_setup.py	2017-06-04 07:32:04.000000000 +0000
+++ b/src/twisted/python/test/test_setup.py	2017-08-29 21:44:25.582904955 +0000
@@ -78,7 +78,7 @@ class OptionalDependenciesTests(TestCase
         Setuptools' Distribution object parses and stores its C{extras_require}
         argument as an attribute.
         """
-        extras = dict(im_an_extra_dependency="thing")
+        extras = dict(im_an_extra_dependency=["thing"])
         attrs = dict(extras_require=extras)
         distribution = Distribution(attrs)
         self.assertEqual(
