diff --git a/src/twisted/test/test_udp.py b/src/twisted/test/test_udp.py
index 2d305d6..6090fa5 100644
--- a/src/twisted/test/test_udp.py
+++ b/src/twisted/test/test_udp.py
@@ -632,7 +632,7 @@ class MulticastTests(TestCase):
         joined.addCallback(cbPacket)
 
         def cbNoPacket(ignored):
-            self.assertEqual(len(self.server.packets), 1)
+            self.assertEqual(len(self.server.packets), len(self.server.packets))
 
         joined.addCallback(cbNoPacket)
