--- a/server/pkgmgmtrestapi.c	2020-11-03 17:46:59.971765087 +0530
+++ b/server/pkgmgmtrestapi.c	2020-11-03 17:49:54.621185855 +0530
@@ -748,9 +748,12 @@ get_repodata_json_string(
         json_object_set_new(pRepoObj, "name", json_string(pRepoData->pszName));
         json_object_set_new(pRepoObj, "baseurl", json_string(pRepoData->pszBaseUrl));
         json_object_set_new(pRepoObj, "gpgkey", pszUrlGPGKeyArray);
-        for(i = 0; pRepoData->ppszUrlGPGKeys[i]; i++)
+        if (pRepoData->ppszUrlGPGKeys != NULL)
         {
-            json_array_append_new(pszUrlGPGKeyArray, json_string(pRepoData->ppszUrlGPGKeys[i]));
+            for(i = 0; pRepoData->ppszUrlGPGKeys[i]; i++)
+            {
+                json_array_append_new(pszUrlGPGKeyArray, json_string(pRepoData->ppszUrlGPGKeys[i]));
+            }
         }
         json_object_set_new(pRepoObj, "metadata_expire", json_integer(0));
         json_object_set_new(pRepoObj, "skip_if_unavailable", json_boolean(pRepoData->nSkipIfUnavailable));
