diff --git a/vsf_findlibs.sh b/vsf_findlibs.sh
index 4538685..6e65e2e 100755
--- a/vsf_findlibs.sh
+++ b/vsf_findlibs.sh
@@ -66,7 +66,7 @@ locate_library /usr/shlib/librt.so && echo "-lrt";
 locate_library /usr/lib/libsendfile.so && echo "-lsendfile";
 
 # OpenSSL
-if find_func SSL_library_init ssl.o; then
+if find_func SSL_CTX_new ssl.o; then
   echo "-lssl -lcrypto";
 elif find_func SSL_new ssl.o; then
   echo "-lssl -lcrypto";
