From 21a5aa20b1509547fa5aed43b8f8dc554a0fee49 Mon Sep 17 00:00:00 2001
From: Ashwin Dayanand Kamat <kashwindayan@vmware.com>
Date: Tue, 14 Dec 2021 14:16:11 +0000
Subject: [PATCH] numactl: fix physcpubind for single cpu

Signed-off-by: Ashwin Dayanand Kamat <kashwindayan@vmware.com>
---
 test/bind_range | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/test/bind_range b/test/bind_range
index 6d67afe..ab2c18d 100755
--- a/test/bind_range
+++ b/test/bind_range
@@ -42,9 +42,11 @@ function check_physcpubind
 {
 	reset_mask
 	set_cpu_affinity 0
-	numactl --physcpubind=$HIGHESTCPU ls > /dev/null 2>&1
-	if [ $? == 0 ] ; then # shouldn't pass so easy
-		return 1;
+	if [ $HIGHESTCPU != 0 ] ; then # system has single cpu
+		numactl --physcpubind=$HIGHESTCPU ls > /dev/null 2>&1
+		if [ $? == 0 ] ; then # shouldn't pass so easy
+			return 1;
+		fi
 	fi
 	numactl --all --physcpubind=$HIGHESTCPU ls > /dev/null 2>&1
 	if [ $? == 1 ] ; then # shouldn't fail
-- 
2.30.0

