From b4cfb50f42950b0f2f382865f23e9ff02ba05aa4 Mon Sep 17 00:00:00 2001
From: Philip Withnall <pwithnall@endlessos.org>
Date: Tue, 25 Oct 2022 18:03:56 +0100
Subject: [PATCH 45/49] gvariant: Fix a leak of a GVariantTypeInfo on an error
 handling path

Signed-off-by: Philip Withnall <pwithnall@endlessos.org>
---
 glib/gvariant-core.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/glib/gvariant-core.c b/glib/gvariant-core.c
index dbcf60f69..2d2508244 100644
--- a/glib/gvariant-core.c
+++ b/glib/gvariant-core.c
@@ -1183,6 +1183,7 @@ g_variant_get_child_value (GVariant *value,
         G_VARIANT_MAX_RECURSION_DEPTH - value->depth)
       {
         g_assert (g_variant_is_of_type (value, G_VARIANT_TYPE_VARIANT));
+        g_variant_type_info_unref (s_child.type_info);
         return g_variant_new_tuple (NULL, 0);
       }
 
-- 
2.35.6

