diff --git a/cli/netmgrcli.c b/cli/netmgrcli.c
index 8c7396f..3b7f535 100644
--- a/cli/netmgrcli.c
+++ b/cli/netmgrcli.c
@@ -1695,10 +1695,10 @@ NETMGRCLI_CMD_MAP cmdMap[] =
      "--set --duid <DUID string> --interface <interface name>",
      "get or set DHCP DUID, optionally per interface"
     },
-    {"if_iaid ",
+    {"if_iaid",
      cli_if_iaid,
      "--set --iaid <IAID value> --interface <interface name>",
-     "get or set interface IAID"
+     "\t get or set interface IAID"
     },
     {"ntp_servers",
      cli_ntp_servers,
