diff -ru psmisc-22.21/src/fuser.c psmisc-22.21-modified/src/fuser.c
--- psmisc-22.21/src/fuser.c	2014-01-31 05:07:32.000000000 -0800
+++ psmisc-22.21-modified/src/fuser.c	2016-03-11 17:43:56.981300845 -0800
@@ -1174,7 +1174,7 @@
 		usage(_("No process specification given"));
 
 	/* Check if -M flag was used and if so check mounts */
-	if (opts * OPT_ISMOUNTPOINT) {
+	if (opts & OPT_ISMOUNTPOINT) {
 	    check_mountpoints(&mounts, &names_head, &names_tail);
 	}
 
