diff -rup multipath-tools-0.5.0/libmultipath/Makefile multipath-tools-0.5.0-new/libmultipath/Makefile
--- multipath-tools-0.5.0/libmultipath/Makefile	2013-12-17 13:40:41.000000000 -0800
+++ multipath-tools-0.5.0-new/libmultipath/Makefile	2016-11-22 10:38:00.195458156 -0800
@@ -9,7 +9,11 @@ DEVLIB = libmultipath.so
 LIBS = $(DEVLIB).$(SONAME)
 LIBDEPS = -lpthread -ldl -ldevmapper -ludev
 ifdef SYSTEMD
-	LIBDEPS += -lsystemd-daemon
+       ifeq ($(shell test $(SYSTEMD) -gt 209 && echo 1), 1)
+               LIBDEPS += -lsystemd
+       else
+               LIBDEPS += -lsystemd-daemon
+       endif
 endif
 
 OBJS = memory.o parser.o vector.o devmapper.o callout.o \
diff -rup multipath-tools-0.5.0/multipathd/Makefile multipath-tools-0.5.0-new/multipathd/Makefile
--- multipath-tools-0.5.0/multipathd/Makefile	2013-12-17 13:40:41.000000000 -0800
+++ multipath-tools-0.5.0-new/multipathd/Makefile	2016-11-22 10:37:40.535458600 -0800
@@ -11,7 +11,11 @@ ifdef SYSTEMD
 endif
 LDFLAGS += -lpthread -ldevmapper -lreadline
 ifdef SYSTEMD
-	LDFLAGS += -lsystemd-daemon
+       ifeq ($(shell test $(SYSTEMD) -gt 209 && echo 1), 1)
+               LDFLAGS += -lsystemd
+       else
+               LDFLAGS += -lsystemd-daemon
+       endif
 endif
 LDFLAGS += -ludev -ldl \
 	-L$(multipathdir) -lmultipath -L$(mpathpersistdir) -lmpathpersist
