From 6c4b8a0e31552f9628e6662b2abff3a64a7b735c Mon Sep 17 00:00:00 2001
From: Daniel Wagner <wagi@monom.org>
Date: Fri, 7 Jun 2024 15:58:21 +0200
Subject: [PATCH 351/351] Linux 4.19.315-rt135 REBASE

Signed-off-by: Daniel Wagner <wagi@monom.org>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 1199ebade17b..e3026053f01e 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt16
+-rt135
-- 
2.45.1

