From c5fd83a417518e5393b94fe63217fb2ae5b00751 Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <sshedi@vmware.com>
Date: Fri, 1 Oct 2021 18:43:48 +0530
Subject: [PATCH 2/8] commit buffer cache to disk after ending rpm transaction

Signed-off-by: Shreenidhi Shedi <sshedi@vmware.com>
---
 lib/rpmts.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/rpmts.cc b/lib/rpmts.cc
index ebbdcaa..dbf5d3f 100644
--- a/lib/rpmts.cc
+++ b/lib/rpmts.cc
@@ -1129,6 +1129,7 @@ rpmtxn rpmtxnBegin(rpmts ts, rpmtxnFlags flags)
 
 rpmtxn rpmtxnEnd(rpmtxn txn)
 {
+    sync();
     if (txn) {
 	rpmlockRelease(txn->lock);
 	if (txn->flags & RPMTXN_WRITE)
-- 
2.43.7

