From 62c4f890fb0fd381be5551a3784d606233fceeef Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <shreenidhi.shedi@broadcom.com>
Date: Mon, 12 Jan 2026 17:18:55 +0000
Subject: [PATCH 7/8] replace format with string concatenation

Signed-off-by: Shreenidhi Shedi <shreenidhi.shedi@broadcom.com>
---
 tools/elfdeps.cc | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/tools/elfdeps.cc b/tools/elfdeps.cc
index 17884b2..7d0a210 100644
--- a/tools/elfdeps.cc
+++ b/tools/elfdeps.cc
@@ -1,6 +1,5 @@
 #include "system.h"
 
-#include <format>
 #include <string>
 #include <vector>
 
@@ -105,7 +104,7 @@ static void addSoDep(std::vector<std::string> & deps,
     if (ver.empty() && marker.empty()) {
 	addDep(deps, soname);
     } else {
-	auto dep = std::format("{}({}){}", soname, ver, marker);
+    auto dep = soname + "(" + ver + ")" + marker;
 	addDep(deps, dep);
     }
 }
-- 
2.43.7

