From cadc6827aeed436cce7a105dba33a49d6944069e Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <sshedi@vmware.com>
Date: Sun, 5 Nov 2023 11:29:06 +0530
Subject: [PATCH] dilute user, group requires

Signed-off-by: Shreenidhi Shedi <sshedi@vmware.com>
---
 macros.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/macros.in b/macros.in
index a047d1e38..5f4d6909e 100644
--- a/macros.in
+++ b/macros.in
@@ -244,7 +244,7 @@ Supplements:   (%{name} = %{version}-%{release} and langpacks-%{1})\
 %clamp_mtime_to_source_date_epoch 0
 
 #	If enabled, dilute user() and group() requires into recommends
-#%_use_weak_usergroup_deps 1
+%_use_weak_usergroup_deps 1
 
 #	The directory where newly built binary packages will be written.
 %_rpmdir		%{_topdir}/RPMS
-- 
2.25.1

