--- a/scripts/find-debuginfo.sh	2017-04-23 19:30:21.289059468 +0000
+++ b/scripts/find-debuginfo.sh	2017-04-23 20:24:52.579860254 +0000
@@ -314,7 +314,7 @@
     $strict && exit 2
   fi
 
-  [ type gdb-add-index >/dev/null 2>&1 && gdb-add-index "$f" > /dev/null 2>&1
+  type gdb-add-index >/dev/null 2>&1 && gdb-add-index "$f" > /dev/null 2>&1
 
   # A binary already copied into /usr/lib/debug doesn't get stripped,
   # just has its file names collected and adjusted.
