diff --color -rupN rpm-rpm-4.16.0-release/scripts/find-debuginfo.sh rpm-rpm-4.16.0-release-debug/scripts/find-debuginfo.sh
--- rpm-rpm-4.16.0-release/scripts/find-debuginfo.sh	2020-09-30 07:47:16.000000000 +0000
+++ rpm-rpm-4.16.0-release-debug/scripts/find-debuginfo.sh	2020-10-13 06:58:35.329328146 +0000
@@ -584,10 +584,10 @@ if [ -s "$SOURCEFILE" ]; then
 fi
 
 if [ -d "${RPM_BUILD_ROOT}/usr/lib" ] || [ -d "${RPM_BUILD_ROOT}/usr/src" ]; then
-  ((nout > 0)) ||
-  test ! -d "${RPM_BUILD_ROOT}/usr/lib" ||
-  (cd "${RPM_BUILD_ROOT}/usr/lib"; find debug -type d) |
-  sed 's,^,%dir /usr/lib/,' >> "$LISTFILE"
+#  ((nout > 0)) ||
+#  test ! -d "${RPM_BUILD_ROOT}/usr/lib" ||
+#  (cd "${RPM_BUILD_ROOT}/usr/lib"; find debug -type d) |
+#  sed 's,^,%dir /usr/lib/,' >> "$LISTFILE"
 
   (cd "${RPM_BUILD_ROOT}/usr"
    test ! -d lib/debug || find lib/debug ! -type d
@@ -659,7 +659,7 @@ while ((i < nout)); do
   mv "${LISTFILE}.new" "$LISTFILE"
   ((++i))
 done
-if ((nout > 0)); then
+if ((0 && nout > 0)); then
   # Generate %dir lines for each output list.
   generate_percent_dir()
   {
