diff -Naur a/scripts/find-debuginfo.sh b/scripts/find-debuginfo.sh
--- a/scripts/find-debuginfo.sh	2012-11-07 04:55:24.000000000 -0800
+++ b/scripts/find-debuginfo.sh	2016-08-26 17:51:50.000000000 -0700
@@ -217,8 +217,10 @@
     eval linked=\$linked_$inum
     if [ -n "$linked" ]; then
       eval id=\$linkedid_$inum
-      make_id_dup_link "$id" "$dn/$(basename $f)"
-      make_id_dup_link "$id" "/usr/lib/debug$dn/$bn" .debug
+      if [ -n "$id" ]; then
+        make_id_dup_link "$id" "$dn/$(basename $f)"
+        make_id_dup_link "$id" "/usr/lib/debug$dn/$bn" .debug
+      fi
       link=$debugfn
       get_debugfn "$linked"
       echo "hard linked $link to $debugfn"
