diff -Naur sqlite-2.8.17/tool/lemon.c sqlite-2.8.17_new/tool/lemon.c
--- sqlite-2.8.17/tool/lemon.c  2005-04-23 22:43:22.000000000 +0000
+++ sqlite-2.8.17_new/tool/lemon.c      2018-09-25 07:23:51.606534628 +0000
@@ -3016,7 +3016,7 @@
   int maxdtlength;          /* Maximum length of any ".datatype" field. */
   char *stddt;              /* Standardized name for a datatype */
   int i,j;                  /* Loop counters */
-  int hash;                 /* For hashing the name of a type */
+  unsigned int hash;        /* For hashing the name of a type */
   char *name;               /* Name of the parser */

   /* Allocate and initialize types[] and allocate stddt[] */
