From 386543bc4495f658dcce6cd4d11e4ba6574a46f5 Mon Sep 17 00:00:00 2001
From: "H.J. Lu" <hjl.tools@gmail.com>
Date: Wed, 26 Aug 2020 04:26:50 +0200
Subject: [PATCH 002/147] NEWS: Update for [BZ #26534] fix

---
 NEWS | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/NEWS b/NEWS
index 485b8ddffa0d41635d12383c2da1e2d4d5f949c6..a141de93dd0ad5892fa81e865df25e5a190f9abc 100644
--- a/NEWS
+++ b/NEWS
@@ -5,6 +5,11 @@ See the end for copying conditions.
 Please send GNU C library bug reports via <https://sourceware.org/bugzilla/>
 using `glibc' in the "product" field.
 
+The following bugs are resolved with this release:
+
+  [26534] libm.so 2.32 SIGILL in pow() due to FMA4 instruction on non-FMA4
+    system
+
 Version 2.32
 
 Major new features:
-- 
2.35.6

