From c3ceb93dc4f67253037644dc8f194831e27f3160 Mon Sep 17 00:00:00 2001
From: Adhemerval Zanella <adhemerval.zanella@linaro.org>
Date: Thu, 26 Feb 2026 10:21:24 -0300
Subject: [PATCH 10/27] Add BZ 33904 entry to NEWS

---
 NEWS | 1 +
 1 file changed, 1 insertion(+)

diff --git a/NEWS b/NEWS
index 1f07633e578493cd67c20f278a4c16889536bab8..cec8b61f5556b6bd43954f87ea714dc1d0a5d58b 100644
--- a/NEWS
+++ b/NEWS
@@ -13,6 +13,7 @@ The following bugs are resolved with this release:
    results
  [33848] build: Build fails at openat2.h, redefinition of 'struct
     open_how'
+ [33904] build: error: '__vasprintf_chk' undeclared here
 
 Version 2.43
 
-- 
2.53.0

