--- ceph-deploy-1.5.39/ceph_deploy/util/pkg_managers.py 2016-08-15 07:34:53.000000000 -0500
+++ ceph-deploy-1.5.39-patched/ceph_deploy/util/pkg_managers.py 2018-05-19 00:18:57.940650317 -0500
@@ -181,6 +181,13 @@
     executable = 'yum'
     name = 'yum'

+class Tdnf(RPMManagerBase):
+    """
+    The Tdnf Package manager
+    """
+
+    executable = 'tdnf'
+    name = 'tdnf'

 class Apt(PackageManager):
     """
