--- b/ceph_deploy/hosts/remotes.py	2016-08-15 12:34:51.000000000 +0000
+++ a/ceph_deploy/hosts/remotes.py	2017-01-23 23:43:38.822797323 +0000
@@ -14,6 +14,12 @@ def platform_information(_linux_distribu
     """ detect platform information from remote host """
     linux_distribution = _linux_distribution or platform.linux_distribution
     distro, release, codename = linux_distribution()
+
+    if 'photon' in distro.lower():
+        distro = 'photon'
+        release = '1.0'
+        codename = 'photon'
+
     if not codename and 'debian' in distro.lower():  # this could be an empty string in Debian
         debian_codenames = {
             '8': 'jessie',
