--- b/ceph_deploy/hosts/__init__.py	2016-02-08 14:11:39.000000000 +0000
+++ a/ceph_deploy/hosts/__init__.py	2017-01-05 02:34:50.662524086 +0000
@@ -7,7 +7,7 @@ on the type of distribution/version we a
 import logging
 from ceph_deploy import exc
 from ceph_deploy.util import versions
-from ceph_deploy.hosts import debian, centos, fedora, suse, remotes, rhel
+from ceph_deploy.hosts import debian, centos, fedora, suse, remotes, rhel, photon
 from ceph_deploy.connection import get_connection
 
 logger = logging.getLogger()
@@ -68,7 +68,7 @@ def get(hostname,
     module.distro = module.normalized_name
     module.is_el = module.normalized_name in ['redhat', 'centos', 'fedora', 'scientific']
     module.is_rpm = module.normalized_name in ['redhat', 'centos',
-                                               'fedora', 'scientific', 'suse']
+                                               'fedora', 'scientific', 'suse', 'photon']
     module.is_deb = not module.is_rpm
     module.release = release
     module.codename = codename
@@ -96,6 +96,7 @@ def _get_distro(distro, fallback=None, u
         'redhat': centos,
         'fedora': fedora,
         'suse': suse,
+        'photon': photon,
         }
 
     if distro == 'redhat' and use_rhceph:
@@ -112,6 +113,8 @@ def _normalized_distro_name(distro):
         return 'scientific'
     elif distro.startswith(('suse', 'opensuse')):
         return 'suse'
+    elif distro.startswith('photon'):
+        return 'photon'
     elif distro.startswith('centos'):
         return 'centos'
     elif distro.startswith('linuxmint'):
