--- ceph-deploy-1.5.39/ceph_deploy/hosts/__init__.py    2017-04-28 12:32:51.000000000 -0500
+++ ceph-deploy-1.5.39-patched/ceph_deploy/hosts/__init__.py    2018-05-19 00:18:57.938650317 -0500
@@ -7,7 +7,7 @@
 import logging
 from ceph_deploy import exc
 from ceph_deploy.util import versions
-from ceph_deploy.hosts import debian, centos, fedora, suse, remotes, rhel
+from ceph_deploy.hosts import debian, centos, fedora, suse, remotes, rhel, photon
 from ceph_deploy.connection import get_connection

 logger = logging.getLogger()
@@ -68,7 +68,7 @@
     module.distro = module.normalized_name
     module.is_el = module.normalized_name in ['redhat', 'centos', 'fedora', 'scientific', 'oracle', 'virtuozzo']
     module.is_rpm = module.normalized_name in ['redhat', 'centos',
-                                               'fedora', 'scientific', 'suse', 'oracle', 'virtuozzo']
+                                               'fedora', 'scientific', 'suse', 'oracle', 'virtuozzo', 'photon']
     module.is_deb = not module.is_rpm
     module.release = release
     module.codename = codename
@@ -97,7 +97,8 @@
         'redhat': centos,
         'fedora': fedora,
         'suse': suse,
-        'virtuozzo' : centos
+        'virtuozzo' : centos,
+        'photon': photon
         }

     if distro == 'redhat' and use_rhceph:
@@ -118,6 +119,8 @@
         return 'suse'
     elif distro.startswith('centos'):
         return 'centos'
+    elif distro.startswith('photon'):
+        return 'photon'
     elif distro.startswith('linuxmint'):
         return 'ubuntu'
     elif distro.startswith('virtuozzo'):
