--- a/ceph_deploy/util/pkg_managers.py	2017-01-03 21:38:42.000000000 +0000
+++ b/ceph_deploy/util/pkg_managers.py	2017-03-30 21:06:24.369709803 +0000
@@ -181,6 +181,13 @@ class Yum(RPMManagerBase):
     executable = 'yum'
     name = 'yum'
 
+class Tdnf(RPMManagerBase):
+    """
+    The Tdnf Package manager
+    """
+
+    executable = 'tdnf'
+    name = 'tdnf'
 
 class Apt(PackageManager):
     """
