--- ceph-deploy-1.5.39/ceph_deploy/hosts/photon/install.py      2018-05-19 00:37:38.354704247 -0500
+++ ceph-deploy-1.5.39-patched/ceph_deploy/hosts/photon/install.py      2018-05-19 00:18:57.939650317 -0500
@@ -0,0 +1,33 @@
+from ceph_deploy.lib import remoto
+from ceph_deploy.util.paths import gpg
+from ceph_deploy.hosts.common import map_components
+
+
+NON_SPLIT_PACKAGES = ['ceph-osd', 'ceph-mon', 'ceph-mds']
+
+def repo_install():
+    pass
+
+def mirror_install():
+    pass
+
+def install(distro, version_kind, version, adjust_repos, **kw):
+    packages = map_components(
+        NON_SPLIT_PACKAGES,
+        kw.pop('components', [])
+    )
+    gpgcheck = kw.pop('gpgcheck', 1)
+
+    logger = distro.conn.logger
+    release = 'el7'
+    machine = distro.machine_type
+
+    if version_kind in ['stable', 'testing']:
+        key = 'release'
+    else:
+        key = 'autobuild'
+
+    print packages
+    distro.packager.install(
+        packages
+    )
