--- ceph-deploy-1.5.39/ceph_deploy/hosts/remotes.py     2017-09-01 06:30:23.000000000 -0500
+++ ceph-deploy-1.5.39-patched/ceph_deploy/hosts/remotes.py     2018-05-19 00:18:57.939650317 -0500
@@ -14,6 +14,12 @@
     """ detect platform information from remote host """
     linux_distribution = _linux_distribution or platform.linux_distribution
     distro, release, codename = linux_distribution()
+
+    if 'photon' in distro.lower():
+        distro = 'photon'
+        release = '2.0'
+        codename = 'photon'
+
     if not codename and 'debian' in distro.lower():  # this could be an empty string in Debian
         debian_codenames = {
             '10': 'buster',
