--- ceph-deploy-1.5.39/ceph_deploy/hosts/photon/__init__.py     2018-05-19 00:37:38.354704247 -0500
+++ ceph-deploy-1.5.39-patched/ceph_deploy/hosts/photon/__init__.py     2018-05-19 00:18:57.939650317 -0500
@@ -0,0 +1,23 @@
+from . import mon  # noqa
+from ceph_deploy.hosts.centos.install import repo_install  # noqa
+from .install import install, mirror_install  # noqa
+from .uninstall import uninstall  # noqa
+from ceph_deploy.util import pkg_managers
+
+# Allow to set some information about this distro
+#
+
+distro = None
+release = None
+codename = None
+
+def choose_init(module):
+    """
+    Select a init system
+
+    Returns the name of a init system (upstart, sysvinit ...).
+    """
+    return 'systemd'
+
+def get_packager(module):
+        return pkg_managers.Yum(module)
