From 1c4595e3d362a241cfcc3feb62bf04592ac55c20 Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <shreenidhi.shedi@broadcom.com>
Date: Wed, 11 Jul 2024 13:56:13 +0000
Subject: [PATCH] Use proxy if available

Signed-off-by: Shreenidhi Shedi <shreenidhi.shedi@broadcom.com>
---
 build.gradle    | 15 +++++++++++++--
 settings.gradle | 18 ++++++++++++++++++
 2 files changed, 31 insertions(+), 2 deletions(-)

diff --git a/build.gradle b/build.gradle
index cacda19..1327eea 100644
--- a/build.gradle
+++ b/build.gradle
@@ -17,9 +17,15 @@ import org.ajoberstar.grgit.Grgit

 import java.nio.charset.StandardCharsets

+
 buildscript {
   repositories {
-    mavenCentral()
+    def proxyUrl = System.getenv("GRADLE_PROXY_URL")
+    if (proxyUrl) {
+      maven { url "${proxyUrl}" }
+    } else {
+      mavenCentral()
+    }
   }
   apply from: "$rootDir/gradle/dependencies.gradle"

@@ -120,7 +126,12 @@ ext {
 allprojects {

   repositories {
-    mavenCentral()
+    def proxyUrl = System.getenv("GRADLE_PROXY_URL")
+    if (proxyUrl) {
+      maven { url "${proxyUrl}" }
+    } else {
+      mavenCentral()
+    }
   }

   dependencyUpdates {
diff --git a/settings.gradle b/settings.gradle
index a406cc0..607aad2 100644
--- a/settings.gradle
+++ b/settings.gradle
@@ -13,6 +13,24 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.

+def proxyUrl = System.getenv("GRADLE_PROXY_URL")
+
+if (proxyUrl) {
+  pluginManagement {
+    repositories {
+      def pluginMirror = System.getenv("GRADLE_PLUGINS_URL_MIRROR1")
+      if (pluginMirror) {
+        maven { url "${pluginMirror}" }
+      }
+
+      pluginMirror = System.getenv("GRADLE_PLUGINS_URL_MIRROR2")
+      if (pluginMirror) {
+        maven { url "${pluginMirror}" }
+      }
+    }
+  }
+}
+
 include 'clients',
     'connect:api',
     'connect:basic-auth-extension',
--
2.39.4

