From 01097db6bcd6ec1a850d61fc27941c6c0b0956da Mon Sep 17 00:00:00 2001
From: Bo Gan <ganb@vmware.com>
Date: Thu, 15 Dec 2022 11:26:04 +0000
Subject: [PATCH] confdata: format change for split script

---
 scripts/kconfig/confdata.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/scripts/kconfig/confdata.c b/scripts/kconfig/confdata.c
index 76193ce5a792..7d004a477b63 100644
--- a/scripts/kconfig/confdata.c
+++ b/scripts/kconfig/confdata.c
@@ -524,7 +524,7 @@ struct comment_style {
 };
 
 static const struct comment_style comment_style_pound = {
-	.decoration = "#",
+	.decoration = "###",
 	.prefix = "#",
 	.postfix = "#",
 };
@@ -840,9 +840,8 @@ int conf_write(const char *name)
 				goto next;
 			str = menu_get_prompt(menu);
 			fprintf(out, "\n"
-				     "#\n"
-				     "# %s\n"
-				     "#\n", str);
+				"#%s %s\n", menu->prompt->type == P_MENU ?
+				"" : "#", str);
 			need_newline = false;
 		} else if (!sym_is_choice(sym) &&
 			   !(sym->flags & SYMBOL_WRITTEN)) {
-- 
2.35.6

