From cea28e356075512497d87b479e8319c30439b4e1 Mon Sep 17 00:00:00 2001
From: Clark Williams <clark.williams@gmail.com>
Date: Thu, 12 Sep 2024 16:45:11 -0500
Subject: [PATCH 63/63] Linux 6.1.109-rt41 REBASE

Signed-off-by: Clark Williams <clark.williams@gmail.com>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index c5b71f9a229d..629e0b4384b8 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt25
+-rt41
-- 
2.46.0

