From 26d18e27c55e5c9e90d40a630bdddfb51a227a4a Mon Sep 17 00:00:00 2001
From: Clark Williams <clark.williams@gmail.com>
Date: Sun, 13 Oct 2024 19:56:10 -0500
Subject: [PATCH 63/63] Linux 6.1.112-rt43 REBASE

Signed-off-by: Clark Williams <clark.williams@gmail.com>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index c5b71f9a229d..41f7b993a830 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt25
+-rt43
-- 
2.47.0

