From 8efa700e76ca084546cfc4931e931c3f8136c118 Mon Sep 17 00:00:00 2001
From: Oliver Kurth <oliver.kurth@broadcom.com>
Date: Thu, 21 May 2026 23:10:42 +0000
Subject: [PATCH 4/7] use SSL_CERT_FILE environment variable

Signed-off-by: Oliver Kurth <oliver.kurth@broadcom.com>
---
 client/repolist.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/client/repolist.c b/client/repolist.c
index 08d2335..65e85ab 100644
--- a/client/repolist.c
+++ b/client/repolist.c
@@ -289,6 +289,16 @@ TDNFLoadRepoData(
         }
     }
 
+    /* Apply SSL_CERT_FILE environment variable, unless overridden by global setopt */
+    if (getenv("SSL_CERT_FILE") &&
+        (!pTdnf->pArgs->cn_setopts || !find_child(pTdnf->pArgs->cn_setopts, TDNF_CONF_KEY_SSL_CA_CERT))) {
+        for (pRepo = pReposAll; pRepo; pRepo = pRepo->pNext) {
+            TDNF_SAFE_FREE_MEMORY(pRepo->pszSSLCaCert);
+            dwError = TDNFSafeAllocateString(getenv("SSL_CERT_FILE"), &pRepo->pszSSLCaCert);
+            BAIL_ON_TDNF_ERROR(dwError);
+        }
+    }
+
     /* look for setopt settings */
     if (pTdnf->pArgs->cn_repoopts != NULL) {
         for (pRepo = pReposAll; pRepo; pRepo = pRepo->pNext) {
-- 
2.43.0

