From 24edf9d1a929f0801ada2b784ed1faaa799f7884 Mon Sep 17 00:00:00 2001
From: Oliver Kurth <oliver.kurth@broadcom.com>
Date: Wed, 29 Apr 2026 20:40:16 +0000
Subject: [PATCH] fix 'assignment discards 'const' qualifier warning'

Signed-off-by: Oliver Kurth <oliver.kurth@broadcom.com>
---
 llconf/nodes.c     | 2 +-
 solv/tdnfpackage.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/llconf/nodes.c b/llconf/nodes.c
index c669399..fb081d8 100644
--- a/llconf/nodes.c
+++ b/llconf/nodes.c
@@ -53,7 +53,7 @@ struct cnfnode *
 create_cnfnode_keyval(const char *keyval)
 {
     struct cnfnode *cn = NULL;
-    char *psep;
+    const char *psep;
 
     psep = strstr(keyval, "=");
     if (psep) {
diff --git a/solv/tdnfpackage.c b/solv/tdnfpackage.c
index 3bc6c6a..9398529 100644
--- a/solv/tdnfpackage.c
+++ b/solv/tdnfpackage.c
@@ -1856,7 +1856,7 @@ check_for_providers(
     char *prv_pkgname
     )
 {
-    char *beg;
+    const char *beg;
     const char *end = NULL;
     uint32_t dwError = 0;
     char pkgname[256] = {0};
-- 
2.43.0

