From 3fce9af4d0d644869e3d817e08fa7a860d6dca72 Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <shreenidhi.shedi@broadcom.com>
Date: Wed, 13 Mar 2024 20:05:01 +0530
Subject: [PATCH] fix syslog-ng profile

Signed-off-by: Shreenidhi Shedi <shreenidhi.shedi@broadcom.com>
---
 profiles/apparmor.d/sbin.syslog-ng | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/profiles/apparmor.d/sbin.syslog-ng b/profiles/apparmor.d/sbin.syslog-ng
index 5ae8587..244c8fc 100644
--- a/profiles/apparmor.d/sbin.syslog-ng
+++ b/profiles/apparmor.d/sbin.syslog-ng
@@ -17,7 +17,8 @@ include <tunables/global>
 #define this to be where syslog-ng is chrooted
 @{CHROOT_BASE}=""
 
-profile syslog-ng /{usr/,}{bin,sbin}/syslog-ng {
+# TODO: attach_disconnected has to be removed at some point
+profile syslog-ng /{usr/,}{bin,sbin}/syslog-ng flags=(attach_disconnected) {
   include <abstractions/base>
   include <abstractions/consoles>
   include <abstractions/nameservice>
@@ -44,6 +45,7 @@ profile syslog-ng /{usr/,}{bin,sbin}/syslog-ng {
   /dev/xconsole rw,
   /dev/kmsg r,
   /etc/machine-id r,
+  /etc/ssl/*.cnf r,
   /etc/syslog-ng/* r,
   /etc/syslog-ng/conf.d/ r,
   /etc/syslog-ng/conf.d/* r,
@@ -54,7 +56,7 @@ profile syslog-ng /{usr/,}{bin,sbin}/syslog-ng {
   /var/lib/syslog-ng/syslog-ng-?????.qf rw,
   # chrooted applications
   @{CHROOT_BASE}/var/lib/*/dev/log w,
-  @{CHROOT_BASE}/var/lib/syslog-ng/syslog-ng.persist* rw,
+  @{CHROOT_BASE}/var/lib/syslog-ng/* rw,
   @{CHROOT_BASE}/var/log/** w,
   @{CHROOT_BASE}/@{run}/syslog-ng.pid krw,
   @{CHROOT_BASE}/@{run}/syslog-ng.ctl rw,
@@ -64,6 +66,7 @@ profile syslog-ng /{usr/,}{bin,sbin}/syslog-ng {
   /{var,var/run,run}/log/journal/*.journal r,
   @{run}/syslog-ng.ctl a,
   @{run}/syslog-ng/additional-log-sockets.conf r,
+  owner /systemd/notify rw,
 
   # Site-specific additions and overrides. See local/README for details.
   include if exists <local/sbin.syslog-ng>
-- 
2.44.0

