From 525f3923bd2a86a096d2230878289afe680bc7ac Mon Sep 17 00:00:00 2001
From: Alexey Makhalov <amakhalov@vmware.com>
Date: Fri, 7 Oct 2022 20:53:16 +0000
Subject: Speedup algos registration in non fips mode

Signed-off-by: Alexey Makhalov <amakhalov@vmware.com>
---
 crypto/fips_canister_wrapper.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/crypto/fips_canister_wrapper.c b/crypto/fips_canister_wrapper.c
index 27776bbffe9f..3191135675fb 100644
--- a/crypto/fips_canister_wrapper.c
+++ b/crypto/fips_canister_wrapper.c
@@ -239,6 +239,17 @@
 
 	if (msg == CRYPTO_MSG_ALG_REGISTER) {
 		struct crypto_alg *alg = (struct crypto_alg *)data;
+		/*
+		 * If FIPS disabled, we want to speed up the registration path
+		 * by marking alg test as passed and returning NOTIFY_STOP.
+		 * If FIPS enabled, starting from 2023 we want to register notifier
+		 * to disallow DES3.
+		 */
+		if (!fips_enabled) {
+			crypto_alg_tested(alg->cra_driver_name, 0);
+			return NOTIFY_STOP;
+		}
+
 		/* Disable non FIPS approved algos */
 		if (fcw_fips_not_allowed_alg(alg->cra_name))
 			return NOTIFY_OK;
-- 
2.23.3

