From 0419aa5453252c38bba6f3a951a3cf4501f1e1b7 Mon Sep 17 00:00:00 2001
From: Alexey Makhalov <amakhalov@vmware.com>
Date: Tue, 22 Feb 2022 22:42:04 +0000
Subject: [PATCH] Speedup algos registration in non fips mode

---
 crypto/fips_canister_wrapper.c | 23 ++++++++++++++++-------
 1 file changed, 19 insertions(+), 7 deletions(-)

diff --git a/crypto/fips_canister_wrapper.c b/crypto/fips_canister_wrapper.c
index 1416e7186..e8b39f273 100644
--- a/crypto/fips_canister_wrapper.c
+++ b/crypto/fips_canister_wrapper.c
@@ -1,4 +1,4 @@
-// Uncomment it in 2023. des3 will not by allowed in FIPS mode.
+// Uncomment it in 2023. des3 will not be allowed in FIPS mode.
 //#define DES3_IS_NOT_ALLOWED
 
 #include <linux/kernel.h>
@@ -9,12 +9,10 @@
 #include <linux/sched.h>
 #include <linux/version.h>
 #include <linux/workqueue.h>
-#ifdef DES3_IS_NOT_ALLOWED
 #include <linux/notifier.h>
 #include <linux/module.h>
 #include <linux/fips.h>
 #include <crypto/algapi.h>
-#endif
 #include <crypto/aead.h>
 #include <crypto/hash.h>
 #include <crypto/akcipher.h>
@@ -24,6 +22,7 @@
 #include <crypto/internal/hash.h>
 #endif
 #include <asm/fpu/api.h>
+#include "internal.h"
 
 
 int fcw_cond_resched(void)
@@ -131,13 +130,16 @@ int fcw_fips_not_allowed_alg(char *alg_name)
 	return 0;
 }
 
-#ifdef DES3_IS_NOT_ALLOWED
 static int crypto_msg_notify(struct notifier_block *this, unsigned long msg,
 			    void *data)
 {
 	if (msg == CRYPTO_MSG_ALG_REGISTER)
 	{
 		struct crypto_alg *alg = (struct crypto_alg *)data;
+		if (!fips_enabled) {
+			crypto_alg_tested(alg->cra_driver_name, 0);
+			return NOTIFY_STOP;
+		}
 		/* Disable non FIPS approved algos */
 		if (fcw_fips_not_allowed_alg(alg->cra_name))
 			return NOTIFY_OK;
@@ -150,12 +152,19 @@ static struct notifier_block crypto_msg_notifier = {
 	.notifier_call = crypto_msg_notify,
 };
 
+/*
+ * If FIPS disabled, we want to speed up the registration path
+ * by marking alg test as passed and returning NOTIFY_STOP.
+ * If FIPS enabled, starting from 2023 we want to register notifier
+ * to disallow DES3.
+ */
 static int __init wrapper_init(void)
 {
-	if (!fips_enabled)
+#ifndef DES3_IS_NOT_ALLOWED
+	if (fips_enabled)
 		return 0;
-
+#endif
 	return crypto_register_notifier(&crypto_msg_notifier);
 }
 arch_initcall(wrapper_init);
-#endif
+
-- 
2.30.0

