From a71a22c075c306d3ff252bf879ede69cc5e25477 Mon Sep 17 00:00:00 2001
From: Nitesh Kumar <kunitesh@vmware.com>
Date: Mon, 13 Feb 2023 22:51:58 +0530
Subject: [PATCH] Set proper config path in include section

---
 config/jail.conf | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/config/jail.conf b/config/jail.conf
index fe8db52..ae06348 100644
--- a/config/jail.conf
+++ b/config/jail.conf
@@ -33,7 +33,8 @@
 [INCLUDES]
 
 #before = paths-distro.conf
-before = paths-debian.conf
+#before = paths-debian.conf
+before = paths-common.conf
 
 # The DEFAULT allows a global definition of the options. They can be overridden
 # in each jail afterwards.
-- 
2.17.1

