From 02eae6084472f65ea4c9a965d21ae3303d1abe9b Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <sshedi@vmware.com>
Date: Mon, 28 Mar 2022 20:40:42 +0530
Subject: [PATCH 18/35] system fstool

Signed-off-by: Shreenidhi Shedi <sshedi@vmware.com>
---
 policy/modules/system/fstools.te | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/policy/modules/system/fstools.te b/policy/modules/system/fstools.te
index 0085819..14e20a4 100644
--- a/policy/modules/system/fstools.te
+++ b/policy/modules/system/fstools.te
@@ -185,6 +185,12 @@ ifdef(`distro_redhat',`
 	')
 ')
 
+ifdef(`distro_photon',`
+	optional_policy(`
+		unconfined_domain(fsadm_t)
+	')
+')
+
 optional_policy(`
 	amanda_rw_dumpdates_files(fsadm_t)
 	amanda_append_log_files(fsadm_t)
-- 
2.45.0

