--- a/wscript     2017-08-04 00:21:24.741655672 +0000
+++ b/wscript     2017-08-04 00:22:15.001653875 +0000
@@ -163,6 +163,8 @@
     magic_helper_cmd = os.path.join(Utils.g_module.blddir, 'talloc_test_magic_differs_helper')
     magic_cmd = os.path.join(srcdir, 'lib', 'talloc',
                              'test_magic_differs.sh')
+    if not os.path.exists(magic_cmd):
+        magic_cmd = os.path.join(srcdir, 'test_magic_differs.sh')

     magic_ret = samba_utils.RUN_COMMAND(magic_cmd + " " +  magic_helper_cmd)
     print("magic differs test returned %d" % magic_ret)
