From fc28aacfac1dacf10d2b85ffd6f757fbabb01c30 Mon Sep 17 00:00:00 2001
From: Clark Williams <williams@redhat.com>
Date: Fri, 7 May 2021 15:05:03 -0500
Subject: [PATCH 338/338] Linux 4.19.190-rt79 REBASE

Signed-off-by: Clark Williams <williams@redhat.com>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index c1f2720eaf25..cf94a551bf32 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt76
+-rt79
-- 
2.31.1

