diff -Naur linux-4.9.66/crypto/testmgr.h linux-4.9.66-kat08/crypto/testmgr.h
--- linux-4.9.66/crypto/testmgr.h	2017-12-08 07:39:22.824813034 +0000
+++ linux-4.9.66-kat08/crypto/testmgr.h	2017-12-08 07:47:22.124830175 +0000
@@ -3261,7 +3261,7 @@
 		.ksize	= 32,
 		.plaintext = "abc",
 		.psize	= 3,
-		.digest	= "\xa2\x1b\x1f\x5d\x4c\xf4\xf7\x3a"
+		.digest	= "\x23\x32\x23\x32\x4c\xf4\xf7\x3a"
 			  "\x4d\xd9\x39\x75\x0f\x7a\x06\x6a"
 			  "\x7f\x98\xcc\x13\x1c\xb1\x6a\x66"
 			  "\x92\x75\x90\x21\xcf\xab\x81\x81",
