diff -Naur linux-4.9.66/crypto/testmgr.h linux-4.9.66-kat07/crypto/testmgr.h
--- linux-4.9.66/crypto/testmgr.h	2017-12-08 07:39:22.824813034 +0000
+++ linux-4.9.66-kat07/crypto/testmgr.h	2017-12-08 07:46:38.700828623 +0000
@@ -3193,7 +3193,7 @@
 			"\x48\x61\x73\x68\x20\x4b\x65\x79"
 			"\x20\x46\x69\x72\x73\x74",
 		.psize  = 54,
-		.digest = "\x95\xe9\xa0\xdb\x96\x20\x95\xad"
+		.digest = "\x23\x32\x23\x32\x96\x20\x95\xad"
 			"\xae\xbe\x9b\x2d\x6f\x0d\xbc\xe2"
 			"\xd4\x99\xf1\x12\xf2\xd2\xb7\x27"
 			"\x3f\xa6\x87\x0e",
