diff -Naur linux-4.9.66/crypto/testmgr.h linux-4.9.66-kat01/crypto/testmgr.h
--- linux-4.9.66/crypto/testmgr.h	2017-12-08 07:39:22.824813034 +0000
+++ linux-4.9.66-kat01/crypto/testmgr.h	2017-12-08 07:40:57.968816437 +0000
@@ -15591,7 +15591,7 @@
 			  "\xf6\x9f\x24\x45\xdf\x4f\x9b\x17"
 			  "\xad\x2b\x41\x7b\xe6\x6c\x37\x10",
 		.ilen	= 64,
-		.result	= "\x4f\x02\x1d\xb2\x43\xbc\x63\x3d"
+		.result	= "\x23\x32\x23\x32\x43\xbc\x63\x3d"
 			  "\x71\x78\x18\x3a\x9f\xa0\x71\xe8"
 			  "\xb4\xd9\xad\xa9\xad\x7d\xed\xf4"
 			  "\xe5\xe7\x38\x76\x3f\x69\x14\x5a"
