diff -Naur linux-4.9.66/crypto/testmgr.h linux-4.9.66-kat05/crypto/testmgr.h
--- linux-4.9.66/crypto/testmgr.h	2017-12-08 07:39:22.824813034 +0000
+++ linux-4.9.66-kat05/crypto/testmgr.h	2017-12-08 07:45:23.844825945 +0000
@@ -5530,7 +5530,7 @@
 		.klen	= 24,
 		.input	= "\x18\xd7\x48\xe5\x63\x62\x05\x72",
 		.ilen	= 8,
-		.result	= "\x73\x6f\x6d\x65\x64\x61\x74\x61",
+		.result	= "\x23\x32\x23\x32\x64\x61\x74\x61",
 		.rlen	= 8,
 	}, {
 		.key	= "\x03\x52\x02\x07\x67\x20\x82\x17"
