diff -Naur linux-4.9.66/crypto/testmgr.h linux-4.9.66-kat03/crypto/testmgr.h
--- linux-4.9.66/crypto/testmgr.h	2017-12-08 07:39:22.824813034 +0000
+++ linux-4.9.66-kat03/crypto/testmgr.h	2017-12-08 07:43:55.884822800 +0000
@@ -18490,7 +18490,7 @@
 			  "\xf0\xf1\xf2\xf3\xf4\xf5\xf6\xf7"
 			  "\xf8\xf9\xfa\xfb\xfc\xfd\xfe\xff",
 		.ilen	= 512,
-		.result	= "\x1c\x3b\x3a\x10\x2f\x77\x03\x86"
+		.result	= "\x23\x32\x23\x32\x2f\x77\x03\x86"
 			  "\xe4\x83\x6c\x99\xe3\x70\xcf\x9b"
 			  "\xea\x00\x80\x3f\x5e\x48\x23\x57"
 			  "\xa4\xae\x12\xd4\x14\xa3\xe6\x3b"
