From 1d4dabfa66d8f4064b0dbcad7178c7f2a122ea26 Mon Sep 17 00:00:00 2001
From: Daniel Wagner <wagi@monom.org>
Date: Wed, 22 May 2024 14:04:42 +0200
Subject: [PATCH 351/351] Linux 4.19.312-rt134 REBASE

Signed-off-by: Daniel Wagner <wagi@monom.org>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 1199ebade17b..6067da4c8c99 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt16
+-rt134
-- 
2.44.0

