diff -rauN tcsh-6.19.00/tc.alloc.c tcsh-6.19.00-patch-calloc-gcc-5/tc.alloc.c
--- tcsh-6.19.00/tc.alloc.c	2015-02-22 17:31:54.000000000 +0100
+++ tcsh-6.19.00-patch-calloc-gcc-5/tc.alloc.c	2015-07-10 09:46:35.896959230 +0200
@@ -348,10 +348,12 @@
 {
 #ifndef lint
     char *cp;
+    volatile size_t k;
 
     i *= j;
     cp = xmalloc(i);
-    memset(cp, 0, i);
+    k = i;
+    memset(cp, 0, k);
 
     return ((memalign_t) cp);
 #else
