From 54662cfe7c7eb3ccb802d57366393dacc4ad5e00 Mon Sep 17 00:00:00 2001
From: Clark Williams <clrkwllms@kernel.org>
Date: Tue, 19 Sep 2023 13:17:12 -0500
Subject: [PATCH 62/62] 'Linux 6.1.54-rt15 REBASE'

Signed-off-by: Clark Williams <clrkwllms@kernel.org>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index 9f7d0bdbf..18777ec0c 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt13
+-rt15
-- 
2.41.0

