From 349b051fa66b417b89d2b8eb992b0ae1dca22ea2 Mon Sep 17 00:00:00 2001
From: Vamsi Krishna Brahmajosyula <vamsi-krishna.brahmajosyula@broadcom.com>
Date: Fri, 24 Apr 2026 04:35:47 +0000
Subject: [PATCH] fix memset_explicit usage for newer glibc

---
 lib/replace/replace.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/replace/replace.c b/lib/replace/replace.c
index 68829f2..67663fb 100644
--- a/lib/replace/replace.c
+++ b/lib/replace/replace.c
@@ -970,7 +970,7 @@ int rep_memset_s(void *dest, size_t destsz, int ch, size_t count)
 	}
 
 #if defined(HAVE_MEMSET_EXPLICIT)
-	memset_explicit(dest, destsz, ch, count);
+	memset_explicit(dest, ch, count);
 #else /* HAVE_MEMSET_EXPLICIT */
 	memset(dest, ch, count);
 # if defined(HAVE_GCC_VOLATILE_MEMORY_PROTECTION)
-- 
2.43.7

