diff --git a/Gemfile.jruby-2.5.lock.release b/Gemfile.jruby-2.5.lock.release
index b12ad13..6b6d360 100644
--- a/Gemfile.jruby-2.5.lock.release
+++ b/Gemfile.jruby-2.5.lock.release
@@ -278,7 +278,7 @@ GEM
       logstash-codec-plain
       logstash-core-plugin-api (~> 2.0)
       stud (>= 0.0.22)
-    logstash-input-beats (5.1.8-java)
+    logstash-input-beats (5.1.9-java)
       concurrent-ruby (~> 1.0)
       jar-dependencies (~> 0.3, >= 0.3.4)
       logstash-codec-multiline (>= 2.0.5)
