diff -ru gosc-scripts/imc-shell/imgcust-scripts/PhotonCustomization.sh gosc-scripts-modified/imc-shell/imgcust-scripts/PhotonCustomization.sh
--- gosc-scripts/imc-shell/imgcust-scripts/PhotonCustomization.sh	2015-06-17 16:22:17.000000000 -0700
+++ gosc-scripts-modified/imc-shell/imgcust-scripts/PhotonCustomization.sh	2016-03-25 12:06:46.786688654 -0700
@@ -32,9 +32,11 @@
   if [[ -z "$utc" ]]; then
     Debug "Not changing UTC"
   elif [[ $utc == 1 ]]; then
-    Exec '/usr/bin/timedatectl set-local-rtc 0' '' ''
+    #Exec '/usr/bin/timedatectl set-local-rtc 0' '' ''
+    /usr/bin/timedatectl set-local-rtc 0 2>&1
   else
-    Exec '/usr/bin/timedatectl set-local-rtc 1' '' ''
+    #Exec '/usr/bin/timedatectl set-local-rtc 1' '' ''
+    /usr/bin/timedatectl set-local-rtc 1 2>&1
   fi
 
   local tz=$(ConfigFile_GetTimeZone)
