diff -ru open-vm-tools-stable-10.1.10/open-vm-tools/libDeployPkg/linuxDeployment.c open-vm-tools-stable-10.1.10-modified/open-vm-tools/libDeployPkg/linuxDeployment.c
--- open-vm-tools-stable-10.1.10/open-vm-tools/libDeployPkg/linuxDeployment.c	2017-07-28 14:59:15.000000000 -0700
+++ open-vm-tools-stable-10.1.10-modified/open-vm-tools/libDeployPkg/linuxDeployment.c	2017-08-22 12:30:18.862037086 -0700
@@ -582,13 +582,17 @@
    close(fd);
 
    // Create space and copy the command
-   *command = malloc(VMWAREDEPLOYPKG_CMD_LENGTH);
+   *command = malloc(VMWAREDEPLOYPKG_CMD_LENGTH + 1);
    if (!*command) {
       SetDeployError("Error allocating memory.");
       return FALSE;
    }
 
-   memcpy(*command, hdr.command, VMWAREDEPLOYPKG_CMD_LENGTH);
+   // will pad with '\0' if necessary
+   strncpy(*command, hdr.command, VMWAREDEPLOYPKG_CMD_LENGTH);
+   // hdr.command is a structure of fixed length VMWAREDEPLOYPKG_CMD_LENGTH
+   (*command)[VMWAREDEPLOYPKG_CMD_LENGTH] = '\0';
+
    *archiveType = hdr.payloadType;
    *flags = hdr.reserved;
 
@@ -1094,6 +1098,20 @@
    } else {
       sLog(log_info, "cloud-init is either not installed or the flag to enable \
            cloud-init is not set.\n Executing traditional GOSC workflow");
+      if ((strstr(command, "scripts/Customize.pl") != NULL) || (strstr(command, "scripts/customize.sh") != NULL)) {
+          free (command);
+          const char* PHOTON_OS_LAUNCH_COMMAND = "/bin/bash /usr/share/open-vm-tools/GOSC/gosc-scripts/imc-shell/imgcust-scripts/customize.sh /tmp/.vmware/linux/deploy/cust.cfg";
+
+          // The command will be parsed alter and can be of arbitrary length
+          command = StrUtil_ReplaceAll(PHOTON_OS_LAUNCH_COMMAND, TMP_PATH_VAR, tmpDirPath);
+          if (!command) {
+             SetDeployError("Error allocating memory.");
+             return DEPLOY_ERROR;
+          }
+          _DeployPkg_SkipReboot(true);
+      }
+      // Run the deployment command
+      sLog(log_info, "Launching deployment %s.  \n", command);
       deploymentResult = ForkExecAndWaitCommand(command);
       free(command);
 
