diff -Naur gosc-scripts_bak/imc-shell/imgcust-scripts/CustomizationUtils.sh gosc-scripts/imc-shell/imgcust-scripts/CustomizationUtils.sh
--- gosc-scripts_bak/imc-shell/imgcust-scripts/CustomizationUtils.sh	2020-01-24 16:25:10.000000000 +0530
+++ gosc-scripts/imc-shell/imgcust-scripts/CustomizationUtils.sh	2020-03-10 10:57:55.711029873 +0530
@@ -726,7 +726,7 @@
 PostGCStatus()
 {
   local STATUS=$1
-  local OUTPUT="$(ps -C vmtoolsd -o cmd=)"
+  local OUTPUT="$(which vmtoolsd)"
   local CMD="$OUTPUT --cmd \"info-set guestinfo.gc.status $STATUS\""
   eval $CMD
 }
@@ -1316,7 +1316,7 @@
 #Set deployPackage status
 PostDeployPkgStatus() {
   local MSG=$1
-  local OUTPUT="$(ps -C vmtoolsd -o cmd=)"
+  local OUTPUT="$(which vmtoolsd)"
   local CMD="$OUTPUT --cmd \"deployPkg.update.state 4 6 $MSG\""
   eval $CMD
 }
