diff -ru gosc-scripts/imc-shell/imgcust-scripts/CustomizationUtils.sh gosc-scripts-modified/imc-shell/imgcust-scripts/CustomizationUtils.sh
--- gosc-scripts/imc-shell/imgcust-scripts/CustomizationUtils.sh	2016-02-01 19:02:43.261707000 -0800
+++ gosc-scripts-modified/imc-shell/imgcust-scripts/CustomizationUtils.sh	2016-02-01 19:15:56.697982085 -0800
@@ -392,6 +392,7 @@
 
   Debug "Patching /etc/hosts with [$res]"
   echo "${res}" > /etc/hosts
+  "127.0.0.1   localhost" >> /etc/hosts
 }
 
 # Detects interface name by MAC address out of 'ip addr show' output.
@@ -939,7 +940,7 @@
 
     export formatResult=$formatResult$(${CAT} <<EOF
 
-    - path: /etc/systemd/network/ifcfg-$ifCfg.network
+    - path: /etc/systemd/network/10-$ifCfg.network
       permissions: 0644
       content: |
         [Match]
