diff -ru gosc-scripts/imc-shell/imgcust-scripts/PhotonCustomization.sh gosc-scripts-modify/imc-shell/imgcust-scripts/PhotonCustomization.sh
--- gosc-scripts/imc-shell/imgcust-scripts/PhotonCustomization.sh	2015-09-01 14:19:34.538883000 -0700
+++ gosc-scripts-modify/imc-shell/imgcust-scripts/PhotonCustomization.sh	2015-09-01 17:57:14.490367830 -0700
@@ -44,8 +44,14 @@
   else
     Exec "/usr/bin/timedatectl set-timezone $tz" '' ''
   fi
-  
-  sleep 2
+ 
+  # Avoiding race between systemd cloud-init and GOSC cloud-init
+  Exec "systemctl --no-reload disable cloud-config.service >/dev/null 2>&1"
+  Exec "systemctl --no-reload disable cloud-final.service  >/dev/null 2>&1"
+  Exec "systemctl --no-reload disable cloud-init.service   >/dev/null 2>&1"
+  Exec "systemctl --no-reload disable cloud-init-local.service >/dev/null 2>&1"
+
+
   rm -rf /var/lib/cloud/instance*
 
   GenerateCloudInitConfig content "GetInterfaceByMacAddressIPAddrShow" $tmpOvfEnvFile
