diff -ru gosc-scripts/imc-shell/imgcust-scripts/PhotonCustomization.sh gosc-scripts-modified/imc-shell/imgcust-scripts/PhotonCustomization.sh
--- gosc-scripts/imc-shell/imgcust-scripts/PhotonCustomization.sh	2016-04-29 14:10:39.000000000 -0700
+++ gosc-scripts-modified/imc-shell/imgcust-scripts/PhotonCustomization.sh	2016-06-23 18:09:34.130130006 -0700
@@ -45,6 +45,7 @@
   
   sleep 2
   rm -rf /var/lib/cloud/instance*
+  rm -rf /var/lib/cloud/seed
 
   GenerateCloudInitConfig content "GetInterfaceByMacAddressIPAddrShow" $tmpOvfEnvFile
   # Once the cloud-config file been generated, remove the existing network config files.
