diff -ru gosc-scripts/imc-shell/imgcust-scripts/ConfigFile.sh gosc-scripts-modify/imc-shell/imgcust-scripts/ConfigFile.sh
--- gosc-scripts/imc-shell/imgcust-scripts/ConfigFile.sh	2015-09-11 12:51:24.157078000 -0700
+++ gosc-scripts-modify/imc-shell/imgcust-scripts/ConfigFile.sh	2015-09-11 13:02:42.872916802 -0700
@@ -610,6 +610,19 @@
   echo "$val"
 }
 
+# Retrieves Lightwave PSE Contoller IP.
+#
+# Args:
+#   None
+# Results:
+#   string: PSE Controller IP or empty string in case it's missing or empty
+# Throws:
+#   None
+ConfigFile_GetLWDOMAINIP()
+{
+  echo "$(ConfigFile_GetOptionalString 'LIGHTWAVE|DOMAINIP')"
+}
+
 # Retrieves Lightwave DNS1 IP.
 #
 # Args:
diff -ru gosc-scripts/imc-shell/imgcust-scripts/CustomizationUtils.sh gosc-scripts-modify/imc-shell/imgcust-scripts/CustomizationUtils.sh
--- gosc-scripts/imc-shell/imgcust-scripts/CustomizationUtils.sh	2015-09-11 12:51:24.157078000 -0700
+++ gosc-scripts-modify/imc-shell/imgcust-scripts/CustomizationUtils.sh	2015-09-11 16:24:55.252033382 -0700
@@ -687,6 +687,7 @@
   local lw_dns2=$(ConfigFile_GetLWDNS2)
   local lw_dns3=$(ConfigFile_GetLWDNS3)
   local query_string=''
+  local lwdomain_ip=$(ConfigFile_GetLWDOMAINIP)
   
   Debug "DNS1 $lw_dns1"
   Debug "DNS2 $lw_dns2"
@@ -703,6 +704,12 @@
   local domain_name=$(ConfigFile_GetLWDomain)
   local user_name=$(ConfigFile_GetLWDomainUserName)
   local password=$(ConfigFile_GetLWDomainPwd)
+
+  if [[ -n "$domain_name" ]] && [[ -n "$lwdomain_ip" ]]; then
+    local patch_info="$lwdomain_ip    $domain_name"
+    echo "$patch_info" >> /etc/hosts
+  fi  
+  
   
   # default username is administrator, we will overwrite it if other username is specified 
   
@@ -729,8 +736,11 @@
     if [[ -n "$optional" ]]; then
       execute="$execute $optional"
     fi
+    # sometime vmafd is not running by default, so restarting it.
+    Exec "/opt/likewise/bin/lwsm restart vmafd"   
+
     Debug "Going to execute $execute"
-    $execute    
+    Exec "$execute"    
     Debug "Light wave domain join procedure completed"
   else
     Debug "ERROR :: Either Domain Name or Password Missing: Cannot Join Lighwave domain"
