--- open-vm-tools-9.10.0-2476743.orig/configure.ac	2015-03-19 15:53:25.000000000 -0700
+++ open-vm-tools-9.10.0-2476743/configure.ac	2015-04-21 01:23:00.561586272 -0700
@@ -1041,6 +1041,7 @@
    CPPFLAGS="$CPPFLAGS -D_XOPEN_SOURCE=500"
    CPPFLAGS="$CPPFLAGS -D_BSD_SOURCE"
    CPPFLAGS="$CPPFLAGS -D_SVID_SOURCE"
+   CPPFLAGS="$CPPFLAGS -D_DEFAULT_SOURCE"
 
    LIBVMTOOLS_LIBADD="$LIBVMTOOLS_LIBADD -lrt"
 
--- open-vm-tools-9.10.0-2476743.orig/vgauth/common/VGAuthLog.c	2015-03-19 15:53:25.000000000 -0700
+++ open-vm-tools-9.10.0-2476743/vgauth/common/VGAuthLog.c	2015-04-21 01:24:32.370582529 -0700
@@ -25,6 +25,11 @@
 #ifdef _WIN32
 #include <windows.h>
 #else
+/*
+ * Need GNU definition of strerror_r for better compatibility
+ * across different glibc versions.
+ */
+#define _GNU_SOURCE
 #include <errno.h>
 #include <unistd.h>
 #include <string.h>
