diff -Naur open-vm-tools.orig/lib/wiper/wiperPosix.c open-vm-tools/lib/wiper/wiperPosix.c
--- open-vm-tools.orig/lib/wiper/wiperPosix.c	2016-09-29 03:07:52.000000000 -0700
+++ open-vm-tools/lib/wiper/wiperPosix.c	2017-03-24 13:13:29.118630435 -0700
@@ -29,6 +29,7 @@
 
 #include <stdio.h>
 #include <sys/stat.h>
+#include <sys/sysmacros.h>
 #if defined(__linux__) || defined(sun)
 # include <sys/vfs.h>
 #elif defined(__FreeBSD__) || defined(__APPLE__)
