From 2930d06d7fbd4d5ed66137a7a5861ffb67d9a3a3 Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <sshedi@vmware.com>
Date: Mon, 28 Mar 2022 20:30:58 +0530
Subject: [PATCH 04/35] contrib virt

Signed-off-by: Shreenidhi Shedi <sshedi@vmware.com>
---
 policy/modules/contrib/virt.te | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/policy/modules/contrib/virt.te b/policy/modules/contrib/virt.te
index 340056b..66423fb 100644
--- a/policy/modules/contrib/virt.te
+++ b/policy/modules/contrib/virt.te
@@ -200,7 +200,7 @@ gen_tunable(virt_qemu_ga_read_nonsecurity_files, false)
 
 virt_domain_template(svirt)
 role system_r types svirt_t;
-typealias svirt_t alias qemu_t;
+#typealias svirt_t alias qemu_t;
 
 virt_domain_template(svirt_tcg)
 role system_r types svirt_tcg_t;
-- 
2.45.0

