From 23c3b9976ab58267dba8d4aa61bd78aeb5920a0c Mon Sep 17 00:00:00 2001
From: Clark Williams <williams@redhat.com>
Date: Fri, 3 Sep 2021 10:24:35 -0500
Subject: [PATCH 338/342] Linux 4.19.206-rt87 REBASE

Signed-off-by: Clark Williams <williams@redhat.com>
---
 localversion-rt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/localversion-rt b/localversion-rt
index c1f2720eaf25..bd5aee366a48 100644
--- a/localversion-rt
+++ b/localversion-rt
@@ -1 +1 @@
--rt76
+-rt87
-- 
2.33.1

