From 71e3e7c5d4a16dbaa6d08a781fbe91de1ee49d11 Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <sshedi@vmware.com>
Date: Mon, 28 Mar 2022 20:44:31 +0530
Subject: [PATCH 12/39] system systemd

Signed-off-by: Shreenidhi Shedi <sshedi@vmware.com>
---
 policy/modules/system/systemd.te | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/policy/modules/system/systemd.te b/policy/modules/system/systemd.te
index 2321a68..10fcc23 100644
--- a/policy/modules/system/systemd.te
+++ b/policy/modules/system/systemd.te
@@ -175,7 +175,6 @@ systemd_unit_file(systemd_socket_proxyd_unit_file_t)
 systemd_domain_template(systemd_timedated)
 init_nnp_daemon_domain(systemd_timedated_t)
 typeattribute systemd_timedated_t systemd_domain;
-typealias systemd_timedated_t alias gnomeclock_t;
 
 type systemd_timedated_unit_file_t;
 systemd_unit_file(systemd_timedated_unit_file_t)
@@ -1072,12 +1071,14 @@ allow systemd_timedated_t systemd_networkd_var_run_t:dir watch_dir_perms;
 
 list_dirs_pattern(systemd_timedated_t, systemd_networkd_var_run_t, systemd_networkd_var_run_t)
 read_files_pattern(systemd_timedated_t, systemd_networkd_var_run_t, systemd_networkd_var_run_t)
+read_files_pattern(systemd_timedated_t, systemd_resolved_var_run_t, systemd_resolved_var_run_t)
 
 corecmd_exec_bin(systemd_timedated_t)
 corecmd_exec_shell(systemd_timedated_t)
 corecmd_dontaudit_access_check_bin(systemd_timedated_t)
 
 corenet_tcp_connect_time_port(systemd_timedated_t)
+corenet_udp_bind_generic_node(systemd_timedated_t)
 
 dev_rw_realtime_clock(systemd_timedated_t)
 dev_write_kmsg(systemd_timedated_t)
@@ -1413,6 +1414,7 @@ read_files_pattern(systemd_resolved_t, systemd_networkd_var_run_t, systemd_netwo
 allow systemd_resolved_t systemd_networkd_var_run_t:dir watch_dir_perms;
 
 kernel_dgram_send(systemd_resolved_t)
+logging_dgram_send(systemd_resolved_t)
 kernel_read_net_sysctls(systemd_resolved_t)
 kernel_read_network_state(systemd_resolved_t)
 
@@ -1426,6 +1428,11 @@ corenet_tcp_connect_llmnr_port(systemd_resolved_t)
 corenet_udp_bind_dns_port(systemd_resolved_t)
 corenet_tcp_bind_dns_port(systemd_resolved_t)
 corenet_udp_bind_howl_port(systemd_resolved_t)
+corenet_udp_bind_generic_node(systemd_resolved_t)
+corenet_tcp_bind_generic_node(systemd_resolved_t)
+
+allow systemd_resolved_t self:udp_socket create_socket_perms;
+allow systemd_resolved_t self:tcp_socket create_socket_perms;
 
 dev_write_kmsg(systemd_resolved_t)
 dev_read_sysfs(systemd_resolved_t)
@@ -1537,6 +1544,7 @@ fs_rw_tracefs_files(systemd_modules_load_t)
 modutils_exec_kmod(systemd_modules_load_t)
 modutils_read_module_config(systemd_modules_load_t)
 modutils_read_module_deps_files(systemd_modules_load_t)
+logging_dgram_send(systemd_modules_load_t)
 
 systemd_read_efivarfs(systemd_modules_load_t)
 
-- 
2.45.2

