From 1b0e1cca75ebbda87ca40c12ebfc1b1989490335 Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <sshedi@vmware.com>
Date: Mon, 28 Mar 2022 20:13:41 +0530
Subject: [PATCH 16/39] admin_usermanage

Signed-off-by: Shreenidhi Shedi <sshedi@vmware.com>
---
 policy/modules/admin/usermanage.te | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/admin/usermanage.te b/policy/modules/admin/usermanage.te
index 6640310..4d1e878 100644
--- a/policy/modules/admin/usermanage.te
+++ b/policy/modules/admin/usermanage.te
@@ -331,6 +331,8 @@ allow passwd_t self:netlink_selinux_socket create_socket_perms;
 
 allow passwd_t crack_db_t:dir list_dir_perms;
 read_files_pattern(passwd_t, crack_db_t, crack_db_t)
+# pw_dict.pwd is a symlink to words.pwd
+read_lnk_files_pattern(passwd_t, crack_db_t, crack_db_t)
 
 kernel_read_kernel_sysctls(passwd_t)
 
-- 
2.45.2

