From e34293cbdcf4f3474c574fa2daafaf1612004b7e Mon Sep 17 00:00:00 2001
From: Alexey Makhalov <amakhalov@vmware.com>
Date: Thu, 3 Mar 2022 00:01:30 +0000
Subject: [PATCH 19/35] iptables: allow kernel_t fifo_files

---
 policy/modules/system/iptables.te | 1 +
 1 file changed, 1 insertion(+)

diff --git a/policy/modules/system/iptables.te b/policy/modules/system/iptables.te
index 3374bff..f38b69a 100644
--- a/policy/modules/system/iptables.te
+++ b/policy/modules/system/iptables.te
@@ -41,6 +41,7 @@ allow iptables_t self:cap_userns { dac_read_search  net_admin net_raw };
 dontaudit iptables_t self:capability { sys_admin sys_tty_config };
 dontaudit iptables_t self:capability2 block_suspend;
 allow iptables_t self:fifo_file rw_fifo_file_perms;
+allow iptables_t kernel_t:fifo_file rw_fifo_file_perms;
 allow iptables_t self:process { sigchld sigkill sigstop signull signal };
 allow iptables_t self:netlink_socket create_socket_perms;
 allow iptables_t self:netlink_generic_socket create_socket_perms;
-- 
2.45.0

