From b4618d074f7c31beb6fa7aeeca73db71a736dfdd Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <sshedi@vmware.com>
Date: Mon, 28 Mar 2022 20:30:24 +0530
Subject: [PATCH 05/20] contrib dbus

Signed-off-by: Shreenidhi Shedi <sshedi@vmware.com>
---
 policy/modules/contrib/dbus.fc | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/policy/modules/contrib/dbus.fc b/policy/modules/contrib/dbus.fc
index 1c1db3d..68bff69 100644
--- a/policy/modules/contrib/dbus.fc
+++ b/policy/modules/contrib/dbus.fc
@@ -32,6 +32,10 @@ ifdef(`distro_gentoo',`
 /var/run/user/%{USERID}/dbus(/.*)? 	gen_context(system_u:object_r:session_dbusd_tmp_t,s0)
 /var/run/user/%{USERID}/dbus-1(/.*)? 	gen_context(system_u:object_r:session_dbusd_tmp_t,s0)
 
+/run/dbus(/.*)?        gen_context(system_u:object_r:system_dbusd_var_run_t,s0)
+
+/run/user/%{USERID}/dbus(/.*)?    gen_context(system_u:object_r:session_dbusd_tmp_t,s0)
+
 ifdef(`distro_redhat',`
 /var/named/chroot/var/run/dbus(/.*)?	gen_context(system_u:object_r:system_dbusd_var_run_t,s0)
 ')
-- 
2.35.1

