diff -aur selinux-policy-ad1d35503f55f535401daa0a59913aa559c38d44/policy/modules/kernel/storage.fc selinux-policy-ad1d35503f55f535401daa0a59913aa559c38d44_2/policy/modules/kernel/storage.fc
--- selinux-policy-ad1d35503f55f535401daa0a59913aa559c38d44/policy/modules/kernel/storage.fc	2020-04-09 16:24:32.000000000 +0000
+++ selinux-policy-ad1d35503f55f535401daa0a59913aa559c38d44_2/policy/modules/kernel/storage.fc	2020-04-24 06:23:07.979181662 +0000
@@ -49,6 +49,9 @@
 ifdef(`distro_redhat', `
 /dev/root		-b	gen_context(system_u:object_r:fixed_disk_device_t,mls_systemhigh)
 ')
+ifdef(`distro_photon', `
+/dev/root		-b	gen_context(system_u:object_r:fixed_disk_device_t,mls_systemhigh)
+')
 /dev/s(cd|r)[^/]*	-b	gen_context(system_u:object_r:removable_device_t,s0)
 /dev/sbpcd.*		-b	gen_context(system_u:object_r:removable_device_t,s0)
 /dev/sg[0-9]+		-c	gen_context(system_u:object_r:scsi_generic_device_t,s0)
