From bf6d9eece0a89b9706aed2d71ba887a6a0003450 Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <sshedi@vmware.com>
Date: Mon, 28 Mar 2022 20:43:22 +0530
Subject: [PATCH 16/20] system modutils

Signed-off-by: Shreenidhi Shedi <sshedi@vmware.com>
---
 policy/modules/system/modutils.fc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/system/modutils.fc b/policy/modules/system/modutils.fc
index fe436a7..b3b24bf 100644
--- a/policy/modules/system/modutils.fc
+++ b/policy/modules/system/modutils.fc
@@ -3,6 +3,7 @@
 /etc/modules\.conf.*	--	gen_context(system_u:object_r:modules_conf_t,s0)
 /etc/modprobe\.conf.*	--	gen_context(system_u:object_r:modules_conf_t,s0)
 /etc/modprobe\.d(/.*)?		gen_context(system_u:object_r:modules_conf_t,s0)
+/etc/modules-load\.d(/.*)?	gen_context(system_u:object_r:modules_conf_t,s0)
 
 ifdef(`distro_gentoo',`
 # gentoo init scripts still manage this file
@@ -33,5 +34,6 @@ ifdef(`distro_gentoo',`
 /usr/lib/modules/[^/]+/modules\..+ --	gen_context(system_u:object_r:modules_dep_t,s0)
 
 /usr/lib/modules/modprobe\.conf -- 	gen_context(system_u:object_r:modules_conf_t,s0)
+/usr/lib/modules-load\.d(/.*)?	gen_context(system_u:object_r:modules_conf_t,s0)
 
 /var/run/tmpfiles.d/kmod.conf --	gen_context(system_u:object_r:kmod_var_run_t,s0)
-- 
2.35.1

