From 0b040d7a0577d442d96b7ed680bc59ea9fa56515 Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <sshedi@vmware.com>
Date: Mon, 28 Mar 2022 20:41:24 +0530
Subject: [PATCH 13/20] system getty

Signed-off-by: Shreenidhi Shedi <sshedi@vmware.com>
---
 policy/modules/system/getty.te | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/policy/modules/system/getty.te b/policy/modules/system/getty.te
index 03748d1..a89b4b2 100644
--- a/policy/modules/system/getty.te
+++ b/policy/modules/system/getty.te
@@ -130,6 +130,10 @@ ifdef(`distro_ubuntu',`
 	')
 ')
 
+ifdef(`distro_photon',`
+	allow getty_t self:capability sys_admin;
+')
+
 tunable_policy(`login_console_enabled',`
 	# Support logging in from /dev/console
 	term_use_console(getty_t)
-- 
2.35.1

