diff -aur selinux-policy-db2614cce37245ba4fc5de73b1f6f33cc46686e4/policy/modules/system/sysnetwork.fc selinux-policy-db2614cce37245ba4fc5de73b1f6f33cc46686e4_2/policy/modules/system/sysnetwork.fc
--- selinux-policy-db2614cce37245ba4fc5de73b1f6f33cc46686e4/policy/modules/system/sysnetwork.fc	2021-09-02 11:10:24.508147768 +0530
+++ selinux-policy-db2614cce37245ba4fc5de73b1f6f33cc46686e4_2/policy/modules/system/sysnetwork.fc	2021-09-02 11:14:20.432141597 +0530
@@ -40,6 +40,14 @@
 /var/run/systemd/resolve/resolv\.conf   --  gen_context(system_u:object_r:net_conf_t,s0)
 /var/run/systemd/resolve/stub-resolv\.conf  gen_context(system_u:object_r:net_conf_t,s0)
 ')
+ifdef(`distro_photon',`
+/etc/sysconfig/network-scripts/.*resolv\.conf -- gen_context(system_u:object_r:net_conf_t,s0)
+/etc/sysconfig/networking(/.*)? gen_context(system_u:object_r:net_conf_t,s0)
+/etc/sysconfig/network-scripts(/.*)? gen_context(system_u:object_r:net_conf_t,s0)
+/var/run/systemd/network(/.*)?  gen_context(system_u:object_r:net_conf_t,s0)
+/var/run/systemd/resolve/resolv\.conf   --  gen_context(system_u:object_r:net_conf_t,s0)
+/var/run/systemd/resolve/stub-resolv\.conf   --  gen_context(system_u:object_r:net_conf_t,s0)
+')
 /var/run/NetworkManager/resolv\.conf.*   --  gen_context(system_u:object_r:net_conf_t,s0)
 
 /var/run/cloud-init(/.*)?     gen_context(system_u:object_r:net_conf_t,s0)
diff -aur selinux-policy-db2614cce37245ba4fc5de73b1f6f33cc46686e4/policy/modules/system/sysnetwork.if selinux-policy-db2614cce37245ba4fc5de73b1f6f33cc46686e4_2/policy/modules/system/sysnetwork.if
--- selinux-policy-db2614cce37245ba4fc5de73b1f6f33cc46686e4/policy/modules/system/sysnetwork.if	2021-09-02 11:10:24.508147768 +0530
+++ selinux-policy-db2614cce37245ba4fc5de73b1f6f33cc46686e4_2/policy/modules/system/sysnetwork.if	2021-09-02 11:11:35.976145899 +0530
@@ -450,6 +450,12 @@
 		list_dirs_pattern($1, net_conf_t, net_conf_t)
 	')
 
+	ifdef(`distro_photon',`
+                files_search_all_pids($1)
+                init_search_pid_dirs($1)
+                list_dirs_pattern($1, net_conf_t, net_conf_t)
+        ')
+
 	ifdef(`distro_redhat',`
 		files_search_all_pids($1)
 		init_search_pid_dirs($1)
