diff --git a/src/a/linux.c b/src/b/linux.c
index 1231ecf..eb63e22 100644
--- a/src/linux.c
+++ b/src/linux.c
@@ -22,7 +22,12 @@
 #include <inttypes.h>
 #include <limits.h>
 #include <linux/ethtool.h>
+#include <linux/version.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 4, 0)
+#include <linux/nvme_ioctl.h>
+#else
 #include <linux/nvme.h>
+#endif
 #include <linux/sockios.h>
 #include <net/if.h>
 #include <scsi/scsi.h>
